// zudoku/dist/vite/output.js
import assert from "node:assert";
import { mkdir, readFile, writeFile } from "node:fs/promises";
import path from "node:path";
import { fileURLToPath } from "node:url";
import { joinUrl } from "../lib/util/joinUrl.js";
const __dirname = path.dirname(fileURLToPath(import.meta.url));
const pkgJsonPath = path.join(__dirname, "../../package.json");
const pkgJson = JSON.parse(await readFile(pkgJsonPath, "utf-8"));
export function generateOutput({ config, redirects, }) {
    const routes = [];
    for (const redirect of redirects) {
        routes.push({
            src: redirect.from,
            dest: redirect.to,
            status: 301,
            headers: { Location: redirect.to },
        });
    }
    if (process.env.VERCEL_SKEW_PROTECTION_ENABLED) {
        assert(process.env.VERCEL_DEPLOYMENT_ID);
        routes.push({
            src: "/.*",
            has: [
                {
                    type: "header",
                    key: "Sec-Fetch-Dest",
                    value: "document",
                },
            ],
            headers: {
                "Set-Cookie": `__vdpl=${process.env.VERCEL_DEPLOYMENT_ID}; Path=${joinUrl(config.basePath)}; SameSite=Strict; Secure; HttpOnly`,
            },
            continue: true,
        });
    }
    const output = {
        version: 3,
        framework: {
            version: pkgJson.version,
        },
        routes,
    };
    return output;
}
export async function writeOutput(dir, { config, redirects, }) {
    const output = generateOutput({ config, redirects });
    // For now we are putting this in the dist folder, eventually we can
    // expand this to support the full vercel build output API
    const outputDir = process.env.VERCEL
        ? path.join(dir, ".vercel/output")
        : path.join(dir, "dist/.output");
    await mkdir(outputDir, { recursive: true });
    const outputFile = path.join(outputDir, "config.json");
    await writeFile(outputFile, JSON.stringify(output, null, 2), "utf-8");
    if (process.env.VERCEL) {
        // eslint-disable-next-line no-console
        console.log("Wrote Vercel output to", outputDir);
    }
}
