// zudoku/lib/ui/Accordion.js
import { j as o } from "../jsx-runtime-C5mzlN2N.js";
import * as e from "@radix-ui/react-accordion";
import { ChevronDown as c } from "lucide-react";
import * as n from "react";
import { c as i } from "../cn-qaFjX9_3.js";
const N = e.Root, m = n.forwardRef(({ className: a, ...t }, r) => /* @__PURE__ */ o.jsx(
  e.Item,
  {
    ref: r,
    className: i("border-b", a),
    ...t
  }
));
m.displayName = "AccordionItem";
const d = n.forwardRef(({ className: a, children: t, ...r }, s) => /* @__PURE__ */ o.jsx(e.Header, { className: "flex", children: /* @__PURE__ */ o.jsxs(
  e.Trigger,
  {
    ref: s,
    className: i(
      "flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180",
      a
    ),
    ...r,
    children: [
      t,
      /* @__PURE__ */ o.jsx(c, { className: "h-4 w-4 shrink-0 transition-transform duration-200" })
    ]
  }
) }));
d.displayName = e.Trigger.displayName;
const l = n.forwardRef(({ className: a, children: t, ...r }, s) => /* @__PURE__ */ o.jsx(
  e.Content,
  {
    ref: s,
    className: "overflow-hidden text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down",
    ...r,
    children: /* @__PURE__ */ o.jsx("div", { className: i("pb-4 pt-0", a), children: t })
  }
));
l.displayName = e.Content.displayName;
export {
  N as Accordion,
  l as AccordionContent,
  m as AccordionItem,
  d as AccordionTrigger
};
