// zudoku/lib/ui/Alert.js
import { j as a } from "../jsx-runtime-C5mzlN2N.js";
import { c as l } from "../index-CPNSgwSb.js";
import * as s from "react";
import { c as i } from "../cn-qaFjX9_3.js";
const d = l(
  "relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-foreground",
  {
    variants: {
      variant: {
        default: "bg-background text-foreground",
        destructive: "border-destructive/50 text-destructive dark:border-destructive [&>svg]:text-destructive"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
), n = s.forwardRef(({ className: t, variant: e, ...r }, o) => /* @__PURE__ */ a.jsx(
  "div",
  {
    ref: o,
    role: "alert",
    className: i(d({ variant: e }), t),
    ...r
  }
));
n.displayName = "Alert";
const c = s.forwardRef(({ className: t, ...e }, r) => /* @__PURE__ */ a.jsx(
  "h5",
  {
    ref: r,
    className: i("mb-1 font-medium leading-none tracking-tight", t),
    ...e
  }
));
c.displayName = "AlertTitle";
const v = s.forwardRef(({ className: t, ...e }, r) => /* @__PURE__ */ a.jsx(
  "div",
  {
    ref: r,
    className: i("text-sm [&_p]:leading-relaxed", t),
    ...e
  }
));
v.displayName = "AlertDescription";
export {
  n as Alert,
  v as AlertDescription,
  c as AlertTitle
};
