// zudoku/lib/ui/Badge.js
import { j as o } from "../jsx-runtime-C5mzlN2N.js";
import { c as n } from "../index-CPNSgwSb.js";
import { c as a } from "../cn-qaFjX9_3.js";
const s = n(
  "inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-medium transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
  {
    variants: {
      variant: {
        default: "border-transparent bg-primary text-primary-foreground hover:bg-primary/80",
        secondary: "border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80",
        muted: "bg-muted text-muted-foreground ",
        destructive: "border-transparent bg-destructive text-destructive-foreground hover:bg-destructive/80",
        outline: "text-foreground"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
function c({ className: r, variant: e, ...t }) {
  return /* @__PURE__ */ o.jsx("div", { className: a(s({ variant: e }), r), ...t });
}
export {
  c as Badge,
  s as badgeVariants
};
