// zudoku/lib/ui/Button.js
import { j as d } from "../jsx-runtime-C5mzlN2N.js";
import { Slot as i } from "@radix-ui/react-slot";
import { c as u } from "../index-CPNSgwSb.js";
import * as c from "react";
import { c as m } from "../cn-qaFjX9_3.js";
const f = u(
  "not-prose inline-flex shrink-0 items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50",
  {
    variants: {
      variant: {
        default: "bg-primary text-primary-foreground shadow hover:bg-primary/90",
        destructive: "bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90",
        outline: "border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground",
        secondary: "bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80",
        ghost: "hover:bg-accent hover:text-accent-foreground",
        link: "text-primary underline-offset-4 hover:underline",
        expand: "flex gap-1.5 border bg-transparent rounded-xl text-muted-foreground hover:text-foreground"
      },
      size: {
        default: "h-9 px-4 py-2",
        sm: "h-8 rounded-md px-3 text-xs",
        lg: "h-10 rounded-md px-8",
        xl: "h-14 rounded-lg px-10 text-lg",
        icon: "h-9 w-9"
      }
    },
    defaultVariants: {
      variant: "default",
      size: "default"
    }
  }
), l = c.forwardRef(
  ({ className: e, variant: r, size: t, asChild: o = !1, ...n }, s) => {
    const a = o ? i : "button";
    return /* @__PURE__ */ d.jsx(
      a,
      {
        className: m(f({ variant: r, size: t, className: e })),
        ref: s,
        ...n
      }
    );
  }
);
l.displayName = "Button";
export {
  l as Button,
  f as buttonVariants
};
