// zudoku/lib/ui/Card.js
import { j as s } from "../jsx-runtime-C5mzlN2N.js";
import * as d from "react";
import { c as o } from "../cn-qaFjX9_3.js";
const t = d.forwardRef(({ className: a, ...e }, r) => /* @__PURE__ */ s.jsx(
  "div",
  {
    ref: r,
    className: o(
      "rounded-xl border bg-card text-card-foreground shadow-sm",
      a
    ),
    ...e
  }
));
t.displayName = "Card";
const i = d.forwardRef(({ className: a, ...e }, r) => /* @__PURE__ */ s.jsx(
  "div",
  {
    ref: r,
    className: o("flex flex-col space-y-1.5 p-6", a),
    ...e
  }
));
i.displayName = "CardHeader";
const m = d.forwardRef(({ className: a, ...e }, r) => /* @__PURE__ */ s.jsx(
  "h3",
  {
    ref: r,
    className: o("font-semibold leading-none tracking-tight", a),
    ...e
  }
));
m.displayName = "CardTitle";
const c = d.forwardRef(({ className: a, ...e }, r) => /* @__PURE__ */ s.jsx(
  "p",
  {
    ref: r,
    className: o("text-sm text-muted-foreground", a),
    ...e
  }
));
c.displayName = "CardDescription";
const n = d.forwardRef(({ className: a, ...e }, r) => /* @__PURE__ */ s.jsx("div", { ref: r, className: o("p-6 pt-0", a), ...e }));
n.displayName = "CardContent";
const f = d.forwardRef(({ className: a, ...e }, r) => /* @__PURE__ */ s.jsx(
  "div",
  {
    ref: r,
    className: o("flex items-center p-6 pt-0", a),
    ...e
  }
));
f.displayName = "CardFooter";
export {
  t as Card,
  n as CardContent,
  c as CardDescription,
  f as CardFooter,
  i as CardHeader,
  m as CardTitle
};
