// zudoku/lib/ui/Command.js
import { j as o } from "../jsx-runtime-C5mzlN2N.js";
import { _ as s } from "../index-CuBIgTKC.js";
import { Search as n } from "lucide-react";
import * as d from "react";
import { D as p, a as l } from "../Dialog-ByYz4ABw.js";
import { c as t } from "../cn-qaFjX9_3.js";
const m = d.forwardRef(({ className: a, ...e }, r) => /* @__PURE__ */ o.jsx(
  s,
  {
    ref: r,
    className: t(
      "flex h-full w-full flex-col overflow-hidden rounded-md bg-popover text-popover-foreground",
      a
    ),
    ...e
  }
));
m.displayName = s.displayName;
const j = ({
  children: a,
  command: e,
  content: r,
  ...i
}) => /* @__PURE__ */ o.jsx(p, { ...i, children: /* @__PURE__ */ o.jsx(
  l,
  {
    ...r,
    className: t(
      "overflow-hidden p-0 shadow-lg top-[15vh] translate-y-[0%]",
      r == null ? void 0 : r.className
    ),
    "aria-describedby": void 0,
    children: /* @__PURE__ */ o.jsx(
      m,
      {
        ...e,
        className: t(
          "[&_[cmdk-group-heading]]:px-2 [&_[cmdk-group-heading]]:font-medium [&_[cmdk-group-heading]]:text-muted-foreground [&_[cmdk-group]:not([hidden])_~[cmdk-group]]:pt-0 [&_[cmdk-group]]:px-2 [&_[cmdk-input-wrapper]_svg]:h-5 [&_[cmdk-input-wrapper]_svg]:w-5 [&_[cmdk-input]]:h-12 [&_[cmdk-item]]:px-2 [&_[cmdk-item]]:py-3 [&_[cmdk-item]_svg]:h-5 [&_[cmdk-item]_svg]:w-5",
          e == null ? void 0 : e.className
        ),
        children: a
      }
    )
  }
) }), c = d.forwardRef(({ className: a, ...e }, r) => (
  // eslint-disable-next-line react/no-unknown-property
  /* @__PURE__ */ o.jsxs("div", { className: "flex items-center border-b px-3", "cmdk-input-wrapper": "", children: [
    /* @__PURE__ */ o.jsx(n, { className: "mr-2 h-4 w-4 shrink-0 opacity-50" }),
    /* @__PURE__ */ o.jsx(
      s.Input,
      {
        ref: r,
        className: t(
          "flex h-11 w-full rounded-md bg-transparent py-3 text-sm outline-none placeholder:text-muted-foreground disabled:cursor-not-allowed disabled:opacity-50",
          a
        ),
        ...e
      }
    )
  ] })
));
c.displayName = s.Input.displayName;
const u = d.forwardRef(({ className: a, ...e }, r) => (
  // eslint-disable-next-line react/no-unknown-property
  /* @__PURE__ */ o.jsx("div", { className: "flex items-center", "cmdk-input-wrapper": "", children: /* @__PURE__ */ o.jsx(
    s.Input,
    {
      ref: r,
      className: t(
        "flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none disabled:cursor-not-allowed disabled:opacity-50",
        "focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring",
        a
      ),
      ...e
    }
  ) })
));
u.displayName = s.Input.displayName;
const f = d.forwardRef(({ className: a, ...e }, r) => /* @__PURE__ */ o.jsx(
  s.List,
  {
    ref: r,
    className: t("max-h-[300px] overflow-y-auto overflow-x-hidden", a),
    ...e
  }
));
f.displayName = s.List.displayName;
const x = d.forwardRef((a, e) => /* @__PURE__ */ o.jsx(
  s.Empty,
  {
    ref: e,
    className: "py-6 text-center text-sm",
    ...a
  }
));
x.displayName = s.Empty.displayName;
const g = d.forwardRef(({ className: a, ...e }, r) => /* @__PURE__ */ o.jsx(
  s.Group,
  {
    ref: r,
    className: t(
      "overflow-hidden p-1 text-foreground [&_[cmdk-group-heading]]:px-2 [&_[cmdk-group-heading]]:py-1.5 [&_[cmdk-group-heading]]:text-xs [&_[cmdk-group-heading]]:font-medium [&_[cmdk-group-heading]]:text-muted-foreground",
      a
    ),
    ...e
  }
));
g.displayName = s.Group.displayName;
const h = d.forwardRef(({ className: a, ...e }, r) => /* @__PURE__ */ o.jsx(
  s.Separator,
  {
    ref: r,
    className: t("-mx-1 h-px bg-border", a),
    ...e
  }
));
h.displayName = s.Separator.displayName;
const y = d.forwardRef(({ className: a, ...e }, r) => /* @__PURE__ */ o.jsx(
  s.Item,
  {
    ref: r,
    className: t(
      "relative flex cursor-default gap-2 select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none data-[disabled=true]:pointer-events-none data-[selected='true']:bg-accent data-[selected=true]:text-accent-foreground data-[disabled=true]:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0",
      a
    ),
    ...e
  }
));
y.displayName = s.Item.displayName;
const N = ({
  className: a,
  ...e
}) => /* @__PURE__ */ o.jsx(
  "span",
  {
    className: t(
      "ml-auto text-xs tracking-widest text-muted-foreground",
      a
    ),
    ...e
  }
);
N.displayName = "CommandShortcut";
export {
  m as Command,
  j as CommandDialog,
  x as CommandEmpty,
  g as CommandGroup,
  u as CommandInlineInput,
  c as CommandInput,
  y as CommandItem,
  f as CommandList,
  h as CommandSeparator,
  N as CommandShortcut
};
