// zudoku/lib/ui/Form.js
import { j as i } from "../jsx-runtime-C5mzlN2N.js";
import { Slot as F } from "@radix-ui/react-slot";
import * as s from "react";
import { u, F as x, C as p } from "../index.esm-D2ZUREQN.js";
import { c as a } from "../cn-qaFjX9_3.js";
import { Label as I } from "./Label.js";
const $ = x, f = s.createContext(
  {}
), D = ({
  ...e
}) => /* @__PURE__ */ i.jsx(f.Provider, { value: { name: e.name }, children: /* @__PURE__ */ i.jsx(p, { ...e }) }), d = () => {
  const e = s.useContext(f), o = s.useContext(l), { getFieldState: r, formState: t } = u(), m = r(e.name, t);
  if (!e)
    throw new Error("useFormField should be used within <FormField>");
  const { id: n } = o;
  return {
    id: n,
    name: e.name,
    formItemId: `${n}-form-item`,
    formDescriptionId: `${n}-form-item-description`,
    formMessageId: `${n}-form-item-message`,
    ...m
  };
}, l = s.createContext(
  {}
), C = s.forwardRef(({ className: e, ...o }, r) => {
  const t = s.useId();
  return /* @__PURE__ */ i.jsx(l.Provider, { value: { id: t }, children: /* @__PURE__ */ i.jsx("div", { ref: r, className: a("space-y-2", e), ...o }) });
});
C.displayName = "FormItem";
const g = s.forwardRef(({ className: e, ...o }, r) => {
  const { error: t, formItemId: m } = d();
  return /* @__PURE__ */ i.jsx(
    I,
    {
      ref: r,
      className: a(t && "text-destructive", e),
      htmlFor: m,
      ...o
    }
  );
});
g.displayName = "FormLabel";
const j = s.forwardRef(({ ...e }, o) => {
  const { error: r, formItemId: t, formDescriptionId: m, formMessageId: n } = d();
  return /* @__PURE__ */ i.jsx(
    F,
    {
      ref: o,
      id: t,
      "aria-describedby": r ? `${m} ${n}` : `${m}`,
      "aria-invalid": !!r,
      ...e
    }
  );
});
j.displayName = "FormControl";
const v = s.forwardRef(({ className: e, ...o }, r) => {
  const { formDescriptionId: t } = d();
  return /* @__PURE__ */ i.jsx(
    "p",
    {
      ref: r,
      id: t,
      className: a("text-sm text-muted-foreground", e),
      ...o
    }
  );
});
v.displayName = "FormDescription";
const N = s.forwardRef(({ className: e, children: o, ...r }, t) => {
  const { error: m, formMessageId: n } = d(), c = m ? String(m.message) : o;
  return c ? /* @__PURE__ */ i.jsx(
    "p",
    {
      ref: t,
      id: n,
      className: a("text-sm font-medium text-destructive", e),
      ...r,
      children: c
    }
  ) : null;
});
N.displayName = "FormMessage";
export {
  $ as Form,
  j as FormControl,
  v as FormDescription,
  D as FormField,
  C as FormItem,
  g as FormLabel,
  N as FormMessage,
  d as useFormField
};
