// zudoku/lib/ui/Popover.js
import { j as t } from "../jsx-runtime-C5mzlN2N.js";
import * as o from "@radix-ui/react-popover";
import * as s from "react";
import { c as n } from "../cn-qaFjX9_3.js";
const l = o.Root, c = o.Trigger, m = s.forwardRef(({ className: e, align: a = "center", sideOffset: r = 4, ...d }, i) => /* @__PURE__ */ t.jsx(o.Portal, { children: /* @__PURE__ */ t.jsx(
  o.Content,
  {
    ref: i,
    align: a,
    sideOffset: r,
    className: n(
      "z-50 w-72 rounded-md border bg-popover p-4 text-popover-foreground shadow-md outline-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2",
      e
    ),
    ...d
  }
) }));
m.displayName = o.Content.displayName;
export {
  l as Popover,
  m as PopoverContent,
  c as PopoverTrigger
};
