// zudoku/lib/ui/Tabs.js
import { j as a } from "../jsx-runtime-C5mzlN2N.js";
import * as e from "@radix-ui/react-tabs";
import * as o from "react";
import { c as n } from "../cn-qaFjX9_3.js";
const m = e.Root, r = o.forwardRef(({ className: i, ...s }, t) => /* @__PURE__ */ a.jsx(
  e.List,
  {
    ref: t,
    className: n(
      "inline-flex h-9 items-center justify-center rounded-lg bg-muted p-1 text-muted-foreground",
      i
    ),
    ...s
  }
));
r.displayName = e.List.displayName;
const f = o.forwardRef(({ className: i, ...s }, t) => /* @__PURE__ */ a.jsx(
  e.Trigger,
  {
    ref: t,
    className: n(
      "inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:shadow",
      i
    ),
    ...s
  }
));
f.displayName = e.Trigger.displayName;
const c = o.forwardRef(({ className: i, ...s }, t) => /* @__PURE__ */ a.jsx(
  e.Content,
  {
    ref: t,
    className: n(
      "mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2",
      i
    ),
    ...s
  }
));
c.displayName = e.Content.displayName;
export {
  m as Tabs,
  c as TabsContent,
  r as TabsList,
  f as TabsTrigger
};
