// zudoku/dist/config/validators/InputSidebarSchema.js
import { z } from "zod";
const BadgeSchema = z.object({
    label: z.string(),
    color: z.enum([
        "green",
        "blue",
        "yellow",
        "red",
        "purple",
        "indigo",
        "gray",
        "outline",
    ]),
    placement: z.enum(["start", "end"]).optional(),
    invert: z.boolean().optional(),
});
export const BaseInputSidebarItemCategoryLinkDocSchema = z.object({
    type: z.literal("doc"),
    id: z.string(),
    label: z.string().optional(),
});
export const InputSidebarItemCategoryLinkDocSchema = z.union([
    BaseInputSidebarItemCategoryLinkDocSchema,
    z.string(),
]);
export const BaseInputSidebarItemDocSchema = z.object({
    type: z.literal("doc"),
    id: z.string(),
    icon: z.custom().optional(),
    label: z.string().optional(),
    badge: BadgeSchema.optional(),
});
export const InputSidebarItemDocSchema = z.union([
    BaseInputSidebarItemDocSchema,
    z.string(),
]);
export const InputSidebarItemLinkSchema = z.object({
    type: z.literal("link"),
    label: z.string(),
    href: z.string(),
    description: z.string().optional(),
    badge: BadgeSchema.optional(),
});
export const BaseInputSidebarItemCategorySchema = z.object({
    type: z.literal("category"),
    icon: z.custom().optional(),
    label: z.string(),
    description: z.string().optional(),
    collapsible: z.boolean().optional(),
    collapsed: z.boolean().optional(),
    link: InputSidebarItemCategoryLinkDocSchema.optional(),
});
export const InputSidebarItemSchema = z.lazy(() => z.union([
    InputSidebarItemDocSchema,
    InputSidebarItemLinkSchema,
    InputSidebarItemCategorySchema,
]));
export const InputSidebarItemCategorySchema = BaseInputSidebarItemCategorySchema.extend({
    items: InputSidebarItemSchema.array(),
});
export const InputSidebarSchema = InputSidebarItemSchema.array();
