// zudoku/lib/Dialog-ByYz4ABw.js
import { j as s } from "./jsx-runtime-C5mzlN2N.js";
import * as a from "@radix-ui/react-dialog";
import { X as r } from "lucide-react";
import * as l from "react";
import { c as i } from "./cn-qaFjX9_3.js";
const D = a.Root, j = a.Trigger, c = a.Portal, n = l.forwardRef(({ className: e, ...t }, o) => /* @__PURE__ */ s.jsx(
  a.Overlay,
  {
    ref: o,
    className: i(
      "fixed inset-0 z-50 bg-black/80  data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0",
      e
    ),
    ...t
  }
));
n.displayName = a.Overlay.displayName;
const m = l.forwardRef(({ className: e, children: t, ...o }, d) => /* @__PURE__ */ s.jsxs(c, { children: [
  /* @__PURE__ */ s.jsx(n, {}),
  /* @__PURE__ */ s.jsxs(
    a.Content,
    {
      ref: d,
      className: i(
        "fixed left-[50%] top-[50%] z-50 grid w-full max-w-lg translate-x-[-50%] translate-y-[-50%] gap-4 border bg-background p-6 shadow-lg duration-200 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%] data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%] sm:rounded-lg",
        e
      ),
      ...o,
      children: [
        t,
        /* @__PURE__ */ s.jsxs(a.Close, { className: "absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent data-[state=open]:text-muted-foreground", children: [
          /* @__PURE__ */ s.jsx(r, { className: "h-4 w-4" }),
          /* @__PURE__ */ s.jsx("span", { className: "sr-only", children: "Close" })
        ] })
      ]
    }
  )
] }));
m.displayName = a.Content.displayName;
const f = ({
  className: e,
  ...t
}) => /* @__PURE__ */ s.jsx(
  "div",
  {
    className: i(
      "flex flex-col space-y-1.5 text-center sm:text-left",
      e
    ),
    ...t
  }
);
f.displayName = "DialogHeader";
const p = ({
  className: e,
  ...t
}) => /* @__PURE__ */ s.jsx(
  "div",
  {
    className: i(
      "flex flex-col-reverse sm:flex-row sm:justify-end sm:space-x-2",
      e
    ),
    ...t
  }
);
p.displayName = "DialogFooter";
const g = l.forwardRef(({ className: e, ...t }, o) => /* @__PURE__ */ s.jsx(
  a.Title,
  {
    ref: o,
    className: i(
      "text-lg font-semibold leading-none tracking-tight",
      e
    ),
    ...t
  }
));
g.displayName = a.Title.displayName;
const x = l.forwardRef(({ className: e, ...t }, o) => /* @__PURE__ */ s.jsx(
  a.Description,
  {
    ref: o,
    className: i("text-sm text-muted-foreground", e),
    ...t
  }
));
x.displayName = a.Description.displayName;
export {
  D,
  m as a,
  g as b,
  f as c,
  x as d,
  p as e,
  j as f
};
