// zudoku/lib/index-tpbiZmWp.js
var nr = Object.defineProperty;
var De = (t) => {
  throw TypeError(t);
};
var sr = (t, e, r) => e in t ? nr(t, e, { enumerable: !0, configurable: !0, writable: !0, value: r }) : t[e] = r;
var A = (t, e, r) => sr(t, typeof e != "symbol" ? e + "" : e, r), qe = (t, e, r) => e.has(t) || De("Cannot " + r);
var d = (t, e, r) => (qe(t, e, "read from private field"), r ? r.call(t) : e.get(t)), R = (t, e, r) => e.has(t) ? De("Cannot add the same private member more than once") : e instanceof WeakSet ? e.add(t) : e.set(t, r), M = (t, e, r, s) => (qe(t, e, "write to private field"), s ? s.call(t, r) : e.set(t, r), r);
var ce = (t, e, r, s) => ({
  set _(a) {
    M(t, e, a, r);
  },
  get _() {
    return d(t, e, s);
  }
});
import { B as ot, a as it, H as Te, R as ar } from "./RouteGuard-BZ_VsiXc.js";
import { z as ct, J as or } from "./index-DwT-v3zK.js";
import { R as ir, S as cr, u as ge, f as pe, g as lr, L as Me, O as lt } from "./chunk-KNED5TY2-BUPjb3LQ.js";
import { S as ut, v as dt, Q as ur, n as k, w as $e, x as Ie, y as B, z as dr, A as fr, B as hr, C as mr, D as Fe, E as Qe, F as gr, h as ze, G as Le, H as pr, g as ft, I as ht, e as F, l as ae, J as mt, K as vr, f as ee, u as xr, j as yr, d as br, Z as jr, L as wr, M as Pr } from "./hook-pPrHCB6G.js";
import { E as gt, S as _, a as Sr, R as Cr } from "./SlotletProvider-VUmTNmLZ.js";
import { j as n } from "./jsx-runtime-C5mzlN2N.js";
import { Button as Er } from "./ui/Button.js";
import { Callout as Nr } from "./ui/Callout.js";
import { M as Tr } from "./mutation-Csa9eQDM.js";
import * as K from "react";
import he, { createContext as Oe, StrictMode as pt, useRef as G, useEffect as Q, useState as J, useCallback as W, Suspense as Ae, memo as vt, Component as Mr, createElement as Be, useMemo as be, useContext as Or } from "react";
import * as Ar from "react-dom";
import { C as ve } from "./ClientOnly-E7hGysn1.js";
import { c as O } from "./cn-qaFjX9_3.js";
import { A as Rr, s as kr, a as He, b as _r, H as Dr, P as qr, c as $r, V as Ir, M as Fr } from "./Markdown-C5h6bxbE.js";
import { ExternalLink as Qr, CircleXIcon as zr, ChevronRightIcon as Lr, SearchIcon as Br, SunIcon as Hr, MoonIcon as Kr, MenuIcon as Ur, PanelLeftIcon as Yr } from "lucide-react";
import { S as xt } from "./Spinner-mNLZ6awP.js";
import { C as Zr } from "./CategoryHeading-D06WK_Wo.js";
import { isSearchPlugin as Vr, isProfileMenuPlugin as Wr, isNavigationPlugin as Gr, isEventConsumerPlugin as Xr, needsInitialization as Jr, isApiIdentityPlugin as en, isMdxProviderPlugin as tn, hasHead as rn } from "./zudoku.plugins.js";
import { o as nn } from "./objectEntries-yMIkr2mI.js";
import { B as Ce } from "./Button-BBNrKpQd.js";
import { DropdownMenu as sn, DropdownMenuTrigger as an, DropdownMenuContent as on, DropdownMenuLabel as cn, DropdownMenuSeparator as je, DropdownMenuSub as ln, DropdownMenuSubTrigger as un, DropdownMenuPortal as dn, DropdownMenuSubContent as fn, DropdownMenuItem as hn } from "./ui/DropdownMenu.js";
import { VisuallyHidden as yt } from "@radix-ui/react-visually-hidden";
import { D as mn, a as gn, b as bt, c as jt, d as z } from "./Drawer-CEwbkLDb.js";
import { a as pn } from "./index-CPNSgwSb.js";
const Ee = {}, wt = he.createContext(Ee);
function Pt(t) {
  const e = he.useContext(wt);
  return he.useMemo(
    function() {
      return typeof t == "function" ? t(e) : { ...e, ...t };
    },
    [e, t]
  );
}
function vn(t) {
  let e;
  return t.disableParentContext ? e = typeof t.components == "function" ? t.components(Ee) : t.components || Ee : e = Pt(t.components), he.createElement(
    wt.Provider,
    { value: e },
    t.children
  );
}
var I, nt, xn = (nt = class extends ut {
  constructor(e = {}) {
    super();
    R(this, I);
    this.config = e, M(this, I, /* @__PURE__ */ new Map());
  }
  build(e, r, s) {
    const a = r.queryKey, i = r.queryHash ?? dt(a, r);
    let o = this.get(i);
    return o || (o = new ur({
      client: e,
      queryKey: a,
      queryHash: i,
      options: e.defaultQueryOptions(r),
      state: s,
      defaultOptions: e.getQueryDefaults(a)
    }), this.add(o)), o;
  }
  add(e) {
    d(this, I).has(e.queryHash) || (d(this, I).set(e.queryHash, e), this.notify({
      type: "added",
      query: e
    }));
  }
  remove(e) {
    const r = d(this, I).get(e.queryHash);
    r && (e.destroy(), r === e && d(this, I).delete(e.queryHash), this.notify({ type: "removed", query: e }));
  }
  clear() {
    k.batch(() => {
      this.getAll().forEach((e) => {
        this.remove(e);
      });
    });
  }
  get(e) {
    return d(this, I).get(e);
  }
  getAll() {
    return [...d(this, I).values()];
  }
  find(e) {
    const r = { exact: !0, ...e };
    return this.getAll().find(
      (s) => $e(r, s)
    );
  }
  findAll(e = {}) {
    const r = this.getAll();
    return Object.keys(e).length > 0 ? r.filter((s) => $e(e, s)) : r;
  }
  notify(e) {
    k.batch(() => {
      this.listeners.forEach((r) => {
        r(e);
      });
    });
  }
  onFocus() {
    k.batch(() => {
      this.getAll().forEach((e) => {
        e.onFocus();
      });
    });
  }
  onOnline() {
    k.batch(() => {
      this.getAll().forEach((e) => {
        e.onOnline();
      });
    });
  }
}, I = new WeakMap(), nt), H, $, oe, st, yn = (st = class extends ut {
  constructor(e = {}) {
    super();
    R(this, H);
    R(this, $);
    R(this, oe);
    this.config = e, M(this, H, /* @__PURE__ */ new Set()), M(this, $, /* @__PURE__ */ new Map()), M(this, oe, 0);
  }
  build(e, r, s) {
    const a = new Tr({
      mutationCache: this,
      mutationId: ++ce(this, oe)._,
      options: e.defaultMutationOptions(r),
      state: s
    });
    return this.add(a), a;
  }
  add(e) {
    d(this, H).add(e);
    const r = le(e);
    if (typeof r == "string") {
      const s = d(this, $).get(r);
      s ? s.push(e) : d(this, $).set(r, [e]);
    }
    this.notify({ type: "added", mutation: e });
  }
  remove(e) {
    if (d(this, H).delete(e)) {
      const r = le(e);
      if (typeof r == "string") {
        const s = d(this, $).get(r);
        if (s)
          if (s.length > 1) {
            const a = s.indexOf(e);
            a !== -1 && s.splice(a, 1);
          } else s[0] === e && d(this, $).delete(r);
      }
    }
    this.notify({ type: "removed", mutation: e });
  }
  canRun(e) {
    const r = le(e);
    if (typeof r == "string") {
      const s = d(this, $).get(r), a = s == null ? void 0 : s.find(
        (i) => i.state.status === "pending"
      );
      return !a || a === e;
    } else
      return !0;
  }
  runNext(e) {
    var s;
    const r = le(e);
    if (typeof r == "string") {
      const a = (s = d(this, $).get(r)) == null ? void 0 : s.find((i) => i !== e && i.state.isPaused);
      return (a == null ? void 0 : a.continue()) ?? Promise.resolve();
    } else
      return Promise.resolve();
  }
  clear() {
    k.batch(() => {
      d(this, H).forEach((e) => {
        this.notify({ type: "removed", mutation: e });
      }), d(this, H).clear(), d(this, $).clear();
    });
  }
  getAll() {
    return Array.from(d(this, H));
  }
  find(e) {
    const r = { exact: !0, ...e };
    return this.getAll().find(
      (s) => Ie(r, s)
    );
  }
  findAll(e = {}) {
    return this.getAll().filter((r) => Ie(e, r));
  }
  notify(e) {
    k.batch(() => {
      this.listeners.forEach((r) => {
        r(e);
      });
    });
  }
  resumePausedMutations() {
    const e = this.getAll().filter((r) => r.state.isPaused);
    return k.batch(
      () => Promise.all(
        e.map((r) => r.continue().catch(B))
      )
    );
  }
}, H = new WeakMap(), $ = new WeakMap(), oe = new WeakMap(), st);
function le(t) {
  var e;
  return (e = t.options.scope) == null ? void 0 : e.id;
}
function Ke(t) {
  return {
    onFetch: (e, r) => {
      var b, x, p, y, m;
      const s = e.options, a = (p = (x = (b = e.fetchOptions) == null ? void 0 : b.meta) == null ? void 0 : x.fetchMore) == null ? void 0 : p.direction, i = ((y = e.state.data) == null ? void 0 : y.pages) || [], o = ((m = e.state.data) == null ? void 0 : m.pageParams) || [];
      let f = { pages: [], pageParams: [] }, h = 0;
      const l = async () => {
        let g = !1;
        const C = (v) => {
          Object.defineProperty(v, "signal", {
            enumerable: !0,
            get: () => (e.signal.aborted ? g = !0 : e.signal.addEventListener("abort", () => {
              g = !0;
            }), e.signal)
          });
        }, N = dr(e.options, e.fetchOptions), T = async (v, j, c) => {
          if (g)
            return Promise.reject();
          if (j == null && v.pages.length)
            return Promise.resolve(v);
          const S = {
            client: e.client,
            queryKey: e.queryKey,
            pageParam: j,
            direction: c ? "backward" : "forward",
            meta: e.options.meta
          };
          C(S);
          const D = await N(
            S
          ), { maxPages: q } = e.options, L = c ? fr : hr;
          return {
            pages: L(v.pages, D, q),
            pageParams: L(v.pageParams, j, q)
          };
        };
        if (a && i.length) {
          const v = a === "backward", j = v ? bn : Ue, c = {
            pages: i,
            pageParams: o
          }, S = j(s, c);
          f = await T(c, S, v);
        } else {
          const v = t ?? i.length;
          do {
            const j = h === 0 ? o[0] ?? s.initialPageParam : Ue(s, f);
            if (h > 0 && j == null)
              break;
            f = await T(f, j), h++;
          } while (h < v);
        }
        return f;
      };
      e.options.persister ? e.fetchFn = () => {
        var g, C;
        return (C = (g = e.options).persister) == null ? void 0 : C.call(
          g,
          l,
          {
            client: e.client,
            queryKey: e.queryKey,
            meta: e.options.meta,
            signal: e.signal
          },
          r
        );
      } : e.fetchFn = l;
    }
  };
}
function Ue(t, { pages: e, pageParams: r }) {
  const s = e.length - 1;
  return e.length > 0 ? t.getNextPageParam(
    e[s],
    e,
    r[s],
    r
  ) : void 0;
}
function bn(t, { pages: e, pageParams: r }) {
  var s;
  return e.length > 0 ? (s = t.getPreviousPageParam) == null ? void 0 : s.call(t, e[0], e, r[0], r) : void 0;
}
var E, Y, Z, te, re, V, ne, se, at, jn = (at = class {
  constructor(t = {}) {
    R(this, E);
    R(this, Y);
    R(this, Z);
    R(this, te);
    R(this, re);
    R(this, V);
    R(this, ne);
    R(this, se);
    M(this, E, t.queryCache || new xn()), M(this, Y, t.mutationCache || new yn()), M(this, Z, t.defaultOptions || {}), M(this, te, /* @__PURE__ */ new Map()), M(this, re, /* @__PURE__ */ new Map()), M(this, V, 0);
  }
  mount() {
    ce(this, V)._++, d(this, V) === 1 && (M(this, ne, mr.subscribe(async (t) => {
      t && (await this.resumePausedMutations(), d(this, E).onFocus());
    })), M(this, se, Fe.subscribe(async (t) => {
      t && (await this.resumePausedMutations(), d(this, E).onOnline());
    })));
  }
  unmount() {
    var t, e;
    ce(this, V)._--, d(this, V) === 0 && ((t = d(this, ne)) == null || t.call(this), M(this, ne, void 0), (e = d(this, se)) == null || e.call(this), M(this, se, void 0));
  }
  isFetching(t) {
    return d(this, E).findAll({ ...t, fetchStatus: "fetching" }).length;
  }
  isMutating(t) {
    return d(this, Y).findAll({ ...t, status: "pending" }).length;
  }
  /**
   * Imperative (non-reactive) way to retrieve data for a QueryKey.
   * Should only be used in callbacks or functions where reading the latest data is necessary, e.g. for optimistic updates.
   *
   * Hint: Do not use this function inside a component, because it won't receive updates.
   * Use `useQuery` to create a `QueryObserver` that subscribes to changes.
   */
  getQueryData(t) {
    var r;
    const e = this.defaultQueryOptions({ queryKey: t });
    return (r = d(this, E).get(e.queryHash)) == null ? void 0 : r.state.data;
  }
  ensureQueryData(t) {
    const e = this.defaultQueryOptions(t), r = d(this, E).build(this, e), s = r.state.data;
    return s === void 0 ? this.fetchQuery(t) : (t.revalidateIfStale && r.isStaleByTime(Qe(e.staleTime, r)) && this.prefetchQuery(e), Promise.resolve(s));
  }
  getQueriesData(t) {
    return d(this, E).findAll(t).map(({ queryKey: e, state: r }) => {
      const s = r.data;
      return [e, s];
    });
  }
  setQueryData(t, e, r) {
    const s = this.defaultQueryOptions({ queryKey: t }), a = d(this, E).get(
      s.queryHash
    ), i = a == null ? void 0 : a.state.data, o = gr(e, i);
    if (o !== void 0)
      return d(this, E).build(this, s).setData(o, { ...r, manual: !0 });
  }
  setQueriesData(t, e, r) {
    return k.batch(
      () => d(this, E).findAll(t).map(({ queryKey: s }) => [
        s,
        this.setQueryData(s, e, r)
      ])
    );
  }
  getQueryState(t) {
    var r;
    const e = this.defaultQueryOptions({ queryKey: t });
    return (r = d(this, E).get(
      e.queryHash
    )) == null ? void 0 : r.state;
  }
  removeQueries(t) {
    const e = d(this, E);
    k.batch(() => {
      e.findAll(t).forEach((r) => {
        e.remove(r);
      });
    });
  }
  resetQueries(t, e) {
    const r = d(this, E);
    return k.batch(() => (r.findAll(t).forEach((s) => {
      s.reset();
    }), this.refetchQueries(
      {
        type: "active",
        ...t
      },
      e
    )));
  }
  cancelQueries(t, e = {}) {
    const r = { revert: !0, ...e }, s = k.batch(
      () => d(this, E).findAll(t).map((a) => a.cancel(r))
    );
    return Promise.all(s).then(B).catch(B);
  }
  invalidateQueries(t, e = {}) {
    return k.batch(() => (d(this, E).findAll(t).forEach((r) => {
      r.invalidate();
    }), (t == null ? void 0 : t.refetchType) === "none" ? Promise.resolve() : this.refetchQueries(
      {
        ...t,
        type: (t == null ? void 0 : t.refetchType) ?? (t == null ? void 0 : t.type) ?? "active"
      },
      e
    )));
  }
  refetchQueries(t, e = {}) {
    const r = {
      ...e,
      cancelRefetch: e.cancelRefetch ?? !0
    }, s = k.batch(
      () => d(this, E).findAll(t).filter((a) => !a.isDisabled()).map((a) => {
        let i = a.fetch(void 0, r);
        return r.throwOnError || (i = i.catch(B)), a.state.fetchStatus === "paused" ? Promise.resolve() : i;
      })
    );
    return Promise.all(s).then(B);
  }
  fetchQuery(t) {
    const e = this.defaultQueryOptions(t);
    e.retry === void 0 && (e.retry = !1);
    const r = d(this, E).build(this, e);
    return r.isStaleByTime(
      Qe(e.staleTime, r)
    ) ? r.fetch(e) : Promise.resolve(r.state.data);
  }
  prefetchQuery(t) {
    return this.fetchQuery(t).then(B).catch(B);
  }
  fetchInfiniteQuery(t) {
    return t.behavior = Ke(t.pages), this.fetchQuery(t);
  }
  prefetchInfiniteQuery(t) {
    return this.fetchInfiniteQuery(t).then(B).catch(B);
  }
  ensureInfiniteQueryData(t) {
    return t.behavior = Ke(t.pages), this.ensureQueryData(t);
  }
  resumePausedMutations() {
    return Fe.isOnline() ? d(this, Y).resumePausedMutations() : Promise.resolve();
  }
  getQueryCache() {
    return d(this, E);
  }
  getMutationCache() {
    return d(this, Y);
  }
  getDefaultOptions() {
    return d(this, Z);
  }
  setDefaultOptions(t) {
    M(this, Z, t);
  }
  setQueryDefaults(t, e) {
    d(this, te).set(ze(t), {
      queryKey: t,
      defaultOptions: e
    });
  }
  getQueryDefaults(t) {
    const e = [...d(this, te).values()], r = {};
    return e.forEach((s) => {
      Le(t, s.queryKey) && Object.assign(r, s.defaultOptions);
    }), r;
  }
  setMutationDefaults(t, e) {
    d(this, re).set(ze(t), {
      mutationKey: t,
      defaultOptions: e
    });
  }
  getMutationDefaults(t) {
    const e = [...d(this, re).values()], r = {};
    return e.forEach((s) => {
      Le(t, s.mutationKey) && Object.assign(r, s.defaultOptions);
    }), r;
  }
  defaultQueryOptions(t) {
    if (t._defaulted)
      return t;
    const e = {
      ...d(this, Z).queries,
      ...this.getQueryDefaults(t.queryKey),
      ...t,
      _defaulted: !0
    };
    return e.queryHash || (e.queryHash = dt(
      e.queryKey,
      e
    )), e.refetchOnReconnect === void 0 && (e.refetchOnReconnect = e.networkMode !== "always"), e.throwOnError === void 0 && (e.throwOnError = !!e.suspense), !e.networkMode && e.persister && (e.networkMode = "offlineFirst"), e.queryFn === pr && (e.enabled = !1), e;
  }
  defaultMutationOptions(t) {
    return t != null && t._defaulted ? t : {
      ...d(this, Z).mutations,
      ...(t == null ? void 0 : t.mutationKey) && this.getMutationDefaults(t.mutationKey),
      ...t,
      _defaulted: !0
    };
  }
  clear() {
    d(this, E).clear(), d(this, Y).clear();
  }
}, E = new WeakMap(), Y = new WeakMap(), Z = new WeakMap(), te = new WeakMap(), re = new WeakMap(), V = new WeakMap(), ne = new WeakMap(), se = new WeakMap(), at);
function wn(t) {
  return t;
}
function Ye(t, e, r) {
  var h, l;
  if (typeof e != "object" || e === null)
    return;
  const s = t.getMutationCache(), a = t.getQueryCache(), i = ((h = r == null ? void 0 : r.defaultOptions) == null ? void 0 : h.deserializeData) ?? ((l = t.getDefaultOptions().hydrate) == null ? void 0 : l.deserializeData) ?? wn, o = e.mutations || [], f = e.queries || [];
  o.forEach(({ state: b, ...x }) => {
    var p, y;
    s.build(
      t,
      {
        ...(p = t.getDefaultOptions().hydrate) == null ? void 0 : p.mutations,
        ...(y = r == null ? void 0 : r.defaultOptions) == null ? void 0 : y.mutations,
        ...x
      },
      b
    );
  }), f.forEach(({ queryKey: b, state: x, queryHash: p, meta: y, promise: m }) => {
    var N, T;
    let g = a.get(p);
    const C = x.data === void 0 ? x.data : i(x.data);
    if (g) {
      if (g.state.dataUpdatedAt < x.dataUpdatedAt) {
        const { fetchStatus: v, ...j } = x;
        g.setState({
          ...j,
          data: C
        });
      }
    } else
      g = a.build(
        t,
        {
          ...(N = t.getDefaultOptions().hydrate) == null ? void 0 : N.queries,
          ...(T = r == null ? void 0 : r.defaultOptions) == null ? void 0 : T.queries,
          queryKey: b,
          queryHash: p,
          meta: y
        },
        // Reset fetch status to idle to avoid
        // query being stuck in fetching state upon hydration
        {
          ...x,
          data: C,
          fetchStatus: "idle"
        }
      );
    if (m) {
      const v = Promise.resolve(m).then(i);
      g.fetch(void 0, { initialPromise: v });
    }
  });
}
var Ze = (t, e) => typeof t == "object" && t !== null && e in t, Pn = ({
  children: t,
  options: e = {},
  state: r,
  queryClient: s
}) => {
  const a = ft(s), [i, o] = K.useState(), f = K.useRef(e);
  return f.current = e, K.useMemo(() => {
    if (r) {
      if (typeof r != "object")
        return;
      const h = a.getQueryCache(), l = r.queries || [], b = [], x = [];
      for (const p of l) {
        const y = h.get(p.queryHash);
        if (!y)
          b.push(p);
        else {
          const m = p.state.dataUpdatedAt > y.state.dataUpdatedAt || // RSC special serialized then-able chunks
          Ze(p.promise, "status") && Ze(y.promise, "status") && p.promise.status !== y.promise.status, g = i == null ? void 0 : i.find(
            (C) => C.queryHash === p.queryHash
          );
          m && (!g || p.state.dataUpdatedAt > g.state.dataUpdatedAt) && x.push(p);
        }
      }
      b.length > 0 && Ye(a, { queries: b }, f.current), x.length > 0 && o(
        (p) => p ? [...p, ...x] : x
      );
    }
  }, [a, i, r]), K.useEffect(() => {
    i && (Ye(a, { queries: i }, f.current), o(void 0));
  }, [a, i]), t;
};
function Sn({ error: t }) {
  return /* @__PURE__ */ n.jsx(gt, { error: t });
}
/**
 * react-router v7.5.0
 *
 * Copyright (c) Remix Software Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.md file in the root directory of this source tree.
 *
 * @license MIT
 */
function Cn(t) {
  return /* @__PURE__ */ K.createElement(ir, { flushSync: Ar.flushSync, ...t });
}
const Ne = Oe({ stagger: !1 }), En = new jn({
  defaultOptions: {
    queries: {
      staleTime: 1e3 * 60 * 5
    }
  }
}), Nn = ({
  router: t,
  hydrate: e = !1
}) => /* @__PURE__ */ n.jsx(pt, { children: /* @__PURE__ */ n.jsx(ht, { client: En, children: /* @__PURE__ */ n.jsx(Pn, { state: e ? window.DATA : void 0, children: /* @__PURE__ */ n.jsx(ot, { value: !1, children: /* @__PURE__ */ n.jsx(it, { children: /* @__PURE__ */ n.jsx(Ne.Provider, { value: { stagger: !e }, children: /* @__PURE__ */ n.jsx(Cn, { router: t }) }) }) }) }) }) }), Tn = ({
  router: t,
  context: e,
  queryClient: r,
  helmetContext: s,
  bypassProtection: a = !1
}) => /* @__PURE__ */ n.jsx(pt, { children: /* @__PURE__ */ n.jsx(ht, { client: r, children: /* @__PURE__ */ n.jsx(it, { context: s, children: /* @__PURE__ */ n.jsx(ot, { value: a, children: /* @__PURE__ */ n.jsx(cr, { router: t, context: e }) }) }) }) }), Mn = () => {
  const t = ge(), e = G(t.pathname);
  Q(() => {
    const r = e.current !== t.pathname, s = t.hash !== "";
    r && !s && window.scrollTo(0, 0), e.current = t.pathname;
  }, [t.pathname, t.hash]);
}, On = ({
  icon: t
}) => typeof t == "string" ? /* @__PURE__ */ n.jsx(
  "img",
  {
    src: `https://cdn.simpleicons.org/${t}/000000/ffffff`,
    className: "size-5",
    alt: t
  }
) : t, An = (t) => /^https?:/.test(t), Rn = () => {
  var r;
  const { page: t } = ys(), e = t == null ? void 0 : t.footer;
  return e ? /* @__PURE__ */ n.jsx("footer", { className: "border-t bg-background", children: /* @__PURE__ */ n.jsxs("div", { className: "mx-auto max-w-screen-2xl px-4 lg:px-8 py-8 pt-20", children: [
    /* @__PURE__ */ n.jsxs(
      "div",
      {
        className: O("flex flex-row gap-8", {
          "justify-center": !e.position || e.position === "center",
          "justify-start": e.position === "start",
          "justify-end": e.position === "end"
        }),
        children: [
          /* @__PURE__ */ n.jsx(_, { name: "footer-before" }),
          e.columns && /* @__PURE__ */ n.jsx(
            "div",
            {
              className: "w-full md:max-w-screen-md grid grid-cols-[1fr_1fr] gap-8 md:grid-cols-[repeat(var(--columns),minmax(0,1fr))]",
              style: { "--columns": e.columns.length },
              children: e.columns.map((s) => /* @__PURE__ */ n.jsxs(
                "div",
                {
                  className: O({
                    "justify-self-center": !s.position || s.position === "center",
                    "justify-self-start": s.position === "start",
                    "justify-self-end": s.position === "end"
                  }),
                  children: [
                    /* @__PURE__ */ n.jsx("span", { className: "text-sm font-semibold", children: s.title }),
                    /* @__PURE__ */ n.jsx("ul", { className: "mt-4 space-y-2", children: s.links.map((a, i) => {
                      const o = "flex flex-row gap-1 items-center text-sm text-muted-foreground hover:text-accent-foreground";
                      return /* @__PURE__ */ n.jsx("li", { children: An(a.href) ? /* @__PURE__ */ n.jsxs(
                        "a",
                        {
                          href: a.href,
                          target: "_blank",
                          rel: "noopener noreferrer",
                          className: o,
                          children: [
                            /* @__PURE__ */ n.jsx("span", { children: a.label }),
                            /* @__PURE__ */ n.jsx(Qr, { size: 12 })
                          ]
                        }
                      ) : /* @__PURE__ */ n.jsx(
                        Rr,
                        {
                          to: a.href + i,
                          className: o,
                          children: /* @__PURE__ */ n.jsx("span", { children: a.label })
                        }
                      ) }, a.href + i);
                    }) })
                  ]
                },
                s.title
              ))
            }
          ),
          /* @__PURE__ */ n.jsx(_, { name: "footer-after" })
        ]
      }
    ),
    /* @__PURE__ */ n.jsxs(
      "div",
      {
        className: O(
          "flex items-center justify-between",
          e.columns && "border-t mt-8 pt-8"
        ),
        children: [
          e.logo && /* @__PURE__ */ n.jsxs(n.Fragment, { children: [
            /* @__PURE__ */ n.jsx(
              "img",
              {
                src: e.logo.src.light,
                alt: e.logo.alt,
                className: "w-8 dark:hidden",
                style: { width: e.logo.width }
              }
            ),
            /* @__PURE__ */ n.jsx(
              "img",
              {
                src: e.logo.src.dark,
                alt: e.logo.alt,
                className: "w-8 hidden dark:block",
                style: { width: e.logo.width }
              }
            )
          ] }),
          e.copyright && /* @__PURE__ */ n.jsx("p", { className: "text-sm text-muted-foreground", children: e.copyright }),
          /* @__PURE__ */ n.jsx("div", { className: "flex items-center gap-2", children: (r = e.social) == null ? void 0 : r.map((s) => /* @__PURE__ */ n.jsxs(
            "a",
            {
              href: s.href,
              target: "_blank",
              rel: "noopener noreferrer",
              className: "w-auto gap-2 flex text-muted-foreground hover:text-accent-foreground",
              children: [
                /* @__PURE__ */ n.jsx(On, { icon: s.icon }),
                s.label
              ]
            },
            s.href
          )) })
        ]
      }
    )
  ] }) }) : null;
};
function kn({
  className: t,
  ...e
}) {
  return /* @__PURE__ */ n.jsx(
    "div",
    {
      className: O("animate-pulse rounded-md bg-muted", t),
      ...e
    }
  );
}
const Ve = {
  info: "bg-blue-500",
  note: "bg-gray-500",
  tip: "bg-green-600",
  caution: "bg-orange-500",
  danger: "bg-rose-500"
}, _n = () => {
  const { page: t } = F(), [e, r] = J(!0);
  if (!(t != null && t.banner) || !e)
    return /* @__PURE__ */ n.jsx("style", { children: ":root { --banner-height: 0px; }" });
  const s = t.banner.color && t.banner.color in Ve ? Ve[t.banner.color] : t.banner.color ? void 0 : "bg-primary", a = s ? {} : { backgroundColor: t.banner.color };
  return /* @__PURE__ */ n.jsxs(
    "div",
    {
      className: O(
        "relative text-primary-foreground text-sm font-medium px-4 py-2 flex gap-2 items-center",
        s
      ),
      "data-pagefind-ignore": "all",
      style: a,
      children: [
        /* @__PURE__ */ n.jsx("div", { className: "w-full", children: t.banner.message }),
        t.banner.dismissible && /* @__PURE__ */ n.jsx(
          "button",
          {
            type: "button",
            className: "md:absolute md:right-4 -m-1.5 p-1.5 hover:bg-accent-foreground/10 rounded-md",
            onClick: () => r(!1),
            children: /* @__PURE__ */ n.jsx(zr, { size: 16 })
          }
        )
      ]
    }
  );
}, St = (t) => /* @__PURE__ */ n.jsxs(
  "svg",
  {
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 132 100",
    fill: "none",
    ...t,
    children: [
      /* @__PURE__ */ n.jsx(
        "path",
        {
          fill: "currentColor",
          fillRule: "evenodd",
          d: "M80.092 2.963A4.66 4.66 0 0 1 84.449 0h34.049c6.325 0 10.835 6.135 8.948 12.172L116.653 46.71a4.688 4.688 0 0 1-4.474 3.29H75c-5.178 0-7.813 4.687-9.375 9.374-1.288 3.864-11.07 28.963-14.467 37.662A4.66 4.66 0 0 1 46.801 100H12.75c-6.324 0-10.834-6.134-8.947-12.171l10.793-34.54A4.688 4.688 0 0 1 19.071 50H56.25c5.178 0 7.813-4.687 9.375-9.375 1.288-3.864 11.07-28.962 14.467-37.662Z",
          clipRule: "evenodd"
        }
      ),
      /* @__PURE__ */ n.jsx(
        "path",
        {
          fill: "currentColor",
          d: "M83.54 57.813a7.813 7.813 0 0 0-7.316 5.07L63.888 95.777c-.766 2.043.744 4.222 2.926 4.222h36.828c5.211 0 9.875-3.232 11.704-8.11l8.821-23.522c1.915-5.107-1.861-10.555-7.315-10.555H83.539ZM29.17 0a12.5 12.5 0 0 0-11.704 8.111l-8.82 23.521c-1.915 5.107 1.86 10.556 7.315 10.556h33.312a7.813 7.813 0 0 0 7.316-5.07L68.924 4.223C69.691 2.18 68.18 0 65.998 0H29.17Z"
        }
      )
    ]
  }
);
St.displayName = "ZudokuLogo";
const Ct = ({ className: t }) => /* @__PURE__ */ n.jsxs(
  "a",
  {
    href: "https://zudoku.dev",
    target: "_blank",
    rel: "noopener noreferrer",
    className: O(
      "flex justify-between items-center w-full border border-transparent hover:border-border rounded-full hover:shadow-sm h-7 px-3 text-nowrap hover:bg-muted/80 transition-all",
      t
    ),
    children: [
      /* @__PURE__ */ n.jsxs("div", { className: "opacity-70 hover:opacity-100 transition-opacity gap-1.5 text-[11px] font-medium rounded-full h-7 flex items-center text-nowrap", children: [
        /* @__PURE__ */ n.jsx(St, { className: "w-3.5 h-3.5 dark:fill-white" }),
        "powered by Zudoku"
      ] }),
      /* @__PURE__ */ n.jsx("div", { className: "text-xs font-medium opacity-70 hover:text-foreground transition-colors cursor-pointer", children: /* @__PURE__ */ n.jsx(Lr, { size: 12, absoluteStrokeWidth: !0, strokeWidth: 1.5 }) })
    ]
  }
);
function Dn() {
  const t = navigator.userAgent.toLowerCase();
  if (t.includes("win")) return "Windows";
  if (t.includes("mac")) return "macOS";
  if (t.includes("linux")) return "Linux";
}
const Et = ({ className: t }) => {
  const e = F(), [r, s] = J(!1), a = W(() => s(!1), []);
  Q(() => {
    if (r)
      return;
    function o(f) {
      f.key === "k" && (f.metaKey || f.ctrlKey) && (f.preventDefault(), s(!0));
    }
    return window.addEventListener("keydown", o), () => {
      window.removeEventListener("keydown", o);
    };
  }, [r, s]);
  const i = e.plugins.find(Vr);
  return i ? /* @__PURE__ */ n.jsxs("div", { className: t, children: [
    /* @__PURE__ */ n.jsxs(
      "button",
      {
        type: "button",
        onClick: () => s(!0),
        className: "flex items-center border border-input hover:bg-accent hover:text-accent-foreground p-4 relative h-8 justify-start rounded-lg bg-background text-sm text-muted-foreground shadow-none w-full sm:w-72",
        children: [
          /* @__PURE__ */ n.jsxs("div", { className: "flex items-center gap-2 flex-grow", children: [
            /* @__PURE__ */ n.jsx(Br, { size: 14 }),
            "Search"
          ] }),
          /* @__PURE__ */ n.jsx(ve, { children: /* @__PURE__ */ n.jsx(qn, {}) })
        ]
      }
    ),
    /* @__PURE__ */ n.jsx(Ae, { fallback: null, children: i.renderSearch({
      isOpen: r,
      onClose: a
    }) })
  ] }) : null;
}, qn = () => {
  const t = Dn();
  return /* @__PURE__ */ n.jsxs("kbd", { className: "absolute right-[0.3rem] top-[0.35rem] hidden h-5 select-none items-center gap-1 rounded border bg-muted px-1.5 font-mono text-[11px] font-medium opacity-100 sm:flex", children: [
    t === "macOS" ? "⌘" : "Ctrl",
    "+K"
  ] });
}, Nt = () => {
  const { resolvedTheme: t, setTheme: e } = ct();
  return /* @__PURE__ */ n.jsx(ve, { fallback: /* @__PURE__ */ n.jsx(Ce, { variant: "ghost", size: "icon" }), children: /* @__PURE__ */ n.jsxs(
    "button",
    {
      type: "button",
      className: "flex rounded-full border p-0.5 gap-0.5 group",
      onClick: () => e(t === "dark" ? "light" : "dark"),
      "aria-label": t === "dark" ? "Switch to light mode" : "Switch to dark mode",
      children: [
        /* @__PURE__ */ n.jsx(
          "div",
          {
            className: O(
              "border border-transparent rounded-full p-0.5 [&>svg>circle]:transition-colors [&>svg>path]:transition-transform transition-all [&>svg>path]:duration-200 [&>svg>circle]:duration-500 [&>svg>circle]:fill-transparent",
              t === "light" && "border-border bg-muted",
              t === "dark" && "group-hover:[&>svg>path]:scale-110 group-hover:[&>svg>path]:-translate-x-[1px] group-hover:[&>svg>path]:-translate-y-[1px] group-hover:rotate-[15deg] "
            ),
            children: /* @__PURE__ */ n.jsx(Hr, { size: 16 })
          }
        ),
        /* @__PURE__ */ n.jsx(
          "div",
          {
            className: O(
              "border border-transparent rounded-full p-0.5 transition-transform transform-gpu duration-500",
              t === "dark" && "border-border bg-muted [&>svg>path]:fill-white [&>svg>path]:stroke-transparent",
              t === "light" && "group-hover:rotate-[-10deg] [&>svg>path]:stroke-currentColor"
            ),
            children: /* @__PURE__ */ n.jsx(Kr, { size: 16 })
          }
        )
      ]
    }
  ) });
};
function me() {
  return me = Object.assign ? Object.assign.bind() : function(t) {
    for (var e = 1; e < arguments.length; e++) {
      var r = arguments[e];
      for (var s in r) ({}).hasOwnProperty.call(r, s) && (t[s] = r[s]);
    }
    return t;
  }, me.apply(null, arguments);
}
var ue = { exports: {} }, w = {};
/** @license React v16.13.1
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var We;
function $n() {
  if (We) return w;
  We = 1;
  var t = typeof Symbol == "function" && Symbol.for, e = t ? Symbol.for("react.element") : 60103, r = t ? Symbol.for("react.portal") : 60106, s = t ? Symbol.for("react.fragment") : 60107, a = t ? Symbol.for("react.strict_mode") : 60108, i = t ? Symbol.for("react.profiler") : 60114, o = t ? Symbol.for("react.provider") : 60109, f = t ? Symbol.for("react.context") : 60110, h = t ? Symbol.for("react.async_mode") : 60111, l = t ? Symbol.for("react.concurrent_mode") : 60111, b = t ? Symbol.for("react.forward_ref") : 60112, x = t ? Symbol.for("react.suspense") : 60113, p = t ? Symbol.for("react.suspense_list") : 60120, y = t ? Symbol.for("react.memo") : 60115, m = t ? Symbol.for("react.lazy") : 60116, g = t ? Symbol.for("react.block") : 60121, C = t ? Symbol.for("react.fundamental") : 60117, N = t ? Symbol.for("react.responder") : 60118, T = t ? Symbol.for("react.scope") : 60119;
  function v(c) {
    if (typeof c == "object" && c !== null) {
      var S = c.$$typeof;
      switch (S) {
        case e:
          switch (c = c.type, c) {
            case h:
            case l:
            case s:
            case i:
            case a:
            case x:
              return c;
            default:
              switch (c = c && c.$$typeof, c) {
                case f:
                case b:
                case m:
                case y:
                case o:
                  return c;
                default:
                  return S;
              }
          }
        case r:
          return S;
      }
    }
  }
  function j(c) {
    return v(c) === l;
  }
  return w.AsyncMode = h, w.ConcurrentMode = l, w.ContextConsumer = f, w.ContextProvider = o, w.Element = e, w.ForwardRef = b, w.Fragment = s, w.Lazy = m, w.Memo = y, w.Portal = r, w.Profiler = i, w.StrictMode = a, w.Suspense = x, w.isAsyncMode = function(c) {
    return j(c) || v(c) === h;
  }, w.isConcurrentMode = j, w.isContextConsumer = function(c) {
    return v(c) === f;
  }, w.isContextProvider = function(c) {
    return v(c) === o;
  }, w.isElement = function(c) {
    return typeof c == "object" && c !== null && c.$$typeof === e;
  }, w.isForwardRef = function(c) {
    return v(c) === b;
  }, w.isFragment = function(c) {
    return v(c) === s;
  }, w.isLazy = function(c) {
    return v(c) === m;
  }, w.isMemo = function(c) {
    return v(c) === y;
  }, w.isPortal = function(c) {
    return v(c) === r;
  }, w.isProfiler = function(c) {
    return v(c) === i;
  }, w.isStrictMode = function(c) {
    return v(c) === a;
  }, w.isSuspense = function(c) {
    return v(c) === x;
  }, w.isValidElementType = function(c) {
    return typeof c == "string" || typeof c == "function" || c === s || c === l || c === i || c === a || c === x || c === p || typeof c == "object" && c !== null && (c.$$typeof === m || c.$$typeof === y || c.$$typeof === o || c.$$typeof === f || c.$$typeof === b || c.$$typeof === C || c.$$typeof === N || c.$$typeof === T || c.$$typeof === g);
  }, w.typeOf = v, w;
}
var P = {};
/** @license React v16.13.1
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ge;
function In() {
  return Ge || (Ge = 1, process.env.NODE_ENV !== "production" && function() {
    var t = typeof Symbol == "function" && Symbol.for, e = t ? Symbol.for("react.element") : 60103, r = t ? Symbol.for("react.portal") : 60106, s = t ? Symbol.for("react.fragment") : 60107, a = t ? Symbol.for("react.strict_mode") : 60108, i = t ? Symbol.for("react.profiler") : 60114, o = t ? Symbol.for("react.provider") : 60109, f = t ? Symbol.for("react.context") : 60110, h = t ? Symbol.for("react.async_mode") : 60111, l = t ? Symbol.for("react.concurrent_mode") : 60111, b = t ? Symbol.for("react.forward_ref") : 60112, x = t ? Symbol.for("react.suspense") : 60113, p = t ? Symbol.for("react.suspense_list") : 60120, y = t ? Symbol.for("react.memo") : 60115, m = t ? Symbol.for("react.lazy") : 60116, g = t ? Symbol.for("react.block") : 60121, C = t ? Symbol.for("react.fundamental") : 60117, N = t ? Symbol.for("react.responder") : 60118, T = t ? Symbol.for("react.scope") : 60119;
    function v(u) {
      return typeof u == "string" || typeof u == "function" || // Note: its typeof might be other than 'symbol' or 'number' if it's a polyfill.
      u === s || u === l || u === i || u === a || u === x || u === p || typeof u == "object" && u !== null && (u.$$typeof === m || u.$$typeof === y || u.$$typeof === o || u.$$typeof === f || u.$$typeof === b || u.$$typeof === C || u.$$typeof === N || u.$$typeof === T || u.$$typeof === g);
    }
    function j(u) {
      if (typeof u == "object" && u !== null) {
        var ye = u.$$typeof;
        switch (ye) {
          case e:
            var ie = u.type;
            switch (ie) {
              case h:
              case l:
              case s:
              case i:
              case a:
              case x:
                return ie;
              default:
                var _e = ie && ie.$$typeof;
                switch (_e) {
                  case f:
                  case b:
                  case m:
                  case y:
                  case o:
                    return _e;
                  default:
                    return ye;
                }
            }
          case r:
            return ye;
        }
      }
    }
    var c = h, S = l, D = f, q = o, L = e, U = b, xe = s, X = m, Qt = y, zt = r, Lt = i, Bt = a, Ht = x, Re = !1;
    function Kt(u) {
      return Re || (Re = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 17+. Update your code to use ReactIs.isConcurrentMode() instead. It has the exact same API.")), ke(u) || j(u) === h;
    }
    function ke(u) {
      return j(u) === l;
    }
    function Ut(u) {
      return j(u) === f;
    }
    function Yt(u) {
      return j(u) === o;
    }
    function Zt(u) {
      return typeof u == "object" && u !== null && u.$$typeof === e;
    }
    function Vt(u) {
      return j(u) === b;
    }
    function Wt(u) {
      return j(u) === s;
    }
    function Gt(u) {
      return j(u) === m;
    }
    function Xt(u) {
      return j(u) === y;
    }
    function Jt(u) {
      return j(u) === r;
    }
    function er(u) {
      return j(u) === i;
    }
    function tr(u) {
      return j(u) === a;
    }
    function rr(u) {
      return j(u) === x;
    }
    P.AsyncMode = c, P.ConcurrentMode = S, P.ContextConsumer = D, P.ContextProvider = q, P.Element = L, P.ForwardRef = U, P.Fragment = xe, P.Lazy = X, P.Memo = Qt, P.Portal = zt, P.Profiler = Lt, P.StrictMode = Bt, P.Suspense = Ht, P.isAsyncMode = Kt, P.isConcurrentMode = ke, P.isContextConsumer = Ut, P.isContextProvider = Yt, P.isElement = Zt, P.isForwardRef = Vt, P.isFragment = Wt, P.isLazy = Gt, P.isMemo = Xt, P.isPortal = Jt, P.isProfiler = er, P.isStrictMode = tr, P.isSuspense = rr, P.isValidElementType = v, P.typeOf = j;
  }()), P;
}
var Xe;
function Fn() {
  return Xe || (Xe = 1, process.env.NODE_ENV === "production" ? ue.exports = $n() : ue.exports = In()), ue.exports;
}
var we, Je;
function Qn() {
  if (Je) return we;
  Je = 1;
  var t = Fn(), e = {
    childContextTypes: !0,
    contextType: !0,
    contextTypes: !0,
    defaultProps: !0,
    displayName: !0,
    getDefaultProps: !0,
    getDerivedStateFromError: !0,
    getDerivedStateFromProps: !0,
    mixins: !0,
    propTypes: !0,
    type: !0
  }, r = {
    name: !0,
    length: !0,
    prototype: !0,
    caller: !0,
    callee: !0,
    arguments: !0,
    arity: !0
  }, s = {
    $$typeof: !0,
    render: !0,
    defaultProps: !0,
    displayName: !0,
    propTypes: !0
  }, a = {
    $$typeof: !0,
    compare: !0,
    defaultProps: !0,
    displayName: !0,
    propTypes: !0,
    type: !0
  }, i = {};
  i[t.ForwardRef] = s, i[t.Memo] = a;
  function o(m) {
    return t.isMemo(m) ? a : i[m.$$typeof] || e;
  }
  var f = Object.defineProperty, h = Object.getOwnPropertyNames, l = Object.getOwnPropertySymbols, b = Object.getOwnPropertyDescriptor, x = Object.getPrototypeOf, p = Object.prototype;
  function y(m, g, C) {
    if (typeof g != "string") {
      if (p) {
        var N = x(g);
        N && N !== p && y(m, N, C);
      }
      var T = h(g);
      l && (T = T.concat(l(g)));
      for (var v = o(m), j = o(g), c = 0; c < T.length; ++c) {
        var S = T[c];
        if (!r[S] && !(C && C[S]) && !(j && j[S]) && !(v && v[S])) {
          var D = b(g, S);
          try {
            f(m, S, D);
          } catch {
          }
        }
      }
    }
    return m;
  }
  return we = y, we;
}
Qn();
var Tt = function(e, r, s) {
  return e = e <= s ? e : s, e = e >= r ? e : r, e;
}, zn = function() {
  var e = !1, r = [], s = function() {
    e = !0;
    var f = r.shift();
    if (f)
      return f(s);
    e = !1;
  }, a = function() {
    e = !1, r = [];
  }, i = function(f) {
    r.push(f), !e && r.length === 1 && s();
  };
  return {
    clear: a,
    enqueue: i
  };
}, Ln = function() {
  var e, r = function() {
    e && window.cancelAnimationFrame(e);
  }, s = function(i, o) {
    var f, h, l = function(x) {
      if (h = h || x, f = x - h, f > o) {
        i();
        return;
      }
      e = window.requestAnimationFrame(l);
    };
    e = window.requestAnimationFrame(l);
  };
  return {
    cancel: r,
    schedule: s
  };
}, Bn = function(e) {
  var r = 0;
  return e >= 0 && e < 0.2 ? r = 0.1 : e >= 0.2 && e < 0.5 ? r = 0.04 : e >= 0.5 && e < 0.8 ? r = 0.02 : e >= 0.8 && e < 0.99 && (r = 5e-3), Tt(e + r, 0, 0.994);
}, et = function(e) {
  Q(e, []);
}, Hn = function(e) {
  return ++e % 1e6;
}, Kn = function() {
  var e = J(0), r = e[1];
  return W(function() {
    return r(Hn);
  }, []);
}, Un = function(e) {
  e === void 0 && (e = {});
  var r = Kn(), s = G(me({}, e)), a = W(function() {
    return s.current;
  }, []), i = W(function(o) {
    o && (Object.assign(s.current, o), r());
  }, []);
  return [a, i];
}, Yn = function() {
  var e = G(!0);
  return e.current ? (e.current = !1, !0) : e.current;
}, tt = function(e, r) {
  var s = Yn();
  Q(function() {
    if (!s)
      return e();
  }, r);
}, Mt = function() {
}, rt = {
  isFinished: !0,
  progress: 0,
  sideEffect: Mt
}, Zn = function(e) {
  var r = e === void 0 ? {} : e, s = r.animationDuration, a = s === void 0 ? 200 : s, i = r.incrementDuration, o = i === void 0 ? 800 : i, f = r.isAnimating, h = f === void 0 ? !1 : f, l = r.minimum, b = l === void 0 ? 0.08 : l, x = Un(rt), p = x[0], y = x[1], m = G(null), g = G(null);
  et(function() {
    m.current = zn(), g.current = Ln();
  });
  var C = W(function() {
    var S, D;
    (S = g.current) == null || S.cancel(), (D = m.current) == null || D.clear();
  }, []), N = W(function(S) {
    var D;
    if (S = Tt(S, b, 1), S === 1) {
      var q, L;
      C(), (q = m.current) == null || q.enqueue(function(U) {
        y({
          progress: S,
          sideEffect: function() {
            var X;
            return (X = g.current) == null ? void 0 : X.schedule(U, a);
          }
        });
      }), (L = m.current) == null || L.enqueue(function() {
        y({
          isFinished: !0,
          sideEffect: C
        });
      });
      return;
    }
    (D = m.current) == null || D.enqueue(function(U) {
      y({
        isFinished: !1,
        progress: S,
        sideEffect: function() {
          var X;
          return (X = g.current) == null ? void 0 : X.schedule(U, a);
        }
      });
    });
  }, [a, C, b, m, y, g]), T = W(function() {
    N(Bn(p().progress));
  }, [p, N]), v = W(function() {
    var S = function() {
      var q;
      T(), (q = m.current) == null || q.enqueue(function(L) {
        var U;
        (U = g.current) == null || U.schedule(function() {
          S(), L();
        }, o);
      });
    };
    S();
  }, [o, m, g, T]), j = G(Mt), c = p().sideEffect;
  return Q(function() {
    j.current = T;
  }), et(function() {
    return h && v(), C;
  }), tt(function() {
    p().sideEffect();
  }, [p, c]), tt(function() {
    h ? y(me({}, rt, {
      sideEffect: v
    })) : N(1);
  }, [h, N, y, v]), {
    animationDuration: a,
    isFinished: p().isFinished,
    progress: p().progress
  };
};
const Ot = (t) => (e) => e.display === "hide" ? !1 : e.display === "auth" && t || e.display === "anon" && !t || !e.display || e.display === "always", At = () => {
  const e = pe().state === "loading", [r, s] = J(!1);
  Q(() => {
    const o = setTimeout(() => s(e), 100);
    return () => clearTimeout(o);
  }, [e]);
  const { isFinished: a, progress: i } = Zn({ isAnimating: r });
  return /* @__PURE__ */ n.jsx(
    "div",
    {
      className: "absolute w-0 left-0 right-0 bottom-[-1px] h-[2px] bg-primary transition-all duration-300 ease-in-out",
      style: {
        opacity: a ? 0 : 1,
        width: a ? 0 : `${i * 100}%`
      }
    }
  );
}, Vn = () => {
  const { topNavigation: t } = F(), { isAuthenticated: e } = ae(), r = t.filter(Ot(e));
  return r.length === 0 ? /* @__PURE__ */ n.jsx("style", { children: ":root { --top-nav-height: 0px; }" }) : /* @__PURE__ */ n.jsxs(Ae, { children: [
    /* @__PURE__ */ n.jsxs("div", { className: "items-center justify-between px-8 h-[--top-nav-height] hidden lg:flex text-sm relative", children: [
      /* @__PURE__ */ n.jsx("nav", { className: "text-sm", children: /* @__PURE__ */ n.jsx("ul", { className: "flex flex-row items-center gap-8", children: r.map((s) => /* @__PURE__ */ n.jsx("li", { children: /* @__PURE__ */ n.jsx(Rt, { ...s }) }, s.id)) }) }),
      /* @__PURE__ */ n.jsx(_, { name: "top-navigation-side" })
    ] }),
    /* @__PURE__ */ n.jsx(At, {})
  ] });
}, Rt = ({
  id: t,
  label: e,
  default: r
}) => {
  var l;
  const { sidebars: s } = F(), a = s[t], i = mt(), o = !!pe().location, f = ((l = i.topNavItem) == null ? void 0 : l.id) === t && !o, h = r ?? (a ? vr(a, (b) => {
    if (b.type === "doc") return ee(b.id);
  }) : ee(t)) ?? ee(t);
  return (
    // We don't use isActive here because it has to be inside the sidebar,
    // the top nav id doesn't necessarily start with the sidebar id
    /* @__PURE__ */ n.jsx(
      lr,
      {
        className: ({ isPending: b }) => pn(
          "block lg:py-3.5 font-medium -mb-px",
          f || b ? "border-primary text-foreground" : "border-transparent text-foreground/75 hover:text-foreground hover:border-accent-foreground/25"
        ),
        to: h,
        children: e
      }
    )
  );
}, Wn = () => {
  var i;
  const { topNavigation: t, options: e } = F(), { isAuthenticated: r } = ae(), [s, a] = J(!1);
  return /* @__PURE__ */ n.jsxs(
    mn,
    {
      direction: "right",
      open: s,
      onOpenChange: (o) => a(o),
      children: [
        /* @__PURE__ */ n.jsxs("div", { className: "flex lg:hidden justify-self-end", children: [
          /* @__PURE__ */ n.jsx(gn, { className: "lg:hidden", children: /* @__PURE__ */ n.jsx(Ur, { size: 22 }) }),
          /* @__PURE__ */ n.jsx(At, {})
        ] }),
        /* @__PURE__ */ n.jsx(
          bt,
          {
            className: "lg:hidden h-[100dvh] right-0 left-auto w-[320px] rounded-none",
            "aria-describedby": void 0,
            children: /* @__PURE__ */ n.jsxs("div", { className: "p-4 overflow-y-auto overscroll-none h-full flex flex-col justify-between", children: [
              /* @__PURE__ */ n.jsxs("div", { children: [
                /* @__PURE__ */ n.jsx(yt, { children: /* @__PURE__ */ n.jsx(jt, { children: "Navigation" }) }),
                /* @__PURE__ */ n.jsx(Et, { className: "flex p-4" }),
                /* @__PURE__ */ n.jsxs("ul", { className: "flex flex-col items-center gap-4 p-4", children: [
                  /* @__PURE__ */ n.jsx("li", { children: /* @__PURE__ */ n.jsx(Nt, {}) }),
                  t.filter(Ot(r)).map((o) => /* @__PURE__ */ n.jsx("li", { children: /* @__PURE__ */ n.jsx("button", { type: "button", onClick: () => a(!1), children: /* @__PURE__ */ n.jsx(Rt, { ...o }) }) }, o.label))
                ] })
              ] }),
              ((i = e.page) == null ? void 0 : i.showPoweredBy) !== !1 && /* @__PURE__ */ n.jsx(Ct, { className: "flex-grow-0 justify-center gap-1" })
            ] })
          }
        )
      ]
    }
  );
}, fe = ({ item: t }) => t.children ? /* @__PURE__ */ n.jsxs(ln, { children: [
  /* @__PURE__ */ n.jsx(un, { children: t.label }),
  /* @__PURE__ */ n.jsx(dn, { children: /* @__PURE__ */ n.jsx(fn, { children: t.children.map((e, r) => (
    // eslint-disable-next-line react/no-array-index-key
    /* @__PURE__ */ n.jsx(fe, { item: e }, r)
  )) }) })
] }, t.label) : /* @__PURE__ */ n.jsx(Me, { to: t.path ?? "", children: /* @__PURE__ */ n.jsxs(hn, { className: "flex gap-2", children: [
  t.icon && /* @__PURE__ */ n.jsx(t.icon, { size: 16, strokeWidth: 1, absoluteStrokeWidth: !0 }),
  t.label
] }, t.label) }), kt = vt(function() {
  const e = ae(), { isAuthenticated: r, profile: s, isAuthEnabled: a } = ae(), i = F(), { page: o, plugins: f } = i, h = f.filter((l) => Wr(l)).flatMap((l) => l.getProfileMenuItems(i)).sort((l) => l.weight ?? 0);
  return /* @__PURE__ */ n.jsxs("header", { className: "sticky lg:top-0 z-10 bg-background/80 backdrop-blur w-full", children: [
    /* @__PURE__ */ n.jsx(_n, {}),
    /* @__PURE__ */ n.jsx("div", { className: "border-b", children: /* @__PURE__ */ n.jsxs("div", { className: "max-w-screen-2xl mx-auto flex relative items-center justify-between px-4 lg:px-8 h-[--top-header-height] border-transparent", children: [
      /* @__PURE__ */ n.jsx("div", { className: "flex", children: /* @__PURE__ */ n.jsx(Me, { to: "/", children: /* @__PURE__ */ n.jsxs("div", { className: "flex items-center gap-3.5", children: [
        (o == null ? void 0 : o.logo) && /* @__PURE__ */ n.jsxs(n.Fragment, { children: [
          /* @__PURE__ */ n.jsx(
            "img",
            {
              src: /https?:\/\//.test(o.logo.src.light) ? o.logo.src.light : ee(
                "/",
                o.logo.src.light
              ),
              alt: o.logo.alt ?? o.pageTitle,
              style: { width: o.logo.width },
              className: "h-10 dark:hidden",
              loading: "lazy"
            }
          ),
          /* @__PURE__ */ n.jsx(
            "img",
            {
              src: /https?:\/\//.test(o.logo.src.dark) ? o.logo.src.dark : ee(
                "/",
                o.logo.src.dark
              ),
              alt: o.logo.alt ?? o.pageTitle,
              style: { width: o.logo.width },
              className: "h-10 hidden dark:block",
              loading: "lazy"
            }
          )
        ] }),
        /* @__PURE__ */ n.jsx("span", { className: "font-bold text-2xl text-foreground/85 tracking-wide", children: o == null ? void 0 : o.pageTitle })
      ] }) }) }),
      /* @__PURE__ */ n.jsx("div", { className: "absolute inset-x-0 justify-center items-center hidden lg:flex w-full pointer-events-none", children: /* @__PURE__ */ n.jsx(Et, { className: "pointer-events-auto" }) }),
      /* @__PURE__ */ n.jsxs("div", { className: "flex items-center gap-8", children: [
        /* @__PURE__ */ n.jsx(Wn, {}),
        /* @__PURE__ */ n.jsxs("div", { className: "hidden lg:flex items-center justify-self-end text-sm gap-2", children: [
          /* @__PURE__ */ n.jsx(_, { name: "head-navigation-start" }),
          a && /* @__PURE__ */ n.jsx(
            ve,
            {
              fallback: /* @__PURE__ */ n.jsx(kn, { className: "rounded h-5 w-24 mr-4" }),
              children: r ? Object.values(h).length > 0 && /* @__PURE__ */ n.jsxs(sn, { modal: !1, children: [
                /* @__PURE__ */ n.jsx(an, { asChild: !0, children: /* @__PURE__ */ n.jsx(Ce, { variant: "ghost", children: s != null && s.name ? `${s.name}` : "My Account" }) }),
                /* @__PURE__ */ n.jsxs(on, { className: "w-56", children: [
                  /* @__PURE__ */ n.jsxs(cn, { children: [
                    s != null && s.name ? `${s.name}` : "My Account",
                    (s == null ? void 0 : s.email) && /* @__PURE__ */ n.jsx("div", { className: "font-normal text-muted-foreground", children: s.email })
                  ] }),
                  h.filter((l) => l.category === "top").length > 0 && /* @__PURE__ */ n.jsx(je, {}),
                  h.filter((l) => l.category === "top").map((l) => /* @__PURE__ */ n.jsx(fe, { item: l }, l.label)),
                  h.filter(
                    (l) => !l.category || l.category === "middle"
                  ).length > 0 && /* @__PURE__ */ n.jsx(je, {}),
                  h.filter(
                    (l) => !l.category || l.category === "middle"
                  ).map((l) => /* @__PURE__ */ n.jsx(fe, { item: l }, l.label)),
                  h.filter((l) => l.category === "bottom").length > 0 && /* @__PURE__ */ n.jsx(je, {}),
                  h.filter((l) => l.category === "bottom").map((l) => /* @__PURE__ */ n.jsx(fe, { item: l }, l.label))
                ] })
              ] }) : /* @__PURE__ */ n.jsx(Ce, { variant: "ghost", onClick: () => e.login(), children: "Login" })
            }
          ),
          /* @__PURE__ */ n.jsx(_, { name: "head-navigation-end" }),
          /* @__PURE__ */ n.jsx(Nt, {})
        ] })
      ] })
    ] }) }),
    /* @__PURE__ */ n.jsx("div", { className: "border-b hidden lg:block", children: /* @__PURE__ */ n.jsxs("div", { className: "max-w-screen-2xl mx-auto border-transparent", children: [
      /* @__PURE__ */ n.jsx(_, { name: "top-navigation-before" }),
      /* @__PURE__ */ n.jsx(Vn, {}),
      /* @__PURE__ */ n.jsx(_, { name: "top-navigation-after" })
    ] }) })
  ] });
}), _t = ({
  shouldScaleBackground: t = !0,
  ...e
}) => /* @__PURE__ */ n.jsx(
  z.Root,
  {
    shouldScaleBackground: t,
    ...e
  }
);
_t.displayName = "Drawer";
const Gn = z.Trigger, Xn = z.Portal, Dt = K.forwardRef(({ className: t, ...e }, r) => /* @__PURE__ */ n.jsx(
  z.Overlay,
  {
    ref: r,
    className: O("fixed inset-0 z-50 bg-black/80", t),
    ...e
  }
));
Dt.displayName = z.Overlay.displayName;
const Jn = K.forwardRef(
  ({ className: t, children: e, hideBar: r = !0, ...s }, a) => /* @__PURE__ */ n.jsxs(Xn, { children: [
    /* @__PURE__ */ n.jsx(Dt, {}),
    /* @__PURE__ */ n.jsxs(
      z.Content,
      {
        ref: a,
        className: O(
          "fixed inset-x-0 bottom-0 z-50 mt-24 flex h-auto flex-col rounded-t-[10px] border bg-background",
          t
        ),
        ...s,
        children: [
          !r && /* @__PURE__ */ n.jsx("div", { className: "mx-auto mt-4 h-2 w-[100px] rounded-full bg-muted" }),
          e
        ]
      }
    )
  ] })
);
Jn.displayName = "DrawerContent";
const es = K.forwardRef(({ className: t, ...e }, r) => /* @__PURE__ */ n.jsx(
  z.Title,
  {
    ref: r,
    className: O(
      "text-lg font-semibold leading-none tracking-tight",
      t
    ),
    ...e
  }
));
es.displayName = z.Title.displayName;
const ts = K.forwardRef(({ className: t, ...e }, r) => /* @__PURE__ */ n.jsx(
  z.Description,
  {
    ref: r,
    className: O("text-sm text-muted-foreground", t),
    ...e
  }
));
ts.displayName = z.Description.displayName;
const qt = ({
  children: t,
  className: e
}) => {
  var a;
  const { options: r } = F(), s = G(null);
  return Q(() => {
    var o;
    const i = (o = s.current) == null ? void 0 : o.querySelector('[aria-current="page"]');
    kr(i ?? null);
  }, []), /* @__PURE__ */ n.jsxs("div", { className: "grid sticky top-[--header-height] lg:h-[calc(100vh-var(--header-height))] grid-rows-[1fr_min-content] border-r", children: [
    /* @__PURE__ */ n.jsx(
      "nav",
      {
        ref: s,
        className: O(
          "hidden max-w-[calc(var(--side-nav-width)+var(--padding-nav-item))] lg:flex scrollbar flex-col overflow-y-auto shrink-0 text-sm pe-3 ps-4 lg:ps-8",
          "-mx-[--padding-nav-item] pb-[8vh] pt-[--padding-content-top] scroll-pt-2 gap-1",
          // Revert the padding/margin on the first child
          "-mt-2.5",
          e
        ),
        style: {
          maskImage: "linear-gradient(180deg, transparent 1%, rgba(0, 0, 0, 1) 20px, rgba(0, 0, 0, 1) 90%, transparent 99%)"
        },
        children: t
      }
    ),
    /* @__PURE__ */ n.jsx("div", { className: "bg-background border-t p-2 mx-5  gap-2 items-center mt-2 drop-shadow-[0_-3px_1px_rgba(0,0,0,0.015)] hidden lg:[&:has(>_:nth-child(1):last-child)]:flex", children: ((a = r.page) == null ? void 0 : a.showPoweredBy) !== !1 && /* @__PURE__ */ n.jsx(Ct, {}) })
  ] });
};
qt.displayName = "SidebarWrapper";
const rs = ({
  onRequestClose: t,
  sidebar: e
}) => /* @__PURE__ */ n.jsxs(n.Fragment, { children: [
  /* @__PURE__ */ n.jsxs(qt, { children: [
    /* @__PURE__ */ n.jsx(_, { name: "zudoku-before-navigation" }),
    e.map((r) => /* @__PURE__ */ n.jsx(
      He,
      {
        item: r
      },
      ("id" in r ? r.id : "") + ("href" in r ? r.href : "") + r.label
    )),
    /* @__PURE__ */ n.jsx(_, { name: "zudoku-after-navigation" })
  ] }),
  /* @__PURE__ */ n.jsx(
    bt,
    {
      className: "lg:hidden h-[100dvh] left-0 w-[320px] rounded-none",
      "aria-describedby": void 0,
      children: /* @__PURE__ */ n.jsxs("div", { className: "p-4 overflow-y-auto overscroll-none", children: [
        /* @__PURE__ */ n.jsx(yt, { children: /* @__PURE__ */ n.jsx(jt, { children: "Sidebar" }) }),
        e.map((r) => /* @__PURE__ */ n.jsx(
          He,
          {
            item: r,
            onRequestClose: t
          },
          r.label
        ))
      ] })
    }
  )
] }), ns = ({ children: t }) => {
  const [e, r] = J(!1), { sidebar: s } = mt(), a = s.length > 0, i = pe().state === "loading";
  return /* @__PURE__ */ n.jsxs(
    _t,
    {
      direction: "left",
      open: e,
      onOpenChange: (o) => r(o),
      children: [
        a && /* @__PURE__ */ n.jsx(
          rs,
          {
            onRequestClose: () => r(!1),
            sidebar: s
          }
        ),
        a && /* @__PURE__ */ n.jsx("div", { className: "lg:hidden -mx-4 px-4 py-2 sticky bg-background/80 backdrop-blur z-10 top-0 left-0 right-0 border-b", children: /* @__PURE__ */ n.jsxs(Gn, { className: "flex items-center gap-2 px-4", children: [
          /* @__PURE__ */ n.jsx(Yr, { size: 16, strokeWidth: 1.5 }),
          /* @__PURE__ */ n.jsx("span", { className: "text-sm", children: "Menu" })
        ] }) }),
        /* @__PURE__ */ n.jsxs(
          "main",
          {
            "data-pagefind-body": !0,
            className: O(
              "px-4 lg:pe-8 lg:px-8",
              !a && "col-span-full",
              i && "animate-pulse"
            ),
            children: [
              /* @__PURE__ */ n.jsx(_, { name: "zudoku-before-content" }),
              t,
              /* @__PURE__ */ n.jsx(_, { name: "zudoku-after-content" })
            ]
          }
        )
      ]
    }
  );
}, ss = () => /* @__PURE__ */ n.jsx("main", { className: "col-span-full row-span-full grid place-items-center", children: /* @__PURE__ */ n.jsx(xt, {}) }), as = ({ children: t }) => {
  const { meta: e, authentication: r, options: s } = F(), a = ge();
  return _r(), Mn(), Q(() => {
    var i;
    (i = r == null ? void 0 : r.onPageLoad) == null || i.call(r);
  }, [r]), /* @__PURE__ */ n.jsxs(n.Fragment, { children: [
    !1,
    /* @__PURE__ */ n.jsxs(Te, { titleTemplate: e == null ? void 0 : e.title, children: [
      s.canonicalUrlOrigin && /* @__PURE__ */ n.jsx(
        "link",
        {
          rel: "canonical",
          href: ee(
            s.canonicalUrlOrigin,
            s.basePath,
            a.pathname
          )
        }
      ),
      (e == null ? void 0 : e.description) && /* @__PURE__ */ n.jsx("meta", { name: "description", content: e.description }),
      (e == null ? void 0 : e.favicon) && /* @__PURE__ */ n.jsx("link", { rel: "icon", href: e.favicon })
    ] }),
    /* @__PURE__ */ n.jsx(_, { name: "layout-before-head" }),
    /* @__PURE__ */ n.jsx(kt, {}),
    /* @__PURE__ */ n.jsx(_, { name: "layout-after-head" }),
    /* @__PURE__ */ n.jsx(
      "div",
      {
        className: O(
          "grid max-w-screen-2xl w-full lg:mx-auto",
          "has-[:only-child]:grid-rows-1 grid-rows-[0_min-content_1fr] lg:grid-rows-[min-content_1fr]",
          "grid-cols-1 lg:grid-cols-[var(--side-nav-width)_1fr]"
        ),
        children: /* @__PURE__ */ n.jsx(Ae, { fallback: /* @__PURE__ */ n.jsx(ss, {}), children: /* @__PURE__ */ n.jsx(ns, { children: t ?? /* @__PURE__ */ n.jsx(lt, {}) }) })
      }
    ),
    /* @__PURE__ */ n.jsx(Rn, {})
  ] });
}, os = ({
  title: t = "An error occurred",
  message: e,
  category: r
}) => /* @__PURE__ */ n.jsxs("div", { className: qr + " h-full pt-[--padding-content-top]", children: [
  r && /* @__PURE__ */ n.jsx(Zr, { children: r }),
  t && /* @__PURE__ */ n.jsx(Dr, { level: 1, className: "flex gap-3.5 items-center", children: t }),
  /* @__PURE__ */ n.jsx("p", { children: e })
] }), is = (t) => {
  switch (t) {
    case 400:
      return {
        title: "Bad Request",
        message: "The request could not be understood by the server due to malformed syntax."
      };
    case 403:
      return {
        title: "Forbidden",
        message: "You don't have permission to access this resource."
      };
    case 404:
      return {
        title: "Not Found",
        message: "The requested resource could not be found."
      };
    case 405:
      return {
        title: "Method Not Allowed",
        message: "The request method is not supported for the requested resource."
      };
    case 414:
      return {
        title: "Request URI Too Large",
        message: "The request URI is too large."
      };
    case 416:
      return {
        title: "Range Not Satisfiable",
        message: "The server cannot satisfy the request range."
      };
    case 500:
      return {
        title: "Internal Server Error",
        message: "An unexpected error occurred while processing your request."
      };
    case 501:
      return {
        title: "Not Implemented",
        message: "The server does not support the functionality required to fulfill the request."
      };
    case 502:
      return {
        title: "Bad Gateway",
        message: "The server received an invalid response from the upstream server."
      };
    case 503:
      return {
        title: "Service Unavailable",
        message: "The server is temporarily unable to handle the request."
      };
    case 504:
      return {
        title: "Gateway Timeout",
        message: "The server did not receive a timely response from the upstream server."
      };
    default:
      return {
        title: "An error occurred",
        message: "Something went wrong while processing your request."
      };
  }
}, cs = ({ statusCode: t, message: e }) => {
  const r = is(t);
  return /* @__PURE__ */ n.jsx(
    os,
    {
      title: r.title,
      message: e ?? r.message,
      category: t
    }
  );
}, ls = Oe(null), Pe = {
  didCatch: !1,
  error: null
};
class us extends Mr {
  constructor(e) {
    super(e), this.resetErrorBoundary = this.resetErrorBoundary.bind(this), this.state = Pe;
  }
  static getDerivedStateFromError(e) {
    return {
      didCatch: !0,
      error: e
    };
  }
  resetErrorBoundary() {
    const {
      error: e
    } = this.state;
    if (e !== null) {
      for (var r, s, a = arguments.length, i = new Array(a), o = 0; o < a; o++)
        i[o] = arguments[o];
      (r = (s = this.props).onReset) === null || r === void 0 || r.call(s, {
        args: i,
        reason: "imperative-api"
      }), this.setState(Pe);
    }
  }
  componentDidCatch(e, r) {
    var s, a;
    (s = (a = this.props).onError) === null || s === void 0 || s.call(a, e, r);
  }
  componentDidUpdate(e, r) {
    const {
      didCatch: s
    } = this.state, {
      resetKeys: a
    } = this.props;
    if (s && r.error !== null && ds(e.resetKeys, a)) {
      var i, o;
      (i = (o = this.props).onReset) === null || i === void 0 || i.call(o, {
        next: a,
        prev: e.resetKeys,
        reason: "keys"
      }), this.setState(Pe);
    }
  }
  render() {
    const {
      children: e,
      fallbackRender: r,
      FallbackComponent: s,
      fallback: a
    } = this.props, {
      didCatch: i,
      error: o
    } = this.state;
    let f = e;
    if (i) {
      const h = {
        error: o,
        resetErrorBoundary: this.resetErrorBoundary
      };
      if (typeof r == "function")
        f = r(h);
      else if (s)
        f = Be(s, h);
      else if (a !== void 0)
        f = a;
      else
        throw o;
    }
    return Be(ls.Provider, {
      value: {
        didCatch: i,
        error: o,
        resetErrorBoundary: this.resetErrorBoundary
      }
    }, f);
  }
}
function ds() {
  let t = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : [], e = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : [];
  return t.length !== e.length || t.some((r, s) => !Object.is(r, e[s]));
}
let fs = () => ({
  emit(t, ...e) {
    for (let r = this.events[t] || [], s = 0, a = r.length; s < a; s++)
      r[s](...e);
  },
  events: {},
  on(t, e) {
    var r;
    return ((r = this.events)[t] || (r[t] = [])).push(e), () => {
      var s;
      this.events[t] = (s = this.events[t]) == null ? void 0 : s.filter((a) => e !== a);
    };
  }
});
class hs {
  constructor(e, r) {
    A(this, "plugins");
    A(this, "sidebars");
    A(this, "topNavigation");
    A(this, "meta");
    A(this, "page");
    A(this, "authentication");
    A(this, "navigationPlugins");
    A(this, "emitter", fs());
    A(this, "initialize", async () => {
      await Promise.all(
        this.plugins.filter(Jr).map((e) => {
          var r;
          return (r = e.initialize) == null ? void 0 : r.call(e, this);
        })
      );
    });
    A(this, "getApiIdentities", async () => (await Promise.all(
      this.plugins.filter(en).map((r) => r.getIdentities(this))
    )).flat());
    A(this, "emitEvent", (e, ...r) => this.emitter.emit(e, ...r));
    A(this, "getPluginSidebar", async (e) => (await Promise.all(
      this.navigationPlugins.map(
        (s) => {
          var a;
          return (a = s.getSidebar) == null ? void 0 : a.call(s, yr(e), this);
        }
      )
    )).flatMap((s) => s ?? []));
    A(this, "signRequest", async (e) => {
      if (!this.authentication)
        throw new Error("No authentication provider configured");
      return await this.authentication.signRequest(e);
    });
    this.options = e, this.queryClient = r, this.plugins = e.plugins ?? [], this.topNavigation = e.topNavigation ?? [], this.sidebars = e.sidebars ?? {}, this.navigationPlugins = this.plugins.filter(Gr), this.authentication = e.authentication, this.meta = e.metadata, this.page = e.page, this.plugins.forEach((s) => {
      Xr(s) && nn(s.events).forEach(([a, i]) => {
        this.emitter.on(a, i);
      });
    }), xr.subscribe((s, a) => {
      this.emitEvent("auth", {
        prev: a,
        next: s
      });
    });
  }
  addEventListener(e, r) {
    return this.emitter.on(e, r);
  }
}
function ms({ error: t, resetErrorBoundary: e }) {
  return /* @__PURE__ */ n.jsx(gt, { error: t });
}
const de = globalThis;
(!de.requestIdleCallback || !de.cancelIdleCallback) && (de.requestIdleCallback = (t) => setTimeout(t, 1), de.cancelIdleCallback = clearTimeout);
const $t = {
  Header: kt
}, gs = Oe($t), ps = gs.Provider, vs = () => {
  const t = ge(), e = F(), r = G(void 0);
  return Q(() => {
    e.emitEvent("location", {
      from: r.current,
      to: t
    }), r.current = t;
  }, [e, t]), null;
}, xs = ({
  children: t,
  context: e
}) => (br({
  queryFn: async () => (await e.initialize(), !0),
  queryKey: ["zudoku-initialize"]
}), /* @__PURE__ */ n.jsx(jr.Provider, { value: e, children: t }));
let Se;
const It = vt(
  ({ children: t, ...e }) => {
    var p, y;
    const r = be(
      () => ({ ...$t, ...e.overrides }),
      [e.overrides]
    ), s = ge(), a = be(() => {
      var g;
      return {
        ...(e.plugins ?? []).filter(tn).flatMap(
          (C) => C.getMdxComponents ? [C.getMdxComponents()] : []
        ).reduce(
          (C, N) => ({ ...C, ...N }),
          {}
        ),
        ...$r,
        ...(g = e.mdx) == null ? void 0 : g.components
      };
    }, [(p = e.mdx) == null ? void 0 : p.components, e.plugins]), { stagger: i } = Or(Ne), [o, f] = J(!1), h = be(
      () => o ? { stagger: !0 } : { stagger: i },
      [i, o]
    ), l = pe(), b = ft();
    Q(() => {
      o || f(!0);
    }, [o, l.location]), Se || (Se = new hs(e, b));
    const x = (y = e.plugins) == null ? void 0 : y.flatMap(
      (m) => {
        var g;
        return rn(m) ? ((g = m.getHead) == null ? void 0 : g.call(m, { location: s })) ?? [] : [];
      }
    ).map((m, g) => /* @__PURE__ */ n.jsx(Te, { children: m }, g));
    return /* @__PURE__ */ n.jsxs(n.Fragment, { children: [
      x,
      /* @__PURE__ */ n.jsx(Ne.Provider, { value: h, children: /* @__PURE__ */ n.jsxs(xs, { context: Se, children: [
        /* @__PURE__ */ n.jsx(vs, {}),
        /* @__PURE__ */ n.jsx(vn, { components: a, children: /* @__PURE__ */ n.jsx(or, { attribute: "class", disableTransitionOnChange: !0, children: /* @__PURE__ */ n.jsx(ps, { value: r, children: /* @__PURE__ */ n.jsx(Sr, { slotlets: e.slotlets, children: /* @__PURE__ */ n.jsx(Ir, { children: t ?? /* @__PURE__ */ n.jsx(lt, {}) }) }) }) }) })
      ] }) })
    ] });
  }
);
It.displayName = "ZudokoInner";
const Ft = (t) => /* @__PURE__ */ n.jsx(us, { FallbackComponent: ms, children: /* @__PURE__ */ n.jsx(It, { ...t }) });
Ft.displayName = "Zudoku";
const Hs = Pt, Ks = as, Us = Cr, Ys = Sn, Zs = Nn, Vs = Tn, Ws = ar, Gs = Te, ys = F, Xs = ae, Js = wr, ea = Pr, ta = Ft, ra = cs, na = Nr, sa = Fr, aa = xt, oa = ve, ia = Er, ca = Me, la = ct;
export {
  ia as B,
  ea as C,
  Gs as H,
  ca as L,
  sa as M,
  Us as R,
  Ys as S,
  ta as Z,
  Xs as a,
  ys as b,
  Hs as c,
  Ks as d,
  Zs as e,
  Vs as f,
  Ws as g,
  Js as h,
  ra as i,
  na as j,
  aa as k,
  oa as l,
  la as m,
  Pt as u
};
