// zudoku/lib/OperationList-0bhBU7ME.js
import { j as $ } from "./jsx-runtime-C5mzlN2N.js";
import { d as yr, u as Pa, m as qa, t as Ca } from "./hook-pPrHCB6G.js";
import { H as Ea } from "./RouteGuard-BZ_VsiXc.js";
import { ChevronsUpDownIcon as Yn, CheckIcon as Ra, CopyIcon as ka, UnfoldVerticalIcon as Na, FoldVerticalIcon as Aa, ChevronsDownUpIcon as Ia } from "lucide-react";
import { b as Ua, a as La } from "./chunk-KNED5TY2-BUPjb3LQ.js";
import { c as _e, u as mr, d as Xn, C as Zn, b as ea, a as ur, P as Fa, m as ta, e as _a, U as Ba } from "./index-Ba_X7EnR.js";
import { S as Ha, a as Ma, b as za, c as Wa, e as Ja } from "./Select-Dg5R11Dx.js";
import { C as Ga } from "./CategoryHeading-D06WK_Wo.js";
import { I as Va, S as ra, M as De, H as Ie, P as fr } from "./Markdown-C5h6bxbE.js";
import { P as Da } from "./Pagination-Bt1czAiJ.js";
import { c as pe } from "./cn-qaFjX9_3.js";
import { useTransition as na, useState as xe, useRef as aa, useEffect as gr, useLayoutEffect as Qa, useMemo as oa } from "react";
import { Button as Ka } from "./ui/Button.js";
import { u as Qe, P as Ya, E as Ar, a as Xa, g as Za, S as Ir } from "./SchemaView-DCSWtnYr.js";
import { u as vr } from "./context-Lrf2Y9bR.js";
import { VisuallyHidden as pr } from "@radix-ui/react-visually-hidden";
import { c as eo } from "./index-CPNSgwSb.js";
import { Tabs as to, TabsList as ro, TabsTrigger as no, TabsContent as ao } from "./ui/Tabs.js";
import { Card as oo } from "./ui/Card.js";
import { g as io, c as Ke, a as so } from "./_commonjsHelpers-B4e78b8K.js";
import { a as lo, b as co } from "./index-tpbiZmWp.js";
import { S as ia } from "./SyntaxHighlight-CxhyyMkF.js";
import { B as Ur } from "./Button-BBNrKpQd.js";
import * as Ge from "@radix-ui/react-tabs";
function uo(e, n) {
  return n;
}
const Ye = ({
  value: e,
  onChange: n,
  className: t,
  options: a,
  showChevrons: c = !0
}) => /* @__PURE__ */ $.jsxs("div", { className: "grid", children: [
  /* @__PURE__ */ $.jsx(
    "select",
    {
      className: pe(
        "w-full row-start-1 col-start-1 border border-input text-foreground px-2 py-1 pe-6",
        "rounded-md appearance-none bg-zinc-50 hover:bg-white dark:bg-zinc-800 hover:dark:bg-zinc-800/75",
        t
      ),
      value: e,
      onChange: n,
      children: a.map((s) => /* @__PURE__ */ $.jsx("option", { value: s.value, children: s.label }, s.value))
    }
  ),
  /* @__PURE__ */ $.jsx(
    "div",
    {
      className: pe(
        !c && "hidden",
        "row-start-1 col-start-1 self-center justify-self-end relative end-2 pointer-events-none"
      ),
      children: /* @__PURE__ */ $.jsx(Yn, { size: 14 })
    }
  )
] }), fo = _e(
  /* GraphQL */
  `
  query ServersQuery($input: JSON!, $type: SchemaType!) {
    schema(input: $input, type: $type) {
      url
      servers {
        url
      }
    }
  }
`
), Lr = ({ url: e }) => {
  const [n, t] = xe(!1);
  return /* @__PURE__ */ $.jsx(
    Ka,
    {
      onClick: () => {
        navigator.clipboard.writeText(e).then(() => {
          t(!0), setTimeout(() => t(!1), 2e3);
        });
      },
      variant: "ghost",
      size: "icon",
      children: n ? /* @__PURE__ */ $.jsx(Ra, { className: "text-green-600", size: 14 }) : /* @__PURE__ */ $.jsx(ka, { size: 14, strokeWidth: 1.3 })
    }
  );
}, po = () => {
  const { input: e, type: n } = vr(), t = Qe(fo, { input: e, type: n }), a = yr(t), [, c] = na(), { selectedServer: s, setSelectedServer: r } = mr(
    a.data.schema.servers
  ), { servers: o } = a.data.schema;
  if (o.length === 0) return null;
  const i = o.at(0);
  return o.length === 1 ? /* @__PURE__ */ $.jsxs("div", { className: "flex items-center gap-2", children: [
    /* @__PURE__ */ $.jsx("span", { className: "font-medium text-sm", children: "Endpoint:" }),
    /* @__PURE__ */ $.jsx(Va, { className: "text-xs px-2 py-1.5", selectOnClick: !0, children: i.url }),
    /* @__PURE__ */ $.jsx(Lr, { url: i.url })
  ] }) : /* @__PURE__ */ $.jsxs("div", { className: "flex flex-wrap items-center gap-2", children: [
    /* @__PURE__ */ $.jsx("span", { className: "font-medium text-sm", children: "Endpoint" }),
    /* @__PURE__ */ $.jsx(
      Ye,
      {
        className: "font-mono text-xs bg-border/50 dark:bg-border/70 py-1.5 max-w-[450px] truncate",
        onChange: (l) => c(() => r(l.target.value)),
        value: s,
        showChevrons: o.length > 1,
        options: o.map((l) => ({
          value: l.url,
          label: l.url
        }))
      }
    ),
    /* @__PURE__ */ $.jsx(Lr, { url: s })
  ] });
}, ho = eo(
  "inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-medium transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
  {
    variants: {
      variant: {
        default: "border-transparent bg-primary text-primary-foreground hover:bg-primary/80",
        secondary: "border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80",
        muted: "bg-muted text-muted-foreground ",
        destructive: "border-transparent bg-destructive text-destructive-foreground hover:bg-destructive/80",
        outline: "text-foreground"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
function Fr({ className: e, variant: n, ...t }) {
  return /* @__PURE__ */ $.jsx("div", { className: pe(ho({ variant: n }), e), ...t });
}
const yo = (e, n) => e ? n(e) : void 0, mo = (e) => e.schema != null && typeof e.schema == "object" ? e.schema : {
  type: "string"
}, go = ({
  parameter: e,
  group: n,
  id: t
}) => {
  const a = mo(e);
  return /* @__PURE__ */ $.jsxs("li", { className: "p-4 bg-border/20 text-sm flex flex-col gap-1.5", children: [
    /* @__PURE__ */ $.jsxs("div", { className: "flex items-center gap-2", children: [
      /* @__PURE__ */ $.jsx(ra, { asChild: !0, children: /* @__PURE__ */ $.jsx("code", { children: n === "path" ? /* @__PURE__ */ $.jsx(
        Xn,
        {
          name: e.name,
          backgroundOpacity: "15%",
          className: "px-2",
          slug: `${t}-${e.name}`
        }
      ) : e.name }) }),
      /* @__PURE__ */ $.jsx(
        Ya,
        {
          schema: a,
          extraItems: [
            e.required && /* @__PURE__ */ $.jsx("span", { className: "text-primary", children: "required" }),
            e.style && `style: ${e.style}`,
            e.explode && `explode: ${e.explode}`
          ]
        }
      )
    ] }),
    e.description && /* @__PURE__ */ $.jsx(
      De,
      {
        content: e.description,
        className: "text-sm prose-p:my-1 prose-code:whitespace-pre-line"
      }
    ),
    a.type === "array" && a.items.enum ? /* @__PURE__ */ $.jsx(Ar, { values: a.items.enum }) : a.enum && /* @__PURE__ */ $.jsx(Ar, { values: a.enum }),
    /* @__PURE__ */ $.jsx(Xa, { schema: a })
  ] });
}, vo = ({
  summary: e,
  group: n,
  parameters: t,
  id: a
}) => /* @__PURE__ */ $.jsxs($.Fragment, { children: [
  /* @__PURE__ */ $.jsxs(Ie, { level: 3, id: `${a}/${n}-parameters`, className: "capitalize", children: [
    e && /* @__PURE__ */ $.jsxs(pr, { children: [
      e,
      " › "
    ] }),
    n === "header" ? "Headers" : `${n} Parameters`
  ] }),
  /* @__PURE__ */ $.jsx(oo, { children: /* @__PURE__ */ $.jsx("ul", { className: "list-none m-0 px-0 divide-y ", children: t.sort((c, s) => c.required === s.required ? 0 : c.required ? -1 : 1).map((c) => /* @__PURE__ */ $.jsx(
    go,
    {
      parameter: c,
      id: a,
      group: n
    },
    `${c.name}-${c.in}`
  )) }) })
] });
var _r = (e, n) => {
  const t = e[n.name];
  return t === void 0 ? (e[n.name] = n.value, e) : Array.isArray(t) ? (t.push(n.value), e) : (e[n.name] = [t, n.value], e);
}, Br = "", Hr = `
`, V = class {
  /**
   * Helper object to format and aggragate lines of code.
   * Lines are aggregated in a `code` array, and need to be joined to obtain a proper code snippet.
   */
  constructor({ indent: e, join: n } = {}) {
    this.postProcessors = [], this.code = [], this.indentationCharacter = Br, this.lineJoin = Hr, this.indentLine = (t, a = 0) => `${this.indentationCharacter.repeat(a)}${t}`, this.unshift = (t, a) => {
      const c = this.indentLine(t, a);
      this.code.unshift(c);
    }, this.push = (t, a) => {
      const c = this.indentLine(t, a);
      this.code.push(c);
    }, this.blank = () => {
      this.code.push("");
    }, this.join = () => {
      const t = this.code.join(this.lineJoin);
      return this.postProcessors.reduce((c, s) => s(c), t);
    }, this.addPostProcessor = (t) => {
      this.postProcessors = [...this.postProcessors, t];
    }, this.indentationCharacter = e || Br, this.lineJoin = n ?? Hr;
  }
}, ft, Mr;
function bo() {
  return Mr || (Mr = 1, ft = function(e) {
    return Object.prototype.toString.call(e) === "[object RegExp]";
  }), ft;
}
var pt, zr;
function xo() {
  return zr || (zr = 1, pt = function(e) {
    var n = typeof e;
    return e !== null && (n === "object" || n === "function");
  }), pt;
}
var Je = {}, Wr;
function $o() {
  return Wr || (Wr = 1, Object.defineProperty(Je, "__esModule", { value: !0 }), Je.default = (e) => Object.getOwnPropertySymbols(e).filter((n) => Object.prototype.propertyIsEnumerable.call(e, n))), Je;
}
var dt, Jr;
function wo() {
  if (Jr) return dt;
  Jr = 1;
  const e = bo(), n = xo(), t = $o().default;
  return dt = (a, c, s) => {
    const r = [];
    return function o(i, l, f) {
      l = l || {}, l.indent = l.indent || "	", f = f || "";
      let y;
      l.inlineCharacterLimit === void 0 ? y = {
        newLine: `
`,
        newLineOrSpace: `
`,
        pad: f,
        indent: f + l.indent
      } : y = {
        newLine: "@@__STRINGIFY_OBJECT_NEW_LINE__@@",
        newLineOrSpace: "@@__STRINGIFY_OBJECT_NEW_LINE_OR_SPACE__@@",
        pad: "@@__STRINGIFY_OBJECT_PAD__@@",
        indent: "@@__STRINGIFY_OBJECT_INDENT__@@"
      };
      const g = (m) => {
        if (l.inlineCharacterLimit === void 0)
          return m;
        const x = m.replace(new RegExp(y.newLine, "g"), "").replace(new RegExp(y.newLineOrSpace, "g"), " ").replace(new RegExp(y.pad + "|" + y.indent, "g"), "");
        return x.length <= l.inlineCharacterLimit ? x : m.replace(new RegExp(y.newLine + "|" + y.newLineOrSpace, "g"), `
`).replace(new RegExp(y.pad, "g"), f).replace(new RegExp(y.indent, "g"), f + l.indent);
      };
      if (r.indexOf(i) !== -1)
        return '"[Circular]"';
      if (i == null || typeof i == "number" || typeof i == "boolean" || typeof i == "function" || typeof i == "symbol" || e(i))
        return String(i);
      if (i instanceof Date)
        return `new Date('${i.toISOString()}')`;
      if (Array.isArray(i)) {
        if (i.length === 0)
          return "[]";
        r.push(i);
        const m = "[" + y.newLine + i.map((x, v) => {
          const u = i.length - 1 === v ? y.newLine : "," + y.newLineOrSpace;
          let h = o(x, l, f + l.indent);
          return l.transform && (h = l.transform(i, v, h)), y.indent + h + u;
        }).join("") + y.pad + "]";
        return r.pop(), g(m);
      }
      if (n(i)) {
        let m = Object.keys(i).concat(t(i));
        if (l.filter && (m = m.filter((v) => l.filter(i, v))), m.length === 0)
          return "{}";
        r.push(i);
        const x = "{" + y.newLine + m.map((v, u) => {
          const h = m.length - 1 === u ? y.newLine : "," + y.newLineOrSpace, d = typeof v == "symbol", j = !d && /^[a-z$_][a-z$_0-9]*$/i.test(v), T = d || j ? v : o(v, l);
          let p = o(i[v], l, f + l.indent);
          return l.transform && (p = l.transform(i, v, p)), y.indent + String(T) + ": " + p + h;
        }).join("") + y.pad + "}";
        return r.pop(), g(x);
      }
      return i = String(i).replace(/[\r\n]/g, (m) => m === `
` ? "\\n" : "\\r"), l.singleQuotes === !1 ? (i = i.replace(/"/g, '\\"'), `"${i}"`) : (i = i.replace(/\\?'/g, "\\'"), `'${i}'`);
    }(a, c, s);
  }, dt;
}
var So = wo();
const ue = /* @__PURE__ */ io(So);
function Le(e, n = {}) {
  const { delimiter: t = '"', escapeChar: a = "\\", escapeNewlines: c = !0 } = n;
  return [...e.toString()].map((r) => r === "\b" ? `${a}b` : r === "	" ? `${a}t` : r === `
` ? c ? `${a}n` : r : r === "\f" ? `${a}f` : r === "\r" ? c ? `${a}r` : r : r === a ? a + a : r === t ? a + t : r < " " || r > "~" ? JSON.stringify(r).slice(1, -1) : r).join("");
}
var Fe = (e) => Le(e, { delimiter: "'" }), le = (e) => Le(e, { delimiter: '"' }), jo = {
  info: {
    key: "libcurl",
    title: "Libcurl",
    link: "http://curl.haxx.se/libcurl",
    description: "Simple REST and HTTP API Client for C",
    extname: ".c"
  },
  convert: ({ method: e, fullUrl: n, headersObj: t, allHeaders: a, postData: c }) => {
    const { push: s, blank: r, join: o } = new V();
    s("CURL *hnd = curl_easy_init();"), r(), s(`curl_easy_setopt(hnd, CURLOPT_CUSTOMREQUEST, "${e.toUpperCase()}");`), s("curl_easy_setopt(hnd, CURLOPT_WRITEDATA, stdout);"), s(`curl_easy_setopt(hnd, CURLOPT_URL, "${n}");`);
    const i = Object.keys(t);
    return i.length && (r(), s("struct curl_slist *headers = NULL;"), i.forEach((l) => {
      s(`headers = curl_slist_append(headers, "${l}: ${le(t[l])}");`);
    }), s("curl_easy_setopt(hnd, CURLOPT_HTTPHEADER, headers);")), a.cookie && (r(), s(`curl_easy_setopt(hnd, CURLOPT_COOKIE, "${a.cookie}");`)), c.text && (r(), s(`curl_easy_setopt(hnd, CURLOPT_POSTFIELDS, ${JSON.stringify(c.text)});`)), r(), s("CURLcode ret = curl_easy_perform(hnd);"), o();
  }
}, Oo = {
  info: {
    key: "c",
    title: "C",
    default: "libcurl",
    cli: "c"
  },
  clientsById: {
    libcurl: jo
  }
}, fe = (e, n) => Object.keys(e).find((t) => t.toLowerCase() === n.toLowerCase()), ge = (e, n) => {
  const t = fe(e, n);
  if (t)
    return e[t];
}, Ze = (e, n) => !!fe(e, n), To = (e) => ["application/json", "application/x-json", "text/json", "text/x-json", "+json"].some(
  (n) => e.indexOf(n) > -1
), Gr = class {
  constructor(e) {
    this.name = "", this.toString = () => `:${this.name}`, this.name = e;
  }
}, Po = class {
  constructor(e) {
    this.path = "", this.toString = () => `(clojure.java.io/file "${this.path}")`, this.path = e;
  }
}, sa = (e) => e === void 0 ? null : e === null ? "null" : e.constructor.name.toLowerCase(), la = (e) => e === void 0 ? !0 : sa(e) === "object" ? Object.keys(e).length === 0 : !1, Vr = (e) => (Object.keys(e).filter((n) => la(e[n])).forEach((n) => {
  delete e[n];
}), e), Ve = (e, n) => {
  const t = " ".repeat(e);
  return n.replace(/\n/g, `
${t}`);
}, dr = (e) => {
  switch (sa(e)) {
    case "string":
      return `"${e.replace(/"/g, '\\"')}"`;
    case "file":
      return e.toString();
    case "keyword":
      return e.toString();
    case "null":
      return "nil";
    case "regexp":
      return `#"${e.source}"`;
    case "object": {
      const n = Object.keys(e).reduce((t, a) => {
        const c = Ve(a.length + 2, dr(e[a]));
        return `${t}:${a} ${c}
 `;
      }, "").trim();
      return `{${Ve(1, n)}}`;
    }
    case "array": {
      const n = e.reduce((t, a) => `${t} ${dr(a)}`, "").trim();
      return `[${Ve(1, n)}]`;
    }
    default:
      return e.toString();
  }
}, qo = {
  info: {
    key: "clj_http",
    title: "clj-http",
    link: "https://github.com/dakrone/clj-http",
    description: "An idiomatic clojure http client wrapping the apache client.",
    extname: ".clj"
  },
  convert: ({ queryObj: e, method: n, postData: t, url: a, allHeaders: c }, s) => {
    const { push: r, join: o } = new V({ indent: s == null ? void 0 : s.indent }), i = ["get", "post", "put", "delete", "patch", "head", "options"];
    if (n = n.toLowerCase(), !i.includes(n))
      return r("Method not supported"), o();
    const l = {
      headers: c,
      "query-params": e
    };
    switch (t.mimeType) {
      case "application/json":
        {
          l["content-type"] = new Gr("json"), l["form-params"] = t.jsonObj;
          const f = fe(l.headers, "content-type");
          f && delete l.headers[f];
        }
        break;
      case "application/x-www-form-urlencoded":
        {
          l["form-params"] = t.paramsObj;
          const f = fe(l.headers, "content-type");
          f && delete l.headers[f];
        }
        break;
      case "text/plain":
        {
          l.body = t.text;
          const f = fe(l.headers, "content-type");
          f && delete l.headers[f];
        }
        break;
      case "multipart/form-data": {
        if (t.params) {
          l.multipart = t.params.map((y) => y.fileName && !y.value ? {
            name: y.name,
            content: new Po(y.fileName)
          } : {
            name: y.name,
            content: y.value
          });
          const f = fe(l.headers, "content-type");
          f && delete l.headers[f];
        }
        break;
      }
    }
    switch (ge(l.headers, "accept")) {
      case "application/json":
        {
          l.accept = new Gr("json");
          const f = fe(l.headers, "accept");
          f && delete l.headers[f];
        }
        break;
    }
    if (r(`(require '[clj-http.client :as client])
`), la(Vr(l)))
      r(`(client/${n} "${a}")`);
    else {
      const f = 11 + n.length + a.length, y = Ve(f, dr(Vr(l)));
      r(`(client/${n} "${a}" ${y})`);
    }
    return o();
  }
}, Co = {
  info: {
    key: "clojure",
    title: "Clojure",
    default: "clj_http"
  },
  clientsById: {
    clj_http: qo
  }
}, Eo = (e) => {
  let n = ge(e, "accept-encoding");
  if (!n)
    return [];
  const t = {
    gzip: "DecompressionMethods.GZip",
    deflate: "DecompressionMethods.Deflate"
  }, a = [];
  return typeof n == "string" && (n = [n]), n.forEach((c) => {
    c.split(",").forEach((s) => {
      const r = /\s*([^;\s]+)/.exec(s);
      if (r) {
        const o = t[r[1]];
        o && a.push(o);
      }
    });
  }), a;
}, Ro = {
  info: {
    key: "httpclient",
    title: "HttpClient",
    link: "https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient",
    description: ".NET Standard HTTP Client",
    extname: ".cs"
  },
  convert: ({ allHeaders: e, postData: n, method: t, fullUrl: a }, c) => {
    var m, x;
    const s = {
      indent: "    ",
      ...c
    }, { push: r, join: o } = new V({ indent: s.indent });
    r("using System.Net.Http.Headers;");
    let i = "";
    const l = !!e.cookie, f = Eo(e);
    (l || f.length) && (i = "clientHandler", r("var clientHandler = new HttpClientHandler"), r("{"), l && r("UseCookies = false,", 1), f.length && r(`AutomaticDecompression = ${f.join(" | ")},`, 1), r("};")), r(`var client = new HttpClient(${i});`), r("var request = new HttpRequestMessage"), r("{");
    const y = ["GET", "POST", "PUT", "DELETE", "PATCH", "HEAD", "OPTIONS", "TRACE"];
    t = t.toUpperCase(), t && y.includes(t) ? t = `HttpMethod.${t[0]}${t.substring(1).toLowerCase()}` : t = `new HttpMethod("${t}")`, r(`Method = ${t},`, 1), r(`RequestUri = new Uri("${a}"),`, 1);
    const g = Object.keys(e).filter((v) => {
      switch (v.toLowerCase()) {
        case "content-type":
        case "content-length":
        case "accept-encoding":
          return !1;
        default:
          return !0;
      }
    });
    if (g.length && (r("Headers =", 1), r("{", 1), g.forEach((v) => {
      r(`{ "${v}", "${le(e[v])}" },`, 2);
    }), r("},", 1)), n.text) {
      const v = n.mimeType;
      switch (v) {
        case "application/x-www-form-urlencoded":
          r("Content = new FormUrlEncodedContent(new Dictionary<string, string>", 1), r("{", 1), (m = n.params) == null || m.forEach((u) => {
            r(`{ "${u.name}", "${u.value}" },`, 2);
          }), r("}),", 1);
          break;
        case "multipart/form-data":
          r("Content = new MultipartFormDataContent", 1), r("{", 1), (x = n.params) == null || x.forEach((u) => {
            r(`new StringContent(${JSON.stringify(u.value || "")})`, 2), r("{", 2), r("Headers =", 3), r("{", 3), u.contentType && r(`ContentType = new MediaTypeHeaderValue("${u.contentType}"),`, 4), r('ContentDisposition = new ContentDispositionHeaderValue("form-data")', 4), r("{", 4), r(`Name = "${u.name}",`, 5), u.fileName && r(`FileName = "${u.fileName}",`, 5), r("}", 4), r("}", 3), r("},", 2);
          }), r("},", 1);
          break;
        default:
          r(`Content = new StringContent(${JSON.stringify(n.text || "")})`, 1), r("{", 1), r("Headers =", 2), r("{", 2), r(`ContentType = new MediaTypeHeaderValue("${v}")`, 3), r("}", 2), r("}", 1);
          break;
      }
    }
    return r("};"), r("using (var response = await client.SendAsync(request))"), r("{"), r("response.EnsureSuccessStatusCode();", 1), r("var body = await response.Content.ReadAsStringAsync();", 1), r("Console.WriteLine(body);", 1), r("}"), o();
  }
};
function ko(e) {
  return e[0].toUpperCase() + e.slice(1).toLowerCase();
}
var No = {
  info: {
    key: "restsharp",
    title: "RestSharp",
    link: "http://restsharp.org/",
    description: "Simple REST and HTTP API Client for .NET",
    extname: ".cs",
    installation: "dotnet add package RestSharp"
  },
  convert: ({ method: e, fullUrl: n, headersObj: t, cookies: a, postData: c, uriObj: s }) => {
    const { push: r, join: o } = new V();
    if (!["GET", "POST", "PUT", "DELETE", "PATCH", "HEAD", "OPTIONS"].includes(
      e.toUpperCase()
    ))
      return "Method not supported";
    r(`using RestSharp;

`), r(`var options = new RestClientOptions("${n}");`), r("var client = new RestClient(options);"), r('var request = new RestRequest("");');
    const l = c.mimeType && c.mimeType === "multipart/form-data";
    switch (l && r("request.AlwaysMultipartFormData = true;"), Object.keys(t).forEach((f) => {
      if (c.mimeType && f.toLowerCase() === "content-type" && c.text) {
        l && c.boundary && r(`request.FormBoundary = "${c.boundary}";`);
        return;
      }
      r(`request.AddHeader("${f}", "${le(t[f])}");`);
    }), a.forEach(({ name: f, value: y }) => {
      r(`request.AddCookie("${f}", "${le(y)}", "${s.pathname}", "${s.host}");`);
    }), c.mimeType) {
      case "multipart/form-data":
        if (!c.params) break;
        c.params.forEach((f) => {
          f.fileName ? r(`request.AddFile("${f.name}", "${f.fileName}");`) : r(`request.AddParameter("${f.name}", "${f.value}");`);
        });
        break;
      case "application/x-www-form-urlencoded":
        if (!c.params) break;
        c.params.forEach((f) => {
          r(`request.AddParameter("${f.name}", "${f.value}");`);
        });
        break;
      case "application/json": {
        if (!c.text) break;
        const f = JSON.stringify(c.text);
        r(`request.AddJsonBody(${f}, false);`);
        break;
      }
      default:
        if (!c.text) break;
        r(`request.AddStringBody("${c.text}", "${c.mimeType}");`);
    }
    return r(`var response = await client.${ko(e)}Async(request);
`), r(`Console.WriteLine("{0}", response.Content);
`), o();
  }
}, Ao = {
  info: {
    key: "csharp",
    title: "C#",
    default: "restsharp",
    cli: "dotnet"
  },
  clientsById: {
    httpclient: Ro,
    restsharp: No
  }
}, Io = {
  info: {
    key: "native",
    title: "NewRequest",
    link: "http://golang.org/pkg/net/http/#NewRequest",
    description: "Golang HTTP client request",
    extname: ".go"
  },
  convert: ({ postData: e, method: n, allHeaders: t, fullUrl: a }, c = {}) => {
    const { blank: s, push: r, join: o } = new V({ indent: "	" }), { showBoilerplate: i = !0, checkErrors: l = !1, printBody: f = !0, timeout: y = -1 } = c, g = l ? "err" : "_", m = i ? 1 : 0, x = () => {
      l && (r("if err != nil {", m), r("panic(err)", m + 1), r("}", m));
    };
    i && (r("package main"), s(), r("import ("), r('"fmt"', m), y > 0 && r('"time"', m), e.text && r('"strings"', m), r('"net/http"', m), f && r('"io"', m), r(")"), s(), r("func main() {"), s());
    const v = y > 0, u = v, h = u ? "client" : "http.DefaultClient";
    return u && (r("client := http.Client{", m), v && r(`Timeout: time.Duration(${y} * time.Second),`, m + 1), r("}", m), s()), r(`url := "${a}"`, m), s(), e.text ? (r(`payload := strings.NewReader(${JSON.stringify(e.text)})`, m), s(), r(`req, ${g} := http.NewRequest("${n}", url, payload)`, m), s()) : (r(`req, ${g} := http.NewRequest("${n}", url, nil)`, m), s()), x(), Object.keys(t).length && (Object.keys(t).forEach((d) => {
      r(`req.Header.Add("${d}", "${le(t[d])}")`, m);
    }), s()), r(`res, ${g} := ${h}.Do(req)`, m), x(), f && (s(), r("defer res.Body.Close()", m), r(`body, ${g} := io.ReadAll(res.Body)`, m), x()), s(), f && r("fmt.Println(string(body))", m), i && (s(), r("}")), o();
  }
}, Uo = {
  info: {
    key: "go",
    title: "Go",
    default: "native",
    cli: "go"
  },
  clientsById: {
    native: Io
  }
}, Dr = `\r
`, Lo = {
  info: {
    key: "http1.1",
    title: "HTTP/1.1",
    link: "https://tools.ietf.org/html/rfc7230",
    description: "HTTP/1.1 request string in accordance with RFC 7230",
    extname: null
  },
  convert: ({ method: e, fullUrl: n, uriObj: t, httpVersion: a, allHeaders: c, postData: s }, r) => {
    const o = {
      absoluteURI: !1,
      autoContentLength: !0,
      autoHost: !0,
      ...r
    }, { blank: i, push: l, join: f } = new V({ indent: "", join: Dr }), y = o.absoluteURI ? n : t.path;
    l(`${e} ${y} ${a}`);
    const g = Object.keys(c);
    if (g.forEach((v) => {
      const u = v.toLowerCase().replace(/(^|-)(\w)/g, (h) => h.toUpperCase());
      l(`${u}: ${c[v]}`);
    }), o.autoHost && !g.includes("host") && l(`Host: ${t.host}`), o.autoContentLength && s.text && !g.includes("content-length")) {
      const u = new TextEncoder().encode(s.text).length.toString();
      l(`Content-Length: ${u}`);
    }
    i();
    const m = f(), x = s.text || "";
    return `${m}${Dr}${x}`;
  }
}, Fo = {
  info: {
    key: "http",
    title: "HTTP",
    default: "http1.1"
  },
  clientsById: {
    "http1.1": Lo
  }
}, _o = {
  info: {
    key: "asynchttp",
    title: "AsyncHttp",
    link: "https://github.com/AsyncHttpClient/async-http-client",
    description: "Asynchronous Http and WebSocket Client library for Java",
    extname: ".java"
  },
  convert: ({ method: e, allHeaders: n, postData: t, fullUrl: a }, c) => {
    const s = {
      indent: "  ",
      ...c
    }, { blank: r, push: o, join: i } = new V({ indent: s.indent });
    return o("AsyncHttpClient client = new DefaultAsyncHttpClient();"), o(`client.prepare("${e.toUpperCase()}", "${a}")`), Object.keys(n).forEach((l) => {
      o(`.setHeader("${l}", "${le(n[l])}")`, 1);
    }), t.text && o(`.setBody(${JSON.stringify(t.text)})`, 1), o(".execute()", 1), o(".toCompletableFuture()", 1), o(".thenAccept(System.out::println)", 1), o(".join();", 1), r(), o("client.close();"), i();
  }
}, Bo = {
  info: {
    key: "nethttp",
    title: "java.net.http",
    link: "https://openjdk.java.net/groups/net/httpclient/intro.html",
    description: "Java Standardized HTTP Client API",
    extname: ".java"
  },
  convert: ({ allHeaders: e, fullUrl: n, method: t, postData: a }, c) => {
    const s = {
      indent: "  ",
      ...c
    }, { push: r, join: o } = new V({ indent: s.indent });
    return r("HttpRequest request = HttpRequest.newBuilder()"), r(`.uri(URI.create("${n}"))`, 2), Object.keys(e).forEach((i) => {
      r(`.header("${i}", "${le(e[i])}")`, 2);
    }), a.text ? r(
      `.method("${t.toUpperCase()}", HttpRequest.BodyPublishers.ofString(${JSON.stringify(a.text)}))`,
      2
    ) : r(`.method("${t.toUpperCase()}", HttpRequest.BodyPublishers.noBody())`, 2), r(".build();", 2), r(
      "HttpResponse<String> response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString());"
    ), r("System.out.println(response.body());"), o();
  }
}, Ho = {
  info: {
    key: "okhttp",
    title: "OkHttp",
    link: "http://square.github.io/okhttp/",
    description: "An HTTP Request Client Library",
    extname: ".java"
  },
  convert: ({ postData: e, method: n, fullUrl: t, allHeaders: a }, c) => {
    const s = {
      indent: "  ",
      ...c
    }, { push: r, blank: o, join: i } = new V({ indent: s.indent }), l = ["GET", "POST", "PUT", "DELETE", "PATCH", "HEAD"], f = ["POST", "PUT", "DELETE", "PATCH"];
    return r("OkHttpClient client = new OkHttpClient();"), o(), e.text && (e.boundary ? r(`MediaType mediaType = MediaType.parse("${e.mimeType}; boundary=${e.boundary}");`) : r(`MediaType mediaType = MediaType.parse("${e.mimeType}");`), r(`RequestBody body = RequestBody.create(mediaType, ${JSON.stringify(e.text)});`)), r("Request request = new Request.Builder()"), r(`.url("${t}")`, 1), l.includes(n.toUpperCase()) ? f.includes(n.toUpperCase()) ? e.text ? r(`.${n.toLowerCase()}(body)`, 1) : r(`.${n.toLowerCase()}(null)`, 1) : r(`.${n.toLowerCase()}()`, 1) : e.text ? r(`.method("${n.toUpperCase()}", body)`, 1) : r(`.method("${n.toUpperCase()}", null)`, 1), Object.keys(a).forEach((y) => {
      r(`.addHeader("${y}", "${le(a[y])}")`, 1);
    }), r(".build();", 1), o(), r("Response response = client.newCall(request).execute();"), i();
  }
}, Mo = {
  info: {
    key: "unirest",
    title: "Unirest",
    link: "http://unirest.io/java.html",
    description: "Lightweight HTTP Request Client Library",
    extname: ".java"
  },
  convert: ({ method: e, allHeaders: n, postData: t, fullUrl: a }, c) => {
    const s = {
      indent: "  ",
      ...c
    }, { join: r, push: o } = new V({ indent: s.indent });
    return ["GET", "POST", "PUT", "DELETE", "PATCH", "HEAD", "OPTIONS"].includes(e.toUpperCase()) ? o(`HttpResponse<String> response = Unirest.${e.toLowerCase()}("${a}")`) : o(`HttpResponse<String> response = Unirest.customMethod("${e.toUpperCase()}","${a}")`), Object.keys(n).forEach((l) => {
      o(`.header("${l}", "${le(n[l])}")`, 1);
    }), t.text && o(`.body(${JSON.stringify(t.text)})`, 1), o(".asString();", 1), r();
  }
}, zo = {
  info: {
    key: "java",
    title: "Java",
    default: "unirest"
  },
  clientsById: {
    asynchttp: _o,
    nethttp: Bo,
    okhttp: Ho,
    unirest: Mo
  }
}, Wo = {
  info: {
    key: "axios",
    title: "Axios",
    link: "https://github.com/axios/axios",
    description: "Promise based HTTP client for the browser and node.js",
    extname: ".js",
    installation: "npm install axios --save"
  },
  convert: ({ allHeaders: e, method: n, url: t, queryObj: a, postData: c }, s) => {
    const r = {
      indent: "  ",
      ...s
    }, { blank: o, push: i, join: l, addPostProcessor: f } = new V({ indent: r.indent });
    i("import axios from 'axios';"), o();
    const y = {
      method: n,
      url: t
    };
    switch (Object.keys(a).length && (y.params = a), Object.keys(e).length && (y.headers = e), c.mimeType) {
      case "application/x-www-form-urlencoded":
        c.params && (i("const encodedParams = new URLSearchParams();"), c.params.forEach((m) => {
          i(`encodedParams.set('${m.name}', '${m.value}');`);
        }), o(), y.data = "encodedParams,", f((m) => m.replace(/'encodedParams,'/, "encodedParams,")));
        break;
      case "application/json":
        c.jsonObj && (y.data = c.jsonObj);
        break;
      case "multipart/form-data":
        if (!c.params)
          break;
        i("const form = new FormData();"), c.params.forEach((m) => {
          i(`form.append('${m.name}', '${m.value || m.fileName || ""}');`);
        }), o(), y.data = "[form]";
        break;
      default:
        c.text && (y.data = c.text);
    }
    const g = ue(y, {
      indent: "  ",
      inlineCharacterLimit: 80
    }).replace('"[form]"', "form");
    return i(`const options = ${g};`), o(), i("axios"), i(".request(options)", 1), i(".then(function (response) {", 1), i("console.log(response.data);", 2), i("})", 1), i(".catch(function (error) {", 1), i("console.error(error);", 2), i("});", 1), l();
  }
}, Jo = {
  info: {
    key: "fetch",
    title: "fetch",
    link: "https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch",
    description: "Perform asynchronous HTTP requests with the Fetch API",
    extname: ".js"
  },
  convert: ({ method: e, allHeaders: n, postData: t, fullUrl: a }, c) => {
    const s = {
      indent: "  ",
      credentials: null,
      ...c
    }, { blank: r, join: o, push: i } = new V({ indent: s.indent }), l = {
      method: e
    };
    switch (Object.keys(n).length && (l.headers = n), s.credentials !== null && (l.credentials = s.credentials), t.mimeType) {
      case "application/x-www-form-urlencoded":
        l.body = t.paramsObj ? t.paramsObj : t.text;
        break;
      case "application/json":
        t.jsonObj && (l.body = t.jsonObj);
        break;
      case "multipart/form-data":
        if (!t.params)
          break;
        const f = fe(n, "content-type");
        f && delete n[f], i("const form = new FormData();"), t.params.forEach((y) => {
          i(`form.append('${y.name}', '${y.value || y.fileName || ""}');`);
        }), r();
        break;
      default:
        t.text && (l.body = t.text);
    }
    return l.headers && !Object.keys(l.headers).length && delete l.headers, i(
      `const options = ${ue(l, {
        indent: s.indent,
        inlineCharacterLimit: 80,
        // The Fetch API body only accepts string parameters, but stringified JSON can be difficult
        // to read, so we keep the object as a literal and use this transform function to wrap the
        // literal in a `JSON.stringify` call.
        transform: (f, y, g) => {
          if (y === "body") {
            if (t.mimeType === "application/x-www-form-urlencoded")
              return `new URLSearchParams(${g})`;
            if (t.mimeType === "application/json")
              return `JSON.stringify(${g})`;
          }
          return g;
        }
      })};`
    ), r(), t.params && t.mimeType === "multipart/form-data" && (i("options.body = form;"), r()), i(`fetch('${a}', options)`), i(".then(response => response.json())", 1), i(".then(response => console.log(response))", 1), i(".catch(err => console.error(err));", 1), o();
  }
}, Go = {
  info: {
    key: "jquery",
    title: "jQuery",
    link: "http://api.jquery.com/jquery.ajax/",
    description: "Perform an asynchronous HTTP (Ajax) requests with jQuery",
    extname: ".js"
  },
  convert: ({ fullUrl: e, method: n, allHeaders: t, postData: a }, c) => {
    var y;
    const s = {
      indent: "  ",
      ...c
    }, { blank: r, push: o, join: i } = new V({ indent: s.indent }), l = {
      async: !0,
      crossDomain: !0,
      url: e,
      method: n,
      headers: t
    };
    switch (a.mimeType) {
      case "application/x-www-form-urlencoded":
        l.data = a.paramsObj ? a.paramsObj : a.text;
        break;
      case "application/json":
        l.processData = !1, l.data = a.text;
        break;
      case "multipart/form-data":
        if (!a.params)
          break;
        if (o("const form = new FormData();"), a.params.forEach((g) => {
          o(`form.append('${g.name}', '${g.value || g.fileName || ""}');`);
        }), l.processData = !1, l.contentType = !1, l.mimeType = "multipart/form-data", l.data = "[form]", Ze(t, "content-type") && (y = ge(t, "content-type")) != null && y.includes("boundary")) {
          const g = fe(t, "content-type");
          g && delete l.headers[g];
        }
        r();
        break;
      default:
        a.text && (l.data = a.text);
    }
    const f = ue(l, { indent: s.indent }).replace("'[form]'", "form");
    return o(`const settings = ${f};`), r(), o("$.ajax(settings).done(function (response) {"), o("console.log(response);", 1), o("});"), i();
  }
}, Vo = {
  info: {
    key: "xhr",
    title: "XMLHttpRequest",
    link: "https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest",
    description: "W3C Standard API that provides scripted client functionality",
    extname: ".js"
  },
  convert: ({ postData: e, allHeaders: n, method: t, fullUrl: a }, c) => {
    var l;
    const s = {
      indent: "  ",
      cors: !0,
      ...c
    }, { blank: r, push: o, join: i } = new V({ indent: s.indent });
    switch (e.mimeType) {
      case "application/json":
        o(
          `const data = JSON.stringify(${ue(e.jsonObj, {
            indent: s.indent
          })});`
        ), r();
        break;
      case "multipart/form-data":
        if (!e.params)
          break;
        if (o("const data = new FormData();"), e.params.forEach((f) => {
          o(`data.append('${f.name}', '${f.value || f.fileName || ""}');`);
        }), Ze(n, "content-type") && (l = ge(n, "content-type")) != null && l.includes("boundary")) {
          const f = fe(n, "content-type");
          f && delete n[f];
        }
        r();
        break;
      default:
        o(`const data = ${e.text ? `'${e.text}'` : "null"};`), r();
    }
    return o("const xhr = new XMLHttpRequest();"), s.cors && o("xhr.withCredentials = true;"), r(), o("xhr.addEventListener('readystatechange', function () {"), o("if (this.readyState === this.DONE) {", 1), o("console.log(this.responseText);", 2), o("}", 1), o("});"), r(), o(`xhr.open('${t}', '${a}');`), Object.keys(n).forEach((f) => {
      o(`xhr.setRequestHeader('${f}', '${Fe(n[f])}');`);
    }), r(), o("xhr.send(data);"), i();
  }
}, Do = {
  info: {
    key: "javascript",
    title: "JavaScript",
    default: "xhr"
  },
  clientsById: {
    xhr: Vo,
    axios: Wo,
    fetch: Jo,
    jquery: Go
  }
}, Qo = {
  info: {
    key: "native",
    title: "Native JSON",
    link: "https://www.json.org/json-en.html",
    description: "A JSON represetation of any HAR payload.",
    extname: ".json"
  },
  convert: ({ postData: e }, n) => {
    const t = {
      indent: "  ",
      ...n
    };
    let a = "";
    switch (e.mimeType) {
      case "application/x-www-form-urlencoded":
        a = e.paramsObj ? e.paramsObj : e.text;
        break;
      case "application/json":
        e.jsonObj && (a = e.jsonObj);
        break;
      case "multipart/form-data":
        if (!e.params)
          break;
        const c = {};
        e.params.forEach((s) => {
          c[s.name] = s.value;
        }), a = c;
        break;
      default:
        e.text && (a = e.text);
    }
    return typeof a > "u" || a === "" ? "No JSON body" : JSON.stringify(a, null, t.indent);
  }
}, Ko = {
  info: {
    key: "json",
    title: "JSON",
    default: "native"
  },
  clientsById: {
    native: Qo
  }
}, Yo = {
  info: {
    key: "okhttp",
    title: "OkHttp",
    link: "http://square.github.io/okhttp/",
    description: "An HTTP Request Client Library",
    extname: ".kt"
  },
  convert: ({ postData: e, fullUrl: n, method: t, allHeaders: a }, c) => {
    const s = {
      indent: "  ",
      ...c
    }, { blank: r, join: o, push: i } = new V({ indent: s.indent }), l = ["GET", "POST", "PUT", "DELETE", "PATCH", "HEAD"], f = ["POST", "PUT", "DELETE", "PATCH"];
    return i("val client = OkHttpClient()"), r(), e.text && (e.boundary ? i(`val mediaType = MediaType.parse("${e.mimeType}; boundary=${e.boundary}")`) : i(`val mediaType = MediaType.parse("${e.mimeType}")`), i(`val body = RequestBody.create(mediaType, ${JSON.stringify(e.text)})`)), i("val request = Request.Builder()"), i(`.url("${n}")`, 1), l.includes(t.toUpperCase()) ? f.includes(t.toUpperCase()) ? e.text ? i(`.${t.toLowerCase()}(body)`, 1) : i(`.${t.toLowerCase()}(null)`, 1) : i(`.${t.toLowerCase()}()`, 1) : e.text ? i(`.method("${t.toUpperCase()}", body)`, 1) : i(`.method("${t.toUpperCase()}", null)`, 1), Object.keys(a).forEach((y) => {
      i(`.addHeader("${y}", "${le(a[y])}")`, 1);
    }), i(".build()", 1), r(), i("val response = client.newCall(request).execute()"), o();
  }
}, Xo = {
  info: {
    key: "kotlin",
    title: "Kotlin",
    default: "okhttp"
  },
  clientsById: {
    okhttp: Yo
  }
}, Zo = {
  info: {
    key: "axios",
    title: "Axios",
    link: "https://github.com/axios/axios",
    description: "Promise based HTTP client for the browser and node.js",
    extname: ".cjs",
    installation: "npm install axios --save"
  },
  convert: ({ method: e, fullUrl: n, allHeaders: t, postData: a }, c) => {
    const s = {
      indent: "  ",
      ...c
    }, { blank: r, join: o, push: i, addPostProcessor: l } = new V({ indent: s.indent });
    i("const axios = require('axios');");
    const f = {
      method: e,
      url: n
    };
    switch (Object.keys(t).length && (f.headers = t), a.mimeType) {
      case "application/x-www-form-urlencoded":
        a.params && (i("const { URLSearchParams } = require('url');"), r(), i("const encodedParams = new URLSearchParams();"), a.params.forEach((g) => {
          i(`encodedParams.set('${g.name}', '${g.value}');`);
        }), r(), f.data = "encodedParams,", l((g) => g.replace(/'encodedParams,'/, "encodedParams,")));
        break;
      case "application/json":
        r(), a.jsonObj && (f.data = a.jsonObj);
        break;
      default:
        r(), a.text && (f.data = a.text);
    }
    const y = ue(f, { indent: "  ", inlineCharacterLimit: 80 });
    return i(`const options = ${y};`), r(), i("axios"), i(".request(options)", 1), i(".then(function (response) {", 1), i("console.log(response.data);", 2), i("})", 1), i(".catch(function (error) {", 1), i("console.error(error);", 2), i("});", 1), o();
  }
}, ei = {
  info: {
    key: "fetch",
    title: "Fetch",
    link: "https://github.com/bitinn/node-fetch",
    description: "Simplified HTTP node-fetch client",
    extname: ".cjs",
    installation: "npm install node-fetch@2 --save"
  },
  convert: ({ method: e, fullUrl: n, postData: t, headersObj: a, cookies: c }, s) => {
    var v;
    const r = {
      indent: "  ",
      ...s
    };
    let o = !1;
    const { blank: i, push: l, join: f, unshift: y } = new V({ indent: r.indent });
    l("const fetch = require('node-fetch');");
    const g = n, m = {
      method: e
    };
    switch (Object.keys(a).length && (m.headers = a), t.mimeType) {
      case "application/x-www-form-urlencoded":
        y("const { URLSearchParams } = require('url');"), l("const encodedParams = new URLSearchParams();"), i(), (v = t.params) == null || v.forEach((h) => {
          l(`encodedParams.set('${h.name}', '${h.value}');`);
        }), m.body = "encodedParams";
        break;
      case "application/json":
        t.jsonObj && (m.body = t.jsonObj);
        break;
      case "multipart/form-data":
        if (!t.params)
          break;
        const u = fe(a, "content-type");
        u && delete a[u], y("const FormData = require('form-data');"), l("const formData = new FormData();"), i(), t.params.forEach((h) => {
          if (!h.fileName && !h.fileName && !h.contentType) {
            l(`formData.append('${h.name}', '${h.value}');`);
            return;
          }
          h.fileName && (o = !0, l(`formData.append('${h.name}', fs.createReadStream('${h.fileName}'));`));
        });
        break;
      default:
        t.text && (m.body = t.text);
    }
    if (c.length) {
      const u = c.map(({ name: h, value: d }) => `${encodeURIComponent(h)}=${encodeURIComponent(d)}`).join("; ");
      m.headers || (m.headers = {}), m.headers.cookie = u;
    }
    i(), l(`const url = '${g}';`), m.headers && !Object.keys(m.headers).length && delete m.headers;
    const x = ue(m, {
      indent: "  ",
      inlineCharacterLimit: 80,
      // The Fetch API body only accepts string parameters, but stringified JSON can be difficult to
      // read, so we keep the object as a literal and use this transform function to wrap the literal
      // in a `JSON.stringify` call.
      transform: (u, h, d) => h === "body" && t.mimeType === "application/json" ? `JSON.stringify(${d})` : d
    });
    return l(`const options = ${x};`), i(), o && y("const fs = require('fs');"), t.params && t.mimeType === "multipart/form-data" && (l("options.body = formData;"), i()), l("fetch(url, options)"), l(".then(res => res.json())", 1), l(".then(json => console.log(json))", 1), l(".catch(err => console.error('error:' + err));", 1), f().replace(/'encodedParams'/, "encodedParams").replace(/"fs\.createReadStream\(\\"(.+)\\"\)"/, 'fs.createReadStream("$1")');
  }
}, ti = {
  info: {
    key: "native",
    title: "HTTP",
    link: "http://nodejs.org/api/http.html#http_http_request_options_callback",
    description: "Node.js native HTTP interface",
    extname: ".cjs"
  },
  convert: ({ uriObj: e, method: n, allHeaders: t, postData: a }, c = {}) => {
    var y;
    const { indent: s = "  " } = c, { blank: r, join: o, push: i, unshift: l } = new V({ indent: s }), f = {
      method: n,
      hostname: e.hostname,
      port: e.port,
      path: e.path,
      headers: t
    };
    switch (i(`const http = require('${(y = e.protocol) == null ? void 0 : y.replace(":", "")}');`), r(), i(`const options = ${ue(f, { indent: s })};`), r(), i("const req = http.request(options, function (res) {"), i("const chunks = [];", 1), r(), i("res.on('data', function (chunk) {", 1), i("chunks.push(chunk);", 2), i("});", 1), r(), i("res.on('end', function () {", 1), i("const body = Buffer.concat(chunks);", 2), i("console.log(body.toString());", 2), i("});", 1), i("});"), r(), a.mimeType) {
      case "application/x-www-form-urlencoded":
        a.paramsObj && (l("const qs = require('querystring');"), i(
          `req.write(qs.stringify(${ue(a.paramsObj, {
            indent: "  ",
            inlineCharacterLimit: 80
          })}));`
        ));
        break;
      case "application/json":
        a.jsonObj && i(
          `req.write(JSON.stringify(${ue(a.jsonObj, {
            indent: "  ",
            inlineCharacterLimit: 80
          })}));`
        );
        break;
      default:
        a.text && i(`req.write(${ue(a.text, { indent: s })});`);
    }
    return i("req.end();"), o();
  }
}, ri = {
  info: {
    key: "request",
    title: "Request",
    link: "https://github.com/request/request",
    description: "Simplified HTTP request client",
    extname: ".cjs",
    installation: "npm install request --save"
  },
  convert: ({ method: e, url: n, fullUrl: t, postData: a, headersObj: c, cookies: s }, r) => {
    const o = {
      indent: "  ",
      ...r
    };
    let i = !1;
    const { push: l, blank: f, join: y, unshift: g, addPostProcessor: m } = new V({ indent: o.indent });
    l("const request = require('request');"), f();
    const x = {
      method: e,
      url: t
    };
    switch (Object.keys(c).length && (x.headers = c), a.mimeType) {
      case "application/x-www-form-urlencoded":
        x.form = a.paramsObj;
        break;
      case "application/json":
        a.jsonObj && (x.body = a.jsonObj, x.json = !0);
        break;
      case "multipart/form-data":
        if (!a.params)
          break;
        x.formData = {}, a.params.forEach((v) => {
          if (!v.fileName && !v.fileName && !v.contentType) {
            x.formData[v.name] = v.value;
            return;
          }
          let u = {};
          v.fileName ? (i = !0, u = {
            value: `fs.createReadStream(${v.fileName})`,
            options: {
              filename: v.fileName,
              contentType: v.contentType ? v.contentType : null
            }
          }) : v.value && (u.value = v.value), x.formData[v.name] = u;
        }), m((v) => v.replace(/'fs\.createReadStream\((.*)\)'/, "fs.createReadStream('$1')"));
        break;
      default:
        a.text && (x.body = a.text);
    }
    return s.length && (x.jar = "JAR", l("const jar = request.jar();"), s.forEach(({ name: v, value: u }) => {
      l(`jar.setCookie(request.cookie('${encodeURIComponent(v)}=${encodeURIComponent(u)}'), '${n}');`);
    }), f(), m((v) => v.replace(/'JAR'/, "jar"))), i && g("const fs = require('fs');"), l(`const options = ${ue(x, { indent: "  ", inlineCharacterLimit: 80 })};`), f(), l("request(options, function (error, response, body) {"), l("if (error) throw new Error(error);", 1), f(), l("console.log(body);", 1), l("});"), y();
  }
}, ni = {
  info: {
    key: "unirest",
    title: "Unirest",
    link: "http://unirest.io/nodejs.html",
    description: "Lightweight HTTP Request Client Library",
    extname: ".cjs"
  },
  convert: ({ method: e, url: n, cookies: t, queryObj: a, postData: c, headersObj: s }, r) => {
    const o = {
      indent: "  ",
      ...r
    };
    let i = !1;
    const { addPostProcessor: l, blank: f, join: y, push: g, unshift: m } = new V({
      indent: o.indent
    });
    switch (g("const unirest = require('unirest');"), f(), g(`const req = unirest('${e}', '${n}');`), f(), t.length && (g("const CookieJar = unirest.jar();"), t.forEach((x) => {
      g(`CookieJar.add('${encodeURIComponent(x.name)}=${encodeURIComponent(x.value)}', '${n}');`);
    }), g("req.jar(CookieJar);"), f()), Object.keys(a).length && (g(`req.query(${ue(a, { indent: o.indent })});`), f()), Object.keys(s).length && (g(`req.headers(${ue(s, { indent: o.indent })});`), f()), c.mimeType) {
      case "application/x-www-form-urlencoded":
        c.paramsObj && (g(`req.form(${ue(c.paramsObj, { indent: o.indent })});`), f());
        break;
      case "application/json":
        c.jsonObj && (g("req.type('json');"), g(`req.send(${ue(c.jsonObj, { indent: o.indent })});`), f());
        break;
      case "multipart/form-data": {
        if (!c.params)
          break;
        const x = [];
        c.params.forEach((v) => {
          const u = {};
          v.fileName && !v.value ? (i = !0, u.body = `fs.createReadStream('${v.fileName}')`, l((h) => h.replace(/'fs\.createReadStream\(\\'(.+)\\'\)'/, "fs.createReadStream('$1')"))) : v.value && (u.body = v.value), u.body && (v.contentType && (u["content-type"] = v.contentType), x.push(u));
        }), g(`req.multipart(${ue(x, { indent: o.indent })});`), f();
        break;
      }
      default:
        c.text && (g(`req.send(${ue(c.text, { indent: o.indent })});`), f());
    }
    return i && m("const fs = require('fs');"), g("req.end(function (res) {"), g("if (res.error) throw new Error(res.error);", 1), f(), g("console.log(res.body);", 1), g("});"), y();
  }
}, ai = {
  info: {
    key: "node",
    title: "Node.js",
    default: "native",
    cli: "node %s"
  },
  clientsById: {
    native: ti,
    request: ri,
    unirest: ni,
    axios: Zo,
    fetch: ei
  }
}, ht = (e, n, t, a) => {
  const c = `${e} *${n} = `, s = hr(t, a ? c.length : void 0);
  return `${c}${s};`;
}, hr = (e, n) => {
  const t = n === void 0 ? ", " : `,
   ${" ".repeat(n)}`;
  switch (Object.prototype.toString.call(e)) {
    case "[object Number]":
      return `@${e}`;
    case "[object Array]":
      return `@[ ${e.map((c) => hr(c)).join(t)} ]`;
    case "[object Object]": {
      const a = [];
      return Object.keys(e).forEach((c) => {
        a.push(`@"${c}": ${hr(e[c])}`);
      }), `@{ ${a.join(t)} }`;
    }
    case "[object Boolean]":
      return e ? "@YES" : "@NO";
    default:
      return e == null ? "" : `@"${e.toString().replace(/"/g, '\\"')}"`;
  }
}, oi = {
  info: {
    key: "nsurlsession",
    title: "NSURLSession",
    link: "https://developer.apple.com/library/mac/documentation/Foundation/Reference/NSURLSession_class/index.html",
    description: "Foundation's NSURLSession request",
    extname: ".m"
  },
  convert: ({ allHeaders: e, postData: n, method: t, fullUrl: a }, c) => {
    var f;
    const s = {
      indent: "    ",
      pretty: !0,
      timeout: 10,
      ...c
    }, { push: r, join: o, blank: i } = new V({ indent: s.indent }), l = {
      hasHeaders: !1,
      hasBody: !1
    };
    if (r("#import <Foundation/Foundation.h>"), Object.keys(e).length && (l.hasHeaders = !0, i(), r(ht("NSDictionary", "headers", e, s.pretty))), n.text || n.jsonObj || n.params)
      switch (l.hasBody = !0, n.mimeType) {
        case "application/x-www-form-urlencoded":
          if ((f = n.params) != null && f.length) {
            i();
            const [y, ...g] = n.params;
            r(
              `NSMutableData *postData = [[NSMutableData alloc] initWithData:[@"${y.name}=${y.value}" dataUsingEncoding:NSUTF8StringEncoding]];`
            ), g.forEach(({ name: m, value: x }) => {
              r(`[postData appendData:[@"&${m}=${x}" dataUsingEncoding:NSUTF8StringEncoding]];`);
            });
          } else
            l.hasBody = !1;
          break;
        case "application/json":
          n.jsonObj && (r(ht("NSDictionary", "parameters", n.jsonObj, s.pretty)), i(), r("NSData *postData = [NSJSONSerialization dataWithJSONObject:parameters options:0 error:nil];"));
          break;
        case "multipart/form-data":
          r(ht("NSArray", "parameters", n.params || [], s.pretty)), r(`NSString *boundary = @"${n.boundary}";`), i(), r("NSError *error;"), r("NSMutableString *body = [NSMutableString string];"), r("for (NSDictionary *param in parameters) {"), r('[body appendFormat:@"--%@\\r\\n", boundary];', 1), r('if (param[@"fileName"]) {', 1), r(
            '[body appendFormat:@"Content-Disposition:form-data; name=\\"%@\\"; filename=\\"%@\\"\\r\\n", param[@"name"], param[@"fileName"]];',
            2
          ), r('[body appendFormat:@"Content-Type: %@\\r\\n\\r\\n", param[@"contentType"]];', 2), r(
            '[body appendFormat:@"%@", [NSString stringWithContentsOfFile:param[@"fileName"] encoding:NSUTF8StringEncoding error:&error]];',
            2
          ), r("if (error) {", 2), r('NSLog(@"%@", error);', 3), r("}", 2), r("} else {", 1), r('[body appendFormat:@"Content-Disposition:form-data; name=\\"%@\\"\\r\\n\\r\\n", param[@"name"]];', 2), r('[body appendFormat:@"%@", param[@"value"]];', 2), r("}", 1), r("}"), r('[body appendFormat:@"\\r\\n--%@--\\r\\n", boundary];'), r("NSData *postData = [body dataUsingEncoding:NSUTF8StringEncoding];");
          break;
        default:
          i(), r(
            `NSData *postData = [[NSData alloc] initWithData:[@"${n.text}" dataUsingEncoding:NSUTF8StringEncoding]];`
          );
      }
    return i(), r(`NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:@"${a}"]`), r("                                                       cachePolicy:NSURLRequestUseProtocolCachePolicy"), r(`                                                   timeoutInterval:${s.timeout.toFixed(1)}];`), r(`[request setHTTPMethod:@"${t}"];`), l.hasHeaders && r("[request setAllHTTPHeaderFields:headers];"), l.hasBody && r("[request setHTTPBody:postData];"), i(), r("NSURLSession *session = [NSURLSession sharedSession];"), r("NSURLSessionDataTask *dataTask = [session dataTaskWithRequest:request"), r(
      "                                            completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {"
    ), r("                                            if (error) {", 1), r('                                            NSLog(@"%@", error);', 2), r("                                            } else {", 1), r(
      "                                            NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *) response;",
      2
    ), r('                                            NSLog(@"%@", httpResponse);', 2), r("                                            }", 1), r("                                            }];"), r("[dataTask resume];"), o();
  }
}, ii = {
  info: {
    key: "objc",
    title: "Objective-C",
    default: "nsurlsession"
  },
  clientsById: {
    nsurlsession: oi
  }
}, si = {
  info: {
    key: "cohttp",
    title: "CoHTTP",
    link: "https://github.com/mirage/ocaml-cohttp",
    description: "Cohttp is a very lightweight HTTP server using Lwt or Async for OCaml",
    extname: ".ml",
    installation: "opam install cohttp-lwt-unix cohttp-async"
  },
  convert: ({ fullUrl: e, allHeaders: n, postData: t, method: a }, c) => {
    const s = {
      indent: "  ",
      ...c
    }, r = ["get", "post", "head", "delete", "patch", "put", "options"], { push: o, blank: i, join: l } = new V({ indent: s.indent });
    o("open Cohttp_lwt_unix"), o("open Cohttp"), o("open Lwt"), i(), o(`let uri = Uri.of_string "${e}" in`);
    const f = Object.keys(n);
    f.length === 1 ? o(
      `let headers = Header.add (Header.init ()) "${f[0]}" "${le(
        n[f[0]]
      )}" in`
    ) : f.length > 1 && (o("let headers = Header.add_list (Header.init ()) ["), f.forEach((x) => {
      o(`("${x}", "${le(n[x])}");`, 1);
    }), o("] in")), t.text && o(`let body = Cohttp_lwt_body.of_string ${JSON.stringify(t.text)} in`), i();
    const y = f.length ? "~headers " : "", g = t.text ? "~body " : "", m = r.includes(a.toLowerCase()) ? `\`${a.toUpperCase()}` : `(Code.method_of_string "${a}")`;
    return o(`Client.call ${y}${g}${m} uri`), o(">>= fun (res, body_stream) ->"), o("(* Do stuff with the result *)", 1), l();
  }
}, li = {
  info: {
    key: "ocaml",
    title: "OCaml",
    default: "cohttp"
  },
  clientsById: {
    cohttp: si
  }
}, te = (e, n, t) => {
  switch (t = t || "", n = n || "", Object.prototype.toString.call(e)) {
    case "[object Boolean]":
      return e;
    case "[object Null]":
      return "null";
    case "[object Undefined]":
      return "null";
    case "[object String]":
      return `'${Le(e, { delimiter: "'", escapeNewlines: !1 })}'`;
    case "[object Number]":
      return e.toString();
    case "[object Array]": {
      const a = e.map((c) => te(c, `${n}${n}`, n)).join(`,
${n}`);
      return `[
${n}${a}
${t}]`;
    }
    case "[object Object]": {
      const a = [];
      for (const c in e)
        Object.prototype.hasOwnProperty.call(e, c) && a.push(`${te(c, n)} => ${te(e[c], `${n}${n}`, n)}`);
      return `[
${n}${a.join(`,
${n}`)}
${t}]`;
    }
    default:
      return "null";
  }
}, Qr = [
  "ACL",
  "BASELINE_CONTROL",
  "CHECKIN",
  "CHECKOUT",
  "CONNECT",
  "COPY",
  "DELETE",
  "GET",
  "HEAD",
  "LABEL",
  "LOCK",
  "MERGE",
  "MKACTIVITY",
  "MKCOL",
  "MKWORKSPACE",
  "MOVE",
  "OPTIONS",
  "POST",
  "PROPFIND",
  "PROPPATCH",
  "PUT",
  "REPORT",
  "TRACE",
  "UNCHECKOUT",
  "UNLOCK",
  "UPDATE",
  "VERSION_CONTROL"
], ci = {
  info: {
    key: "curl",
    title: "cURL",
    link: "http://php.net/manual/en/book.curl.php",
    description: "PHP with ext-curl",
    extname: ".php"
  },
  convert: ({ uriObj: e, postData: n, fullUrl: t, method: a, httpVersion: c, cookies: s, headersObj: r }, o = {}) => {
    const {
      closingTag: i = !1,
      indent: l = "  ",
      maxRedirects: f = 10,
      namedErrors: y = !1,
      noTags: g = !1,
      shortTags: m = !1,
      timeout: x = 30
    } = o, { push: v, blank: u, join: h } = new V({ indent: l });
    g || (v(m ? "<?" : "<?php"), u()), v("$curl = curl_init();"), u();
    const d = [
      {
        escape: !0,
        name: "CURLOPT_PORT",
        value: e.port
      },
      {
        escape: !0,
        name: "CURLOPT_URL",
        value: t
      },
      {
        escape: !1,
        name: "CURLOPT_RETURNTRANSFER",
        value: "true"
      },
      {
        escape: !0,
        name: "CURLOPT_ENCODING",
        value: ""
      },
      {
        escape: !1,
        name: "CURLOPT_MAXREDIRS",
        value: f
      },
      {
        escape: !1,
        name: "CURLOPT_TIMEOUT",
        value: x
      },
      {
        escape: !1,
        name: "CURLOPT_HTTP_VERSION",
        value: c === "HTTP/1.0" ? "CURL_HTTP_VERSION_1_0" : "CURL_HTTP_VERSION_1_1"
      },
      {
        escape: !0,
        name: "CURLOPT_CUSTOMREQUEST",
        value: a
      },
      {
        escape: !n.jsonObj,
        name: "CURLOPT_POSTFIELDS",
        value: n ? n.jsonObj ? `json_encode(${te(n.jsonObj, l.repeat(2), l)})` : n.text : void 0
      }
    ];
    v("curl_setopt_array($curl, [");
    const j = new V({ indent: l, join: `
${l}` });
    d.forEach(({ value: O, name: b, escape: q }) => {
      O != null && j.push(`${b} => ${q ? JSON.stringify(O) : O},`);
    });
    const T = s.map((O) => `${encodeURIComponent(O.name)}=${encodeURIComponent(O.value)}`);
    T.length && j.push(`CURLOPT_COOKIE => "${T.join("; ")}",`);
    const p = Object.keys(r).sort().map((O) => `"${O}: ${le(r[O])}"`);
    return p.length && (j.push("CURLOPT_HTTPHEADER => ["), j.push(p.join(`,
${l}${l}`), 1), j.push("],")), v(j.join(), 1), v("]);"), u(), v("$response = curl_exec($curl);"), v("$err = curl_error($curl);"), u(), v("curl_close($curl);"), u(), v("if ($err) {"), v(y ? 'echo array_flip(get_defined_constants(true)["curl"])[$err];' : 'echo "cURL Error #:" . $err;', 1), v("} else {"), v("echo $response;", 1), v("}"), !g && i && (u(), v("?>")), h();
  }
}, ui = {
  info: {
    key: "guzzle",
    title: "Guzzle",
    link: "http://docs.guzzlephp.org/en/stable/",
    description: "PHP with Guzzle",
    extname: ".php",
    installation: "composer require guzzlehttp/guzzle"
  },
  convert: ({ postData: e, fullUrl: n, method: t, cookies: a, headersObj: c }, s) => {
    var v;
    const r = {
      closingTag: !1,
      indent: "  ",
      noTags: !1,
      shortTags: !1,
      ...s
    }, { push: o, blank: i, join: l } = new V({ indent: r.indent }), { code: f, push: y, join: g } = new V({ indent: r.indent });
    switch (r.noTags || o(r.shortTags ? "<?" : "<?php"), o("require_once('vendor/autoload.php');"), i(), e.mimeType) {
      case "application/x-www-form-urlencoded":
        y(`'form_params' => ${te(e.paramsObj, r.indent + r.indent, r.indent)},`, 1);
        break;
      case "multipart/form-data": {
        const u = [];
        if (e.params && e.params.forEach((h) => {
          if (h.fileName) {
            const d = {
              name: h.name,
              filename: h.fileName,
              contents: h.value
            };
            h.contentType && (d.headers = { "Content-Type": h.contentType }), u.push(d);
          } else h.value && u.push({
            name: h.name,
            contents: h.value
          });
        }), u.length && (y(`'multipart' => ${te(u, r.indent + r.indent, r.indent)}`, 1), Ze(c, "content-type") && (v = ge(c, "content-type")) != null && v.indexOf("boundary"))) {
          const h = fe(c, "content-type");
          h && delete c[h];
        }
        break;
      }
      default:
        e.text && y(`'body' => ${te(e.text)},`, 1);
    }
    const m = Object.keys(c).sort().map(function(u) {
      return `${r.indent}${r.indent}'${u}' => '${Fe(c[u])}',`;
    }), x = a.map((u) => `${encodeURIComponent(u.name)}=${encodeURIComponent(u.value)}`).join("; ");
    return x.length && m.push(`${r.indent}${r.indent}'cookie' => '${Fe(x)}',`), m.length && (y("'headers' => [", 1), y(m.join(`
`)), y("],", 1)), o("$client = new \\GuzzleHttp\\Client();"), i(), f.length ? (o(`$response = $client->request('${t}', '${n}', [`), o(g()), o("]);")) : o(`$response = $client->request('${t}', '${n}');`), i(), o("echo $response->getBody();"), !r.noTags && r.closingTag && (i(), o("?>")), l();
  }
}, fi = {
  info: {
    key: "http1",
    title: "HTTP v1",
    link: "http://php.net/manual/en/book.http.php",
    description: "PHP with pecl/http v1",
    extname: ".php"
  },
  convert: ({ method: e, url: n, postData: t, queryObj: a, headersObj: c, cookiesObj: s }, r = {}) => {
    const { closingTag: o = !1, indent: i = "  ", noTags: l = !1, shortTags: f = !1 } = r, { push: y, blank: g, join: m } = new V({ indent: i });
    switch (l || (y(f ? "<?" : "<?php"), g()), Qr.includes(e.toUpperCase()) || y(`HttpRequest::methodRegister('${e}');`), y("$request = new HttpRequest();"), y(`$request->setUrl(${te(n)});`), Qr.includes(e.toUpperCase()) ? y(`$request->setMethod(HTTP_METH_${e.toUpperCase()});`) : y(`$request->setMethod(HttpRequest::HTTP_METH_${e.toUpperCase()});`), g(), Object.keys(a).length && (y(`$request->setQueryData(${te(a, i)});`), g()), Object.keys(c).length && (y(`$request->setHeaders(${te(c, i)});`), g()), Object.keys(s).length && (y(`$request->setCookies(${te(s, i)});`), g()), t.mimeType) {
      case "application/x-www-form-urlencoded":
        y(`$request->setContentType(${te(t.mimeType)});`), y(`$request->setPostFields(${te(t.paramsObj, i)});`), g();
        break;
      case "application/json":
        y(`$request->setContentType(${te(t.mimeType)});`), y(`$request->setBody(json_encode(${te(t.jsonObj, i)}));`), g();
        break;
      default:
        t.text && (y(`$request->setBody(${te(t.text)});`), g());
    }
    return y("try {"), y("$response = $request->send();", 1), g(), y("echo $response->getBody();", 1), y("} catch (HttpException $ex) {"), y("echo $ex;", 1), y("}"), !l && o && (g(), y("?>")), m();
  }
}, pi = {
  info: {
    key: "http2",
    title: "HTTP v2",
    link: "http://devel-m6w6.rhcloud.com/mdref/http",
    description: "PHP with pecl/http v2",
    extname: ".php"
  },
  convert: ({ postData: e, headersObj: n, method: t, queryObj: a, cookiesObj: c, url: s }, r = {}) => {
    var v;
    const { closingTag: o = !1, indent: i = "  ", noTags: l = !1, shortTags: f = !1 } = r, { push: y, blank: g, join: m } = new V({ indent: i });
    let x = !1;
    switch (l || (y(f ? "<?" : "<?php"), g()), y("$client = new http\\Client;"), y("$request = new http\\Client\\Request;"), g(), e.mimeType) {
      case "application/x-www-form-urlencoded":
        y("$body = new http\\Message\\Body;"), y(`$body->append(new http\\QueryString(${te(e.paramsObj, i)}));`), g(), x = !0;
        break;
      case "multipart/form-data": {
        if (!e.params)
          break;
        const u = [], h = {};
        e.params.forEach(({ name: T, fileName: p, value: O, contentType: b }) => {
          if (p) {
            u.push({
              name: T,
              type: b,
              file: p,
              data: O
            });
            return;
          }
          O && (h[T] = O);
        });
        const d = Object.keys(h).length ? te(h, i) : "null", j = u.length ? te(u, i) : "null";
        if (y("$body = new http\\Message\\Body;"), y(`$body->addForm(${d}, ${j});`), Ze(n, "content-type") && (v = ge(n, "content-type")) != null && v.indexOf("boundary")) {
          const T = fe(n, "content-type");
          T && delete n[T];
        }
        g(), x = !0;
        break;
      }
      case "application/json":
        y("$body = new http\\Message\\Body;"), y(`$body->append(json_encode(${te(e.jsonObj, i)}));`), x = !0;
        break;
      default:
        e.text && (y("$body = new http\\Message\\Body;"), y(`$body->append(${te(e.text)});`), g(), x = !0);
    }
    return y(`$request->setRequestUrl(${te(s)});`), y(`$request->setRequestMethod(${te(t)});`), x && (y("$request->setBody($body);"), g()), Object.keys(a).length && (y(`$request->setQuery(new http\\QueryString(${te(a, i)}));`), g()), Object.keys(n).length && (y(`$request->setHeaders(${te(n, i)});`), g()), Object.keys(c).length && (g(), y(`$client->setCookies(${te(c, i)});`), g()), y("$client->enqueue($request)->send();"), y("$response = $client->getResponse();"), g(), y("echo $response->getBody();"), !l && o && (g(), y("?>")), m();
  }
}, di = {
  info: {
    key: "php",
    title: "PHP",
    default: "curl",
    cli: "php %s"
  },
  clientsById: {
    curl: ci,
    guzzle: ui,
    http1: fi,
    http2: pi
  }
}, ca = (e) => ({ method: t, headersObj: a, cookies: c, uriObj: s, fullUrl: r, postData: o, allHeaders: i }) => {
  const { push: l, join: f } = new V();
  if (!["GET", "POST", "PUT", "DELETE", "PATCH", "HEAD", "OPTIONS"].includes(t.toUpperCase()))
    return "Method not supported";
  const g = [], m = Object.keys(a);
  return m.length && (l("$headers=@{}"), m.forEach((x) => {
    x !== "connection" && l(`$headers.Add("${x}", "${Le(a[x], { escapeChar: "`" })}")`);
  }), g.push("-Headers $headers")), c.length && (l("$session = New-Object Microsoft.PowerShell.Commands.WebRequestSession"), c.forEach((x) => {
    l("$cookie = New-Object System.Net.Cookie"), l(`$cookie.Name = '${x.name}'`), l(`$cookie.Value = '${x.value}'`), l(`$cookie.Domain = '${s.host}'`), l("$session.Cookies.Add($cookie)");
  }), g.push("-WebSession $session")), o.text && (g.push(
    `-ContentType '${Le(ge(i, "content-type"), { delimiter: "'", escapeChar: "`" })}'`
  ), g.push(`-Body '${o.text}'`)), l(`$response = ${e} -Uri '${r}' -Method ${t} ${g.join(" ")}`.trim()), f();
}, hi = {
  info: {
    key: "restmethod",
    title: "Invoke-RestMethod",
    link: "https://docs.microsoft.com/en-us/powershell/module/Microsoft.PowerShell.Utility/Invoke-RestMethod",
    description: "Powershell Invoke-RestMethod client",
    extname: ".ps1"
  },
  convert: ca("Invoke-RestMethod")
}, yi = {
  info: {
    key: "webrequest",
    title: "Invoke-WebRequest",
    link: "https://docs.microsoft.com/en-us/powershell/module/Microsoft.PowerShell.Utility/Invoke-WebRequest",
    description: "Powershell Invoke-WebRequest client",
    extname: ".ps1"
  },
  convert: ca("Invoke-WebRequest")
}, mi = {
  info: {
    key: "powershell",
    title: "Powershell",
    default: "webrequest"
  },
  clientsById: {
    webrequest: yi,
    restmethod: hi
  }
};
function Kr(e, n, t, a, c) {
  const s = a.repeat(c), r = a.repeat(c - 1), o = t ? `,
${s}` : ", ", i = e === "object" ? "{" : "[", l = e === "object" ? "}" : "]";
  return t ? `${i}
${s}${n.join(o)}
${r}${l}` : e === "object" && n.length > 0 ? `${i} ${n.join(o)} ${l}` : `${i}${n.join(o)}${l}`;
}
var Re = (e, n, t) => {
  switch (t = t === void 0 ? 1 : t + 1, Object.prototype.toString.call(e)) {
    case "[object Number]":
      return e;
    case "[object Array]": {
      let a = !1;
      const c = e.map((s) => (Object.prototype.toString.call(s) === "[object Object]" && (a = Object.keys(s).length > 1), Re(s, n, t)));
      return Kr("array", c, a, n.indent, t);
    }
    case "[object Object]": {
      const a = [];
      for (const c in e)
        a.push(`"${c}": ${Re(e[c], n, t)}`);
      return Kr("object", a, n.pretty && a.length > 1, n.indent, t);
    }
    case "[object Null]":
      return "None";
    case "[object Boolean]":
      return e ? "True" : "False";
    default:
      return e == null ? "" : `"${e.toString().replace(/"/g, '\\"')}"`;
  }
}, gi = ["HEAD", "GET", "POST", "PUT", "PATCH", "DELETE", "OPTIONS"], vi = {
  info: {
    key: "requests",
    title: "Requests",
    link: "http://docs.python-requests.org/en/latest/api/#requests.request",
    description: "Requests HTTP library",
    extname: ".py",
    installation: "python -m pip install requests"
  },
  convert: ({ fullUrl: e, postData: n, allHeaders: t, method: a }, c) => {
    const s = {
      indent: "    ",
      pretty: !0,
      ...c
    }, { push: r, blank: o, join: i, addPostProcessor: l } = new V({ indent: s.indent });
    r("import requests"), o(), r(`url = "${e}"`), o();
    const f = t;
    let y = {};
    const g = {};
    let m = !1, x = !1, v = !1;
    switch (n.mimeType) {
      case "application/json":
        n.jsonObj && (r(`payload = ${Re(n.jsonObj, s)}`), v = !0, x = !0);
        break;
      case "multipart/form-data":
        if (!n.params)
          break;
        if (y = {}, n.params.forEach((d) => {
          d.fileName ? (d.contentType ? g[d.name] = `('${d.fileName}', open('${d.fileName}', 'rb'), '${d.contentType}')` : g[d.name] = `('${d.fileName}', open('${d.fileName}', 'rb'))`, m = !0) : (y[d.name] = d.value, x = !0);
        }), m) {
          r(`files = ${Re(g, s)}`), x && r(`payload = ${Re(y, s)}`);
          const d = fe(f, "content-type");
          d && delete f[d];
        } else {
          const d = JSON.stringify(n.text);
          d && (r(`payload = ${d}`), x = !0);
        }
        l(
          (d) => d.replace(/"\('(.+)', open\('(.+)', 'rb'\)\)"/g, '("$1", open("$2", "rb"))').replace(/"\('(.+)', open\('(.+)', 'rb'\), '(.+)'\)"/g, '("$1", open("$2", "rb"), "$3")')
        );
        break;
      default: {
        if (n.mimeType === "application/x-www-form-urlencoded" && n.paramsObj) {
          r(`payload = ${Re(n.paramsObj, s)}`), x = !0;
          break;
        }
        const d = JSON.stringify(n.text);
        d && (r(`payload = ${d}`), x = !0);
      }
    }
    const u = Object.keys(f).length;
    if (u === 0 && (x || m))
      o();
    else if (u === 1)
      Object.keys(f).forEach((d) => {
        r(`headers = {"${d}": "${le(f[d])}"}`), o();
      });
    else if (u > 1) {
      let d = 1;
      r("headers = {"), Object.keys(f).forEach((j) => {
        r(d !== u ? `"${j}": "${le(f[j])}",` : `"${j}": "${le(f[j])}"`, 1), d += 1;
      }), r("}"), o();
    }
    let h = gi.includes(a) ? `response = requests.${a.toLowerCase()}(url` : `response = requests.request("${a}", url`;
    return x && (v ? h += ", json=payload" : h += ", data=payload"), m && (h += ", files=files"), u > 0 && (h += ", headers=headers"), h += ")", r(h), o(), r("print(response.text)"), i();
  }
}, bi = {
  info: {
    key: "python",
    title: "Python",
    default: "requests",
    cli: "python3 %s"
  },
  clientsById: {
    requests: vi
  }
}, xi = {
  info: {
    key: "httr",
    title: "httr",
    link: "https://cran.r-project.org/web/packages/httr/vignettes/quickstart.html",
    description: "httr: Tools for Working with URLs and HTTP",
    extname: ".r"
  },
  convert: ({ url: e, queryObj: n, queryString: t, postData: a, allHeaders: c, method: s }) => {
    const { push: r, blank: o, join: i } = new V();
    r("library(httr)"), o(), r(`url <- "${e}"`), o();
    const l = n;
    delete n.key;
    const f = Object.keys(l).length;
    t.length === 1 ? (r(`queryString <- list(${Object.keys(l)} = "${Object.values(l).toString()}")`), o()) : t.length > 1 && (r("queryString <- list("), Object.keys(l).forEach((p, O) => {
      O !== f - 1 ? r(`  ${p} = "${l[p].toString()}",`) : r(`  ${p} = "${l[p].toString()}"`);
    }), r(")"), o());
    const y = JSON.stringify(a.text);
    if (y && (r(`payload <- ${y}`), o()), a.text || a.jsonObj || a.params)
      switch (a.mimeType) {
        case "application/x-www-form-urlencoded":
          r('encode <- "form"'), o();
          break;
        case "application/json":
          r('encode <- "json"'), o();
          break;
        case "multipart/form-data":
          r('encode <- "multipart"'), o();
          break;
        default:
          r('encode <- "raw"'), o();
          break;
      }
    const g = ge(c, "cookie"), m = ge(c, "accept"), x = g ? `set_cookies(\`${String(g).replace(/;/g, '", `').replace(/` /g, "`").replace(/[=]/g, '` = "')}")` : void 0, v = m ? `accept("${le(m)}")` : void 0, u = `content_type("${le(a.mimeType)}")`, h = Object.entries(c).filter(([p]) => !["cookie", "accept", "content-type"].includes(p.toLowerCase())).map(([p, O]) => `'${p}' = '${Fe(O)}'`).join(", "), d = h ? `add_headers(${h})` : void 0;
    let j = `response <- VERB("${s}", url`;
    y && (j += ", body = payload"), t.length && (j += ", query = queryString");
    const T = [d, u, v, x].filter((p) => !!p).join(", ");
    return T && (j += `, ${T}`), (a.text || a.jsonObj || a.params) && (j += ", encode = encode"), j += ")", r(j), o(), r('content(response, "text")'), i();
  }
}, $i = {
  info: {
    key: "r",
    title: "R",
    default: "httr"
  },
  clientsById: {
    httr: xi
  }
}, wi = {
  info: {
    key: "native",
    title: "net::http",
    link: "http://ruby-doc.org/stdlib-2.2.1/libdoc/net/http/rdoc/Net/HTTP.html",
    description: "Ruby HTTP client",
    extname: ".rb"
  },
  convert: ({ uriObj: e, method: n, fullUrl: t, postData: a, allHeaders: c }) => {
    const { push: s, blank: r, join: o } = new V();
    s("require 'uri'"), s("require 'net/http'"), r();
    const i = n.toUpperCase(), l = [
      "GET",
      "POST",
      "HEAD",
      "DELETE",
      "PATCH",
      "PUT",
      "OPTIONS",
      "COPY",
      "LOCK",
      "UNLOCK",
      "MOVE",
      "TRACE"
    ], f = i.charAt(0) + i.substring(1).toLowerCase();
    l.includes(i) || (s(`class Net::HTTP::${f} < Net::HTTPRequest`), s(`  METHOD = '${i.toUpperCase()}'`), s(`  REQUEST_HAS_BODY = '${a.text ? "true" : "false"}'`), s("  RESPONSE_HAS_BODY = true"), s("end"), r()), s(`url = URI("${t}")`), r(), s("http = Net::HTTP.new(url.host, url.port)"), e.protocol === "https:" && s("http.use_ssl = true"), r(), s(`request = Net::HTTP::${f}.new(url)`);
    const y = Object.keys(c);
    return y.length && y.forEach((g) => {
      s(`request["${g}"] = '${Fe(c[g])}'`);
    }), a.text && s(`request.body = ${JSON.stringify(a.text)}`), r(), s("response = http.request(request)"), s("puts response.read_body"), o();
  }
}, Si = {
  info: {
    key: "ruby",
    title: "Ruby",
    default: "native"
  },
  clientsById: {
    native: wi
  }
}, ce = (e = "") => /^[a-z0-9-_/.@%^=:]+$/i.test(e) ? e : `'${e.replace(/'/g, "'\\''")}'`, ji = (e) => e.replace(/\r/g, "\\r").replace(/\n/g, "\\n"), Oi = {
  "http1.0": "0",
  "url ": "",
  cookie: "b",
  data: "d",
  form: "F",
  globoff: "g",
  header: "H",
  insecure: "k",
  request: "X"
}, Ti = (e) => (n) => {
  if (e) {
    const t = Oi[n];
    return t ? `-${t}` : "";
  }
  return `--${n}`;
}, Pi = {
  info: {
    key: "curl",
    title: "cURL",
    link: "http://curl.haxx.se/",
    description: "cURL is a command line tool and library for transferring data with URL syntax",
    extname: ".sh"
  },
  convert: ({ fullUrl: e, method: n, httpVersion: t, headersObj: a, allHeaders: c, postData: s }, r = {}) => {
    var u;
    const { indent: o = "  ", short: i = !1, binary: l = !1, globOff: f = !1 } = r, y = "  ", { push: g, join: m } = new V({
      ...typeof o == "string" ? { indent: o } : {},
      join: o !== !1 ? ` \\
${o}` : " "
    }), x = Ti(i);
    let v = ce(e);
    if (g(`curl ${x("request")} ${n}`), f && (v = unescape(v), g(x("globoff"))), g(`${x("url ")}${v}`), t === "HTTP/1.0" && g(x("http1.0")), ge(c, "accept-encoding") && g("--compressed"), s.mimeType === "multipart/form-data") {
      const h = fe(a, "content-type");
      if (h) {
        const d = a[h];
        if (h && d) {
          const j = String(d).replace(/; boundary.+?(?=(;|$))/, "");
          a[h] = j, c[h] = j;
        }
      }
    }
    switch (Object.keys(a).sort().forEach((h) => {
      const d = `${h}: ${a[h]}`;
      g(`${x("header")} ${ce(d)}`);
    }), c.cookie && g(`${x("cookie")} ${ce(c.cookie)}`), s.mimeType) {
      case "multipart/form-data":
        (u = s.params) == null || u.forEach((d) => {
          let j = "";
          d.fileName ? j = `${d.name}='@${d.fileName}'` : j = ce(`${d.name}=${d.value}`), g(`${x("form")} ${j}`);
        });
        break;
      case "application/x-www-form-urlencoded":
        s.params ? s.params.forEach((d) => {
          const j = d.name, T = encodeURIComponent(d.name), p = T !== j, O = p ? T : j, b = l ? "--data-binary" : p ? "--data-urlencode" : x("data");
          g(`${b} ${ce(`${O}=${d.value}`)}`);
        }) : g(`${l ? "--data-binary" : x("data")} ${ce(s.text)}`);
        break;
      default:
        if (!s.text)
          break;
        let h = !1;
        if (To(s.mimeType) && s.text.length > 20)
          try {
            const d = JSON.parse(s.text);
            h = !0, s.text.indexOf("'") > 0 ? g(
              `${l ? "--data-binary" : x("data")} @- <<EOF
${JSON.stringify(
                d,
                null,
                y
              )}
EOF`
            ) : g(
              `${l ? "--data-binary" : x("data")} '
${JSON.stringify(d, null, y)}
'`
            );
          } catch {
          }
        h || g(`${l ? "--data-binary" : x("data")} ${ce(s.text)}`);
    }
    return m();
  }
}, qi = {
  info: {
    key: "httpie",
    title: "HTTPie",
    link: "http://httpie.org/",
    description: "a CLI, cURL-like tool for humans",
    extname: ".sh",
    installation: "brew install httpie"
  },
  convert: ({ allHeaders: e, postData: n, queryObj: t, fullUrl: a, method: c, url: s }, r) => {
    const o = {
      body: !1,
      cert: !1,
      headers: !1,
      indent: "  ",
      pretty: !1,
      print: !1,
      queryParams: !1,
      short: !1,
      style: !1,
      timeout: !1,
      verbose: !1,
      verify: !1,
      ...r
    }, { push: i, join: l, unshift: f } = new V({
      indent: o.indent,
      // @ts-expect-error SEEMS LEGIT
      join: o.indent !== !1 ? ` \\
${o.indent}` : " "
    });
    let y = !1;
    const g = [];
    o.headers && g.push(o.short ? "-h" : "--headers"), o.body && g.push(o.short ? "-b" : "--body"), o.verbose && g.push(o.short ? "-v" : "--verbose"), o.print && g.push(`${o.short ? "-p" : "--print"}=${o.print}`), o.verify && g.push(`--verify=${o.verify}`), o.cert && g.push(`--cert=${o.cert}`), o.pretty && g.push(`--pretty=${o.pretty}`), o.style && g.push(`--style=${o.style}`), o.timeout && g.push(`--timeout=${o.timeout}`), o.queryParams && Object.keys(t).forEach((x) => {
      const v = t[x];
      Array.isArray(v) ? v.forEach((u) => {
        i(`${x}==${ce(u)}`);
      }) : i(`${x}==${ce(v)}`);
    }), Object.keys(e).sort().forEach((x) => {
      i(`${x}:${ce(e[x])}`);
    }), n.mimeType === "application/x-www-form-urlencoded" ? n.params && n.params.length && (g.push(o.short ? "-f" : "--form"), n.params.forEach((x) => {
      i(`${x.name}=${ce(x.value)}`);
    })) : y = !0;
    const m = g.length ? `${g.join(" ")} ` : "";
    if (s = ce(o.queryParams ? s : a), f(`http ${m}${c} ${s}`), y && n.text) {
      const x = ce(n.text);
      f(`echo ${x} | `);
    }
    return l();
  }
}, Ci = {
  info: {
    key: "wget",
    title: "Wget",
    link: "https://www.gnu.org/software/wget/",
    description: "a free software package for retrieving files using HTTP, HTTPS",
    extname: ".sh"
  },
  convert: ({ method: e, postData: n, allHeaders: t, fullUrl: a }, c) => {
    const s = {
      indent: "  ",
      short: !1,
      verbose: !1,
      ...c
    }, { push: r, join: o } = new V({
      ...typeof s.indent == "string" ? { indent: s.indent } : {},
      join: s.indent !== !1 ? ` \\
${s.indent}` : " "
    });
    return s.verbose ? r(`wget ${s.short ? "-v" : "--verbose"}`) : r(`wget ${s.short ? "-q" : "--quiet"}`), r(`--method ${ce(e)}`), Object.keys(t).forEach((i) => {
      const l = `${i}: ${t[i]}`;
      r(`--header ${ce(l)}`);
    }), n.text && r(`--body-data ${ji(ce(n.text))}`), r(s.short ? "-O" : "--output-document"), r(`- ${ce(a)}`), o();
  }
}, Ei = {
  info: {
    key: "shell",
    title: "Shell",
    default: "curl",
    cli: "%s"
  },
  clientsById: {
    curl: Pi,
    httpie: qi,
    wget: Ci
  }
}, Yr = (e, n) => n.repeat(e), Xr = (e, n, t, a) => {
  const c = Yr(a, t), s = Yr(a - 1, t), r = n ? `,
${c}` : ", ";
  return n ? `[
${c}${e.join(r)}
${s}]` : `[${e.join(r)}]`;
}, Zr = (e, n, t) => `let ${e} = ${Xe(n, t)}`, Xe = (e, n, t) => {
  switch (t = t === void 0 ? 1 : t + 1, Object.prototype.toString.call(e)) {
    case "[object Number]":
      return e;
    case "[object Array]": {
      let a = !1;
      const c = e.map((s) => (Object.prototype.toString.call(s) === "[object Object]" && (a = Object.keys(s).length > 1), Xe(s, n, t)));
      return Xr(c, a, n.indent, t);
    }
    case "[object Object]": {
      const a = [];
      for (const c in e)
        a.push(`"${c}": ${Xe(e[c], n, t)}`);
      return Xr(
        a,
        // @ts-expect-error needs better types
        n.pretty && a.length > 1,
        // @ts-expect-error needs better types
        n.indent,
        t
      );
    }
    case "[object Boolean]":
      return e.toString();
    default:
      return e == null ? "nil" : `"${e.toString().replace(/"/g, '\\"')}"`;
  }
}, Ri = {
  info: {
    key: "urlsession",
    title: "URLSession",
    link: "https://developer.apple.com/documentation/foundation/urlsession",
    description: "Foundation's URLSession request",
    extname: ".swift"
  },
  convert: ({ allHeaders: e, postData: n, uriObj: t, queryObj: a, method: c }, s) => {
    var g;
    const r = {
      indent: "  ",
      pretty: !0,
      timeout: 10,
      ...s
    }, { push: o, blank: i, join: l } = new V({ indent: r.indent });
    o("import Foundation"), i();
    const f = n.text || n.jsonObj || n.params;
    if (f)
      switch (n.mimeType) {
        case "application/x-www-form-urlencoded":
          if ((g = n.params) != null && g.length) {
            const m = n.params.map((x) => `"${x.name}": "${x.value}"`);
            r.pretty ? (o("let parameters = ["), m.forEach((x) => o(`${x},`, 1)), o("]")) : o(`let parameters = [${m.join(", ")}]`), o('let joinedParameters = parameters.map { "\\($0.key)=\\($0.value)" }.joined(separator: "&")'), o("let postData = Data(joinedParameters.utf8)"), i();
          }
          break;
        case "application/json":
          n.jsonObj && (o(`${Zr("parameters", n.jsonObj, r)} as [String : Any?]`), i(), o("let postData = try JSONSerialization.data(withJSONObject: parameters, options: [])"), i());
          break;
        case "multipart/form-data":
          o(Zr("parameters", n.params, r)), i(), o(`let boundary = "${n.boundary}"`), i(), o('var body = ""'), o("for param in parameters {"), o('let paramName = param["name"]!', 1), o('body += "--\\(boundary)\\r\\n"', 1), o('body += "Content-Disposition:form-data; name=\\"\\(paramName)\\""', 1), o('if let filename = param["fileName"] {', 1), o('let contentType = param["contentType"]!', 2), o("let fileContent = try String(contentsOfFile: filename, encoding: .utf8)", 2), o('body += "; filename=\\"\\(filename)\\"\\r\\n"', 2), o('body += "Content-Type: \\(contentType)\\r\\n\\r\\n"', 2), o("body += fileContent", 2), o('} else if let paramValue = param["value"] {', 1), o('body += "\\r\\n\\r\\n\\(paramValue)"', 2), o("}", 1), o("}"), i(), o("let postData = Data(body.utf8)"), i();
          break;
        default:
          o(`let postData = Data("${n.text}".utf8)`), i();
      }
    o(`let url = URL(string: "${t.href}")!`);
    const y = a ? Object.entries(a) : [];
    return y.length < 1 ? o("var request = URLRequest(url: url)") : (o("var components = URLComponents(url: url, resolvingAgainstBaseURL: true)!"), o("let queryItems: [URLQueryItem] = ["), y.forEach((m) => {
      const x = m[0], v = m[1];
      switch (Object.prototype.toString.call(v)) {
        case "[object String]":
          o(`URLQueryItem(name: "${x}", value: "${v}"),`, 1);
          break;
        case "[object Array]":
          v.forEach((u) => {
            o(`URLQueryItem(name: "${x}", value: "${u}"),`, 1);
          });
          break;
      }
    }), o("]"), o("components.queryItems = components.queryItems.map { $0 + queryItems } ?? queryItems"), i(), o("var request = URLRequest(url: components.url!)")), o(`request.httpMethod = "${c}"`), o(`request.timeoutInterval = ${r.timeout}`), Object.keys(e).length && o(`request.allHTTPHeaderFields = ${Xe(e, r)}`), f && o("request.httpBody = postData"), i(), o("let (data, _) = try await URLSession.shared.data(for: request)"), o("print(String(decoding: data, as: UTF8.self))"), l();
  }
}, ki = {
  info: {
    key: "swift",
    title: "Swift",
    default: "urlsession"
  },
  clientsById: {
    urlsession: Ri
  }
}, Ni = {
  c: Oo,
  clojure: Co,
  csharp: Ao,
  go: Uo,
  http: Fo,
  java: zo,
  javascript: Do,
  json: Ko,
  kotlin: Xo,
  node: ai,
  objc: ii,
  ocaml: li,
  php: di,
  powershell: mi,
  python: bi,
  r: $i,
  ruby: Si,
  shell: Ei,
  swift: ki
}, Oe = {}, Ae = { exports: {} };
/*! https://mths.be/punycode v1.4.1 by @mathias */
var Ai = Ae.exports, en;
function Ii() {
  return en || (en = 1, function(e, n) {
    (function(t) {
      var a = n && !n.nodeType && n, c = e && !e.nodeType && e, s = typeof Ke == "object" && Ke;
      (s.global === s || s.window === s || s.self === s) && (t = s);
      var r, o = 2147483647, i = 36, l = 1, f = 26, y = 38, g = 700, m = 72, x = 128, v = "-", u = /^xn--/, h = /[^\x20-\x7E]/, d = /[\x2E\u3002\uFF0E\uFF61]/g, j = {
        overflow: "Overflow: input needs wider integers to process",
        "not-basic": "Illegal input >= 0x80 (not a basic code point)",
        "invalid-input": "Invalid input"
      }, T = i - l, p = Math.floor, O = String.fromCharCode, b;
      function q(R) {
        throw new RangeError(j[R]);
      }
      function A(R, k) {
        for (var F = R.length, C = []; F--; )
          C[F] = k(R[F]);
        return C;
      }
      function E(R, k) {
        var F = R.split("@"), C = "";
        F.length > 1 && (C = F[0] + "@", R = F[1]), R = R.replace(d, ".");
        var U = R.split("."), J = A(U, k).join(".");
        return C + J;
      }
      function L(R) {
        for (var k = [], F = 0, C = R.length, U, J; F < C; )
          U = R.charCodeAt(F++), U >= 55296 && U <= 56319 && F < C ? (J = R.charCodeAt(F++), (J & 64512) == 56320 ? k.push(((U & 1023) << 10) + (J & 1023) + 65536) : (k.push(U), F--)) : k.push(U);
        return k;
      }
      function P(R) {
        return A(R, function(k) {
          var F = "";
          return k > 65535 && (k -= 65536, F += O(k >>> 10 & 1023 | 55296), k = 56320 | k & 1023), F += O(k), F;
        }).join("");
      }
      function oe(R) {
        return R - 48 < 10 ? R - 22 : R - 65 < 26 ? R - 65 : R - 97 < 26 ? R - 97 : i;
      }
      function H(R, k) {
        return R + 22 + 75 * (R < 26) - ((k != 0) << 5);
      }
      function D(R, k, F) {
        var C = 0;
        for (R = F ? p(R / g) : R >> 1, R += p(R / k); R > T * f >> 1; C += i)
          R = p(R / T);
        return p(C + (T + 1) * R / (R + y));
      }
      function se(R) {
        var k = [], F = R.length, C, U = 0, J = x, z = m, K, ne, G, I, _, B, Q, W, Y;
        for (K = R.lastIndexOf(v), K < 0 && (K = 0), ne = 0; ne < K; ++ne)
          R.charCodeAt(ne) >= 128 && q("not-basic"), k.push(R.charCodeAt(ne));
        for (G = K > 0 ? K + 1 : 0; G < F; ) {
          for (I = U, _ = 1, B = i; G >= F && q("invalid-input"), Q = oe(R.charCodeAt(G++)), (Q >= i || Q > p((o - U) / _)) && q("overflow"), U += Q * _, W = B <= z ? l : B >= z + f ? f : B - z, !(Q < W); B += i)
            Y = i - W, _ > p(o / Y) && q("overflow"), _ *= Y;
          C = k.length + 1, z = D(U - I, C, I == 0), p(U / C) > o - J && q("overflow"), J += p(U / C), U %= C, k.splice(U++, 0, J);
        }
        return P(k);
      }
      function Z(R) {
        var k, F, C, U, J, z, K, ne, G, I, _, B = [], Q, W, Y, ae;
        for (R = L(R), Q = R.length, k = x, F = 0, J = m, z = 0; z < Q; ++z)
          _ = R[z], _ < 128 && B.push(O(_));
        for (C = U = B.length, U && B.push(v); C < Q; ) {
          for (K = o, z = 0; z < Q; ++z)
            _ = R[z], _ >= k && _ < K && (K = _);
          for (W = C + 1, K - k > p((o - F) / W) && q("overflow"), F += (K - k) * W, k = K, z = 0; z < Q; ++z)
            if (_ = R[z], _ < k && ++F > o && q("overflow"), _ == k) {
              for (ne = F, G = i; I = G <= J ? l : G >= J + f ? f : G - J, !(ne < I); G += i)
                ae = ne - I, Y = i - I, B.push(
                  O(H(I + ae % Y, 0))
                ), ne = p(ae / Y);
              B.push(O(H(ne, 0))), J = D(F, W, C == U), F = 0, ++C;
            }
          ++F, ++k;
        }
        return B.join("");
      }
      function N(R) {
        return E(R, function(k) {
          return u.test(k) ? se(k.slice(4).toLowerCase()) : k;
        });
      }
      function re(R) {
        return E(R, function(k) {
          return h.test(k) ? "xn--" + Z(k) : k;
        });
      }
      if (r = {
        /**
         * A string representing the current Punycode.js version number.
         * @memberOf punycode
         * @type String
         */
        version: "1.4.1",
        /**
         * An object of methods to convert from JavaScript's internal character
         * representation (UCS-2) to Unicode code points, and back.
         * @see <https://mathiasbynens.be/notes/javascript-encoding>
         * @memberOf punycode
         * @type Object
         */
        ucs2: {
          decode: L,
          encode: P
        },
        decode: se,
        encode: Z,
        toASCII: re,
        toUnicode: N
      }, a && c)
        if (e.exports == a)
          c.exports = r;
        else
          for (b in r)
            r.hasOwnProperty(b) && (a[b] = r[b]);
      else
        t.punycode = r;
    })(Ai);
  }(Ae, Ae.exports)), Ae.exports;
}
var yt, tn;
function ua() {
  return tn || (tn = 1, yt = Object), yt;
}
var mt, rn;
function Ui() {
  return rn || (rn = 1, mt = Error), mt;
}
var gt, nn;
function Li() {
  return nn || (nn = 1, gt = EvalError), gt;
}
var vt, an;
function Fi() {
  return an || (an = 1, vt = RangeError), vt;
}
var bt, on;
function _i() {
  return on || (on = 1, bt = ReferenceError), bt;
}
var xt, sn;
function fa() {
  return sn || (sn = 1, xt = SyntaxError), xt;
}
var $t, ln;
function Be() {
  return ln || (ln = 1, $t = TypeError), $t;
}
var wt, cn;
function Bi() {
  return cn || (cn = 1, wt = URIError), wt;
}
var St, un;
function Hi() {
  return un || (un = 1, St = Math.abs), St;
}
var jt, fn;
function Mi() {
  return fn || (fn = 1, jt = Math.floor), jt;
}
var Ot, pn;
function zi() {
  return pn || (pn = 1, Ot = Math.max), Ot;
}
var Tt, dn;
function Wi() {
  return dn || (dn = 1, Tt = Math.min), Tt;
}
var Pt, hn;
function Ji() {
  return hn || (hn = 1, Pt = Math.pow), Pt;
}
var qt, yn;
function Gi() {
  return yn || (yn = 1, qt = Math.round), qt;
}
var Ct, mn;
function Vi() {
  return mn || (mn = 1, Ct = Number.isNaN || function(n) {
    return n !== n;
  }), Ct;
}
var Et, gn;
function Di() {
  if (gn) return Et;
  gn = 1;
  var e = /* @__PURE__ */ Vi();
  return Et = function(t) {
    return e(t) || t === 0 ? t : t < 0 ? -1 : 1;
  }, Et;
}
var Rt, vn;
function Qi() {
  return vn || (vn = 1, Rt = Object.getOwnPropertyDescriptor), Rt;
}
var kt, bn;
function et() {
  if (bn) return kt;
  bn = 1;
  var e = /* @__PURE__ */ Qi();
  if (e)
    try {
      e([], "length");
    } catch {
      e = null;
    }
  return kt = e, kt;
}
var Nt, xn;
function tt() {
  if (xn) return Nt;
  xn = 1;
  var e = Object.defineProperty || !1;
  if (e)
    try {
      e({}, "a", { value: 1 });
    } catch {
      e = !1;
    }
  return Nt = e, Nt;
}
var At, $n;
function Ki() {
  return $n || ($n = 1, At = function() {
    if (typeof Symbol != "function" || typeof Object.getOwnPropertySymbols != "function")
      return !1;
    if (typeof Symbol.iterator == "symbol")
      return !0;
    var n = {}, t = Symbol("test"), a = Object(t);
    if (typeof t == "string" || Object.prototype.toString.call(t) !== "[object Symbol]" || Object.prototype.toString.call(a) !== "[object Symbol]")
      return !1;
    var c = 42;
    n[t] = c;
    for (var s in n)
      return !1;
    if (typeof Object.keys == "function" && Object.keys(n).length !== 0 || typeof Object.getOwnPropertyNames == "function" && Object.getOwnPropertyNames(n).length !== 0)
      return !1;
    var r = Object.getOwnPropertySymbols(n);
    if (r.length !== 1 || r[0] !== t || !Object.prototype.propertyIsEnumerable.call(n, t))
      return !1;
    if (typeof Object.getOwnPropertyDescriptor == "function") {
      var o = (
        /** @type {PropertyDescriptor} */
        Object.getOwnPropertyDescriptor(n, t)
      );
      if (o.value !== c || o.enumerable !== !0)
        return !1;
    }
    return !0;
  }), At;
}
var It, wn;
function Yi() {
  if (wn) return It;
  wn = 1;
  var e = typeof Symbol < "u" && Symbol, n = Ki();
  return It = function() {
    return typeof e != "function" || typeof Symbol != "function" || typeof e("foo") != "symbol" || typeof Symbol("bar") != "symbol" ? !1 : n();
  }, It;
}
var Ut, Sn;
function pa() {
  return Sn || (Sn = 1, Ut = typeof Reflect < "u" && Reflect.getPrototypeOf || null), Ut;
}
var Lt, jn;
function da() {
  if (jn) return Lt;
  jn = 1;
  var e = /* @__PURE__ */ ua();
  return Lt = e.getPrototypeOf || null, Lt;
}
var Ft, On;
function Xi() {
  if (On) return Ft;
  On = 1;
  var e = "Function.prototype.bind called on incompatible ", n = Object.prototype.toString, t = Math.max, a = "[object Function]", c = function(i, l) {
    for (var f = [], y = 0; y < i.length; y += 1)
      f[y] = i[y];
    for (var g = 0; g < l.length; g += 1)
      f[g + i.length] = l[g];
    return f;
  }, s = function(i, l) {
    for (var f = [], y = l, g = 0; y < i.length; y += 1, g += 1)
      f[g] = i[y];
    return f;
  }, r = function(o, i) {
    for (var l = "", f = 0; f < o.length; f += 1)
      l += o[f], f + 1 < o.length && (l += i);
    return l;
  };
  return Ft = function(i) {
    var l = this;
    if (typeof l != "function" || n.apply(l) !== a)
      throw new TypeError(e + l);
    for (var f = s(arguments, 1), y, g = function() {
      if (this instanceof y) {
        var h = l.apply(
          this,
          c(f, arguments)
        );
        return Object(h) === h ? h : this;
      }
      return l.apply(
        i,
        c(f, arguments)
      );
    }, m = t(0, l.length - f.length), x = [], v = 0; v < m; v++)
      x[v] = "$" + v;
    if (y = Function("binder", "return function (" + r(x, ",") + "){ return binder.apply(this,arguments); }")(g), l.prototype) {
      var u = function() {
      };
      u.prototype = l.prototype, y.prototype = new u(), u.prototype = null;
    }
    return y;
  }, Ft;
}
var _t, Tn;
function He() {
  if (Tn) return _t;
  Tn = 1;
  var e = Xi();
  return _t = Function.prototype.bind || e, _t;
}
var Bt, Pn;
function br() {
  return Pn || (Pn = 1, Bt = Function.prototype.call), Bt;
}
var Ht, qn;
function xr() {
  return qn || (qn = 1, Ht = Function.prototype.apply), Ht;
}
var Mt, Cn;
function Zi() {
  return Cn || (Cn = 1, Mt = typeof Reflect < "u" && Reflect && Reflect.apply), Mt;
}
var zt, En;
function ha() {
  if (En) return zt;
  En = 1;
  var e = He(), n = xr(), t = br(), a = Zi();
  return zt = a || e.call(t, n), zt;
}
var Wt, Rn;
function ya() {
  if (Rn) return Wt;
  Rn = 1;
  var e = He(), n = /* @__PURE__ */ Be(), t = br(), a = ha();
  return Wt = function(s) {
    if (s.length < 1 || typeof s[0] != "function")
      throw new n("a function is required");
    return a(e, t, s);
  }, Wt;
}
var Jt, kn;
function es() {
  if (kn) return Jt;
  kn = 1;
  var e = ya(), n = /* @__PURE__ */ et(), t;
  try {
    t = /** @type {{ __proto__?: typeof Array.prototype }} */
    [].__proto__ === Array.prototype;
  } catch (r) {
    if (!r || typeof r != "object" || !("code" in r) || r.code !== "ERR_PROTO_ACCESS")
      throw r;
  }
  var a = !!t && n && n(
    Object.prototype,
    /** @type {keyof typeof Object.prototype} */
    "__proto__"
  ), c = Object, s = c.getPrototypeOf;
  return Jt = a && typeof a.get == "function" ? e([a.get]) : typeof s == "function" ? (
    /** @type {import('./get')} */
    function(o) {
      return s(o == null ? o : c(o));
    }
  ) : !1, Jt;
}
var Gt, Nn;
function ts() {
  if (Nn) return Gt;
  Nn = 1;
  var e = pa(), n = da(), t = /* @__PURE__ */ es();
  return Gt = e ? function(c) {
    return e(c);
  } : n ? function(c) {
    if (!c || typeof c != "object" && typeof c != "function")
      throw new TypeError("getProto: not an object");
    return n(c);
  } : t ? function(c) {
    return t(c);
  } : null, Gt;
}
var Vt, An;
function rs() {
  if (An) return Vt;
  An = 1;
  var e = Function.prototype.call, n = Object.prototype.hasOwnProperty, t = He();
  return Vt = t.call(e, n), Vt;
}
var Dt, In;
function $r() {
  if (In) return Dt;
  In = 1;
  var e, n = /* @__PURE__ */ ua(), t = /* @__PURE__ */ Ui(), a = /* @__PURE__ */ Li(), c = /* @__PURE__ */ Fi(), s = /* @__PURE__ */ _i(), r = /* @__PURE__ */ fa(), o = /* @__PURE__ */ Be(), i = /* @__PURE__ */ Bi(), l = /* @__PURE__ */ Hi(), f = /* @__PURE__ */ Mi(), y = /* @__PURE__ */ zi(), g = /* @__PURE__ */ Wi(), m = /* @__PURE__ */ Ji(), x = /* @__PURE__ */ Gi(), v = /* @__PURE__ */ Di(), u = Function, h = function(G) {
    try {
      return u('"use strict"; return (' + G + ").constructor;")();
    } catch {
    }
  }, d = /* @__PURE__ */ et(), j = /* @__PURE__ */ tt(), T = function() {
    throw new o();
  }, p = d ? function() {
    try {
      return arguments.callee, T;
    } catch {
      try {
        return d(arguments, "callee").get;
      } catch {
        return T;
      }
    }
  }() : T, O = Yi()(), b = ts(), q = da(), A = pa(), E = xr(), L = br(), P = {}, oe = typeof Uint8Array > "u" || !b ? e : b(Uint8Array), H = {
    __proto__: null,
    "%AggregateError%": typeof AggregateError > "u" ? e : AggregateError,
    "%Array%": Array,
    "%ArrayBuffer%": typeof ArrayBuffer > "u" ? e : ArrayBuffer,
    "%ArrayIteratorPrototype%": O && b ? b([][Symbol.iterator]()) : e,
    "%AsyncFromSyncIteratorPrototype%": e,
    "%AsyncFunction%": P,
    "%AsyncGenerator%": P,
    "%AsyncGeneratorFunction%": P,
    "%AsyncIteratorPrototype%": P,
    "%Atomics%": typeof Atomics > "u" ? e : Atomics,
    "%BigInt%": typeof BigInt > "u" ? e : BigInt,
    "%BigInt64Array%": typeof BigInt64Array > "u" ? e : BigInt64Array,
    "%BigUint64Array%": typeof BigUint64Array > "u" ? e : BigUint64Array,
    "%Boolean%": Boolean,
    "%DataView%": typeof DataView > "u" ? e : DataView,
    "%Date%": Date,
    "%decodeURI%": decodeURI,
    "%decodeURIComponent%": decodeURIComponent,
    "%encodeURI%": encodeURI,
    "%encodeURIComponent%": encodeURIComponent,
    "%Error%": t,
    "%eval%": eval,
    // eslint-disable-line no-eval
    "%EvalError%": a,
    "%Float16Array%": typeof Float16Array > "u" ? e : Float16Array,
    "%Float32Array%": typeof Float32Array > "u" ? e : Float32Array,
    "%Float64Array%": typeof Float64Array > "u" ? e : Float64Array,
    "%FinalizationRegistry%": typeof FinalizationRegistry > "u" ? e : FinalizationRegistry,
    "%Function%": u,
    "%GeneratorFunction%": P,
    "%Int8Array%": typeof Int8Array > "u" ? e : Int8Array,
    "%Int16Array%": typeof Int16Array > "u" ? e : Int16Array,
    "%Int32Array%": typeof Int32Array > "u" ? e : Int32Array,
    "%isFinite%": isFinite,
    "%isNaN%": isNaN,
    "%IteratorPrototype%": O && b ? b(b([][Symbol.iterator]())) : e,
    "%JSON%": typeof JSON == "object" ? JSON : e,
    "%Map%": typeof Map > "u" ? e : Map,
    "%MapIteratorPrototype%": typeof Map > "u" || !O || !b ? e : b((/* @__PURE__ */ new Map())[Symbol.iterator]()),
    "%Math%": Math,
    "%Number%": Number,
    "%Object%": n,
    "%Object.getOwnPropertyDescriptor%": d,
    "%parseFloat%": parseFloat,
    "%parseInt%": parseInt,
    "%Promise%": typeof Promise > "u" ? e : Promise,
    "%Proxy%": typeof Proxy > "u" ? e : Proxy,
    "%RangeError%": c,
    "%ReferenceError%": s,
    "%Reflect%": typeof Reflect > "u" ? e : Reflect,
    "%RegExp%": RegExp,
    "%Set%": typeof Set > "u" ? e : Set,
    "%SetIteratorPrototype%": typeof Set > "u" || !O || !b ? e : b((/* @__PURE__ */ new Set())[Symbol.iterator]()),
    "%SharedArrayBuffer%": typeof SharedArrayBuffer > "u" ? e : SharedArrayBuffer,
    "%String%": String,
    "%StringIteratorPrototype%": O && b ? b(""[Symbol.iterator]()) : e,
    "%Symbol%": O ? Symbol : e,
    "%SyntaxError%": r,
    "%ThrowTypeError%": p,
    "%TypedArray%": oe,
    "%TypeError%": o,
    "%Uint8Array%": typeof Uint8Array > "u" ? e : Uint8Array,
    "%Uint8ClampedArray%": typeof Uint8ClampedArray > "u" ? e : Uint8ClampedArray,
    "%Uint16Array%": typeof Uint16Array > "u" ? e : Uint16Array,
    "%Uint32Array%": typeof Uint32Array > "u" ? e : Uint32Array,
    "%URIError%": i,
    "%WeakMap%": typeof WeakMap > "u" ? e : WeakMap,
    "%WeakRef%": typeof WeakRef > "u" ? e : WeakRef,
    "%WeakSet%": typeof WeakSet > "u" ? e : WeakSet,
    "%Function.prototype.call%": L,
    "%Function.prototype.apply%": E,
    "%Object.defineProperty%": j,
    "%Object.getPrototypeOf%": q,
    "%Math.abs%": l,
    "%Math.floor%": f,
    "%Math.max%": y,
    "%Math.min%": g,
    "%Math.pow%": m,
    "%Math.round%": x,
    "%Math.sign%": v,
    "%Reflect.getPrototypeOf%": A
  };
  if (b)
    try {
      null.error;
    } catch (G) {
      var D = b(b(G));
      H["%Error.prototype%"] = D;
    }
  var se = function G(I) {
    var _;
    if (I === "%AsyncFunction%")
      _ = h("async function () {}");
    else if (I === "%GeneratorFunction%")
      _ = h("function* () {}");
    else if (I === "%AsyncGeneratorFunction%")
      _ = h("async function* () {}");
    else if (I === "%AsyncGenerator%") {
      var B = G("%AsyncGeneratorFunction%");
      B && (_ = B.prototype);
    } else if (I === "%AsyncIteratorPrototype%") {
      var Q = G("%AsyncGenerator%");
      Q && b && (_ = b(Q.prototype));
    }
    return H[I] = _, _;
  }, Z = {
    __proto__: null,
    "%ArrayBufferPrototype%": ["ArrayBuffer", "prototype"],
    "%ArrayPrototype%": ["Array", "prototype"],
    "%ArrayProto_entries%": ["Array", "prototype", "entries"],
    "%ArrayProto_forEach%": ["Array", "prototype", "forEach"],
    "%ArrayProto_keys%": ["Array", "prototype", "keys"],
    "%ArrayProto_values%": ["Array", "prototype", "values"],
    "%AsyncFunctionPrototype%": ["AsyncFunction", "prototype"],
    "%AsyncGenerator%": ["AsyncGeneratorFunction", "prototype"],
    "%AsyncGeneratorPrototype%": ["AsyncGeneratorFunction", "prototype", "prototype"],
    "%BooleanPrototype%": ["Boolean", "prototype"],
    "%DataViewPrototype%": ["DataView", "prototype"],
    "%DatePrototype%": ["Date", "prototype"],
    "%ErrorPrototype%": ["Error", "prototype"],
    "%EvalErrorPrototype%": ["EvalError", "prototype"],
    "%Float32ArrayPrototype%": ["Float32Array", "prototype"],
    "%Float64ArrayPrototype%": ["Float64Array", "prototype"],
    "%FunctionPrototype%": ["Function", "prototype"],
    "%Generator%": ["GeneratorFunction", "prototype"],
    "%GeneratorPrototype%": ["GeneratorFunction", "prototype", "prototype"],
    "%Int8ArrayPrototype%": ["Int8Array", "prototype"],
    "%Int16ArrayPrototype%": ["Int16Array", "prototype"],
    "%Int32ArrayPrototype%": ["Int32Array", "prototype"],
    "%JSONParse%": ["JSON", "parse"],
    "%JSONStringify%": ["JSON", "stringify"],
    "%MapPrototype%": ["Map", "prototype"],
    "%NumberPrototype%": ["Number", "prototype"],
    "%ObjectPrototype%": ["Object", "prototype"],
    "%ObjProto_toString%": ["Object", "prototype", "toString"],
    "%ObjProto_valueOf%": ["Object", "prototype", "valueOf"],
    "%PromisePrototype%": ["Promise", "prototype"],
    "%PromiseProto_then%": ["Promise", "prototype", "then"],
    "%Promise_all%": ["Promise", "all"],
    "%Promise_reject%": ["Promise", "reject"],
    "%Promise_resolve%": ["Promise", "resolve"],
    "%RangeErrorPrototype%": ["RangeError", "prototype"],
    "%ReferenceErrorPrototype%": ["ReferenceError", "prototype"],
    "%RegExpPrototype%": ["RegExp", "prototype"],
    "%SetPrototype%": ["Set", "prototype"],
    "%SharedArrayBufferPrototype%": ["SharedArrayBuffer", "prototype"],
    "%StringPrototype%": ["String", "prototype"],
    "%SymbolPrototype%": ["Symbol", "prototype"],
    "%SyntaxErrorPrototype%": ["SyntaxError", "prototype"],
    "%TypedArrayPrototype%": ["TypedArray", "prototype"],
    "%TypeErrorPrototype%": ["TypeError", "prototype"],
    "%Uint8ArrayPrototype%": ["Uint8Array", "prototype"],
    "%Uint8ClampedArrayPrototype%": ["Uint8ClampedArray", "prototype"],
    "%Uint16ArrayPrototype%": ["Uint16Array", "prototype"],
    "%Uint32ArrayPrototype%": ["Uint32Array", "prototype"],
    "%URIErrorPrototype%": ["URIError", "prototype"],
    "%WeakMapPrototype%": ["WeakMap", "prototype"],
    "%WeakSetPrototype%": ["WeakSet", "prototype"]
  }, N = He(), re = /* @__PURE__ */ rs(), R = N.call(L, Array.prototype.concat), k = N.call(E, Array.prototype.splice), F = N.call(L, String.prototype.replace), C = N.call(L, String.prototype.slice), U = N.call(L, RegExp.prototype.exec), J = /[^%.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|%$))/g, z = /\\(\\)?/g, K = function(I) {
    var _ = C(I, 0, 1), B = C(I, -1);
    if (_ === "%" && B !== "%")
      throw new r("invalid intrinsic syntax, expected closing `%`");
    if (B === "%" && _ !== "%")
      throw new r("invalid intrinsic syntax, expected opening `%`");
    var Q = [];
    return F(I, J, function(W, Y, ae, he) {
      Q[Q.length] = ae ? F(he, z, "$1") : Y || W;
    }), Q;
  }, ne = function(I, _) {
    var B = I, Q;
    if (re(Z, B) && (Q = Z[B], B = "%" + Q[0] + "%"), re(H, B)) {
      var W = H[B];
      if (W === P && (W = se(B)), typeof W > "u" && !_)
        throw new o("intrinsic " + I + " exists, but is not available. Please file an issue!");
      return {
        alias: Q,
        name: B,
        value: W
      };
    }
    throw new r("intrinsic " + I + " does not exist!");
  };
  return Dt = function(I, _) {
    if (typeof I != "string" || I.length === 0)
      throw new o("intrinsic name must be a non-empty string");
    if (arguments.length > 1 && typeof _ != "boolean")
      throw new o('"allowMissing" argument must be a boolean');
    if (U(/^%?[^%]*%?$/, I) === null)
      throw new r("`%` may not be present anywhere but at the beginning and end of the intrinsic name");
    var B = K(I), Q = B.length > 0 ? B[0] : "", W = ne("%" + Q + "%", _), Y = W.name, ae = W.value, he = !1, $e = W.alias;
    $e && (Q = $e[0], k(B, R([0, 1], $e)));
    for (var Te = 1, we = !0; Te < B.length; Te += 1) {
      var ye = B[Te], Pe = C(ye, 0, 1), qe = C(ye, -1);
      if ((Pe === '"' || Pe === "'" || Pe === "`" || qe === '"' || qe === "'" || qe === "`") && Pe !== qe)
        throw new r("property names with quotes must have matching quotes");
      if ((ye === "constructor" || !we) && (he = !0), Q += "." + ye, Y = "%" + Q + "%", re(H, Y))
        ae = H[Y];
      else if (ae != null) {
        if (!(ye in ae)) {
          if (!_)
            throw new o("base intrinsic for " + I + " exists, but the property is not available.");
          return;
        }
        if (d && Te + 1 >= B.length) {
          var Se = d(ae, ye);
          we = !!Se, we && "get" in Se && !("originalValue" in Se.get) ? ae = Se.get : ae = ae[ye];
        } else
          we = re(ae, ye), ae = ae[ye];
        we && !he && (H[Y] = ae);
      }
    }
    return ae;
  }, Dt;
}
var Qt = { exports: {} }, Kt, Un;
function ns() {
  if (Un) return Kt;
  Un = 1;
  var e = /* @__PURE__ */ tt(), n = /* @__PURE__ */ fa(), t = /* @__PURE__ */ Be(), a = /* @__PURE__ */ et();
  return Kt = function(s, r, o) {
    if (!s || typeof s != "object" && typeof s != "function")
      throw new t("`obj` must be an object or a function`");
    if (typeof r != "string" && typeof r != "symbol")
      throw new t("`property` must be a string or a symbol`");
    if (arguments.length > 3 && typeof arguments[3] != "boolean" && arguments[3] !== null)
      throw new t("`nonEnumerable`, if provided, must be a boolean or null");
    if (arguments.length > 4 && typeof arguments[4] != "boolean" && arguments[4] !== null)
      throw new t("`nonWritable`, if provided, must be a boolean or null");
    if (arguments.length > 5 && typeof arguments[5] != "boolean" && arguments[5] !== null)
      throw new t("`nonConfigurable`, if provided, must be a boolean or null");
    if (arguments.length > 6 && typeof arguments[6] != "boolean")
      throw new t("`loose`, if provided, must be a boolean");
    var i = arguments.length > 3 ? arguments[3] : null, l = arguments.length > 4 ? arguments[4] : null, f = arguments.length > 5 ? arguments[5] : null, y = arguments.length > 6 ? arguments[6] : !1, g = !!a && a(s, r);
    if (e)
      e(s, r, {
        configurable: f === null && g ? g.configurable : !f,
        enumerable: i === null && g ? g.enumerable : !i,
        value: o,
        writable: l === null && g ? g.writable : !l
      });
    else if (y || !i && !l && !f)
      s[r] = o;
    else
      throw new n("This environment does not support defining a property as non-configurable, non-writable, or non-enumerable.");
  }, Kt;
}
var Yt, Ln;
function as() {
  if (Ln) return Yt;
  Ln = 1;
  var e = /* @__PURE__ */ tt(), n = function() {
    return !!e;
  };
  return n.hasArrayLengthDefineBug = function() {
    if (!e)
      return null;
    try {
      return e([], "length", { value: 1 }).length !== 1;
    } catch {
      return !0;
    }
  }, Yt = n, Yt;
}
var Xt, Fn;
function os() {
  if (Fn) return Xt;
  Fn = 1;
  var e = /* @__PURE__ */ $r(), n = /* @__PURE__ */ ns(), t = /* @__PURE__ */ as()(), a = /* @__PURE__ */ et(), c = /* @__PURE__ */ Be(), s = e("%Math.floor%");
  return Xt = function(o, i) {
    if (typeof o != "function")
      throw new c("`fn` is not a function");
    if (typeof i != "number" || i < 0 || i > 4294967295 || s(i) !== i)
      throw new c("`length` must be a positive 32-bit integer");
    var l = arguments.length > 2 && !!arguments[2], f = !0, y = !0;
    if ("length" in o && a) {
      var g = a(o, "length");
      g && !g.configurable && (f = !1), g && !g.writable && (y = !1);
    }
    return (f || y || !l) && (t ? n(
      /** @type {Parameters<define>[0]} */
      o,
      "length",
      i,
      !0,
      !0
    ) : n(
      /** @type {Parameters<define>[0]} */
      o,
      "length",
      i
    )), o;
  }, Xt;
}
var Zt, _n;
function is() {
  if (_n) return Zt;
  _n = 1;
  var e = He(), n = xr(), t = ha();
  return Zt = function() {
    return t(e, n, arguments);
  }, Zt;
}
var Bn;
function ss() {
  return Bn || (Bn = 1, function(e) {
    var n = /* @__PURE__ */ os(), t = /* @__PURE__ */ tt(), a = ya(), c = is();
    e.exports = function(r) {
      var o = a(arguments), i = r.length - (arguments.length - 1);
      return n(
        o,
        1 + (i > 0 ? i : 0),
        !0
      );
    }, t ? t(e.exports, "apply", { value: c }) : e.exports.apply = c;
  }(Qt)), Qt.exports;
}
var er, Hn;
function ls() {
  if (Hn) return er;
  Hn = 1;
  var e = /* @__PURE__ */ $r(), n = ss(), t = n(e("String.prototype.indexOf"));
  return er = function(c, s) {
    var r = e(c, !!s);
    return typeof r == "function" && t(c, ".prototype.") > -1 ? n(r) : r;
  }, er;
}
const cs = {}, us = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: cs
}, Symbol.toStringTag, { value: "Module" })), fs = /* @__PURE__ */ so(us);
var tr, Mn;
function ps() {
  if (Mn) return tr;
  Mn = 1;
  var e = typeof Map == "function" && Map.prototype, n = Object.getOwnPropertyDescriptor && e ? Object.getOwnPropertyDescriptor(Map.prototype, "size") : null, t = e && n && typeof n.get == "function" ? n.get : null, a = e && Map.prototype.forEach, c = typeof Set == "function" && Set.prototype, s = Object.getOwnPropertyDescriptor && c ? Object.getOwnPropertyDescriptor(Set.prototype, "size") : null, r = c && s && typeof s.get == "function" ? s.get : null, o = c && Set.prototype.forEach, i = typeof WeakMap == "function" && WeakMap.prototype, l = i ? WeakMap.prototype.has : null, f = typeof WeakSet == "function" && WeakSet.prototype, y = f ? WeakSet.prototype.has : null, g = typeof WeakRef == "function" && WeakRef.prototype, m = g ? WeakRef.prototype.deref : null, x = Boolean.prototype.valueOf, v = Object.prototype.toString, u = Function.prototype.toString, h = String.prototype.match, d = String.prototype.slice, j = String.prototype.replace, T = String.prototype.toUpperCase, p = String.prototype.toLowerCase, O = RegExp.prototype.test, b = Array.prototype.concat, q = Array.prototype.join, A = Array.prototype.slice, E = Math.floor, L = typeof BigInt == "function" ? BigInt.prototype.valueOf : null, P = Object.getOwnPropertySymbols, oe = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? Symbol.prototype.toString : null, H = typeof Symbol == "function" && typeof Symbol.iterator == "object", D = typeof Symbol == "function" && Symbol.toStringTag && (typeof Symbol.toStringTag === H || !0) ? Symbol.toStringTag : null, se = Object.prototype.propertyIsEnumerable, Z = (typeof Reflect == "function" ? Reflect.getPrototypeOf : Object.getPrototypeOf) || ([].__proto__ === Array.prototype ? function(w) {
    return w.__proto__;
  } : null);
  function N(w, S) {
    if (w === 1 / 0 || w === -1 / 0 || w !== w || w && w > -1e3 && w < 1e3 || O.call(/e/, S))
      return S;
    var X = /[0-9](?=(?:[0-9]{3})+(?![0-9]))/g;
    if (typeof w == "number") {
      var ee = w < 0 ? -E(-w) : E(w);
      if (ee !== w) {
        var ie = String(ee), M = d.call(S, ie.length + 1);
        return j.call(ie, X, "$&_") + "." + j.call(j.call(M, /([0-9]{3})/g, "$&_"), /_$/, "");
      }
    }
    return j.call(S, X, "$&_");
  }
  var re = fs, R = re.custom, k = _(R) ? R : null;
  tr = function w(S, X, ee, ie) {
    var M = X || {};
    if (W(M, "quoteStyle") && M.quoteStyle !== "single" && M.quoteStyle !== "double")
      throw new TypeError('option "quoteStyle" must be "single" or "double"');
    if (W(M, "maxStringLength") && (typeof M.maxStringLength == "number" ? M.maxStringLength < 0 && M.maxStringLength !== 1 / 0 : M.maxStringLength !== null))
      throw new TypeError('option "maxStringLength", if provided, must be a positive integer, Infinity, or `null`');
    var ve = W(M, "customInspect") ? M.customInspect : !0;
    if (typeof ve != "boolean" && ve !== "symbol")
      throw new TypeError("option \"customInspect\", if provided, must be `true`, `false`, or `'symbol'`");
    if (W(M, "indent") && M.indent !== null && M.indent !== "	" && !(parseInt(M.indent, 10) === M.indent && M.indent > 0))
      throw new TypeError('option "indent" must be "\\t", an integer > 0, or `null`');
    if (W(M, "numericSeparator") && typeof M.numericSeparator != "boolean")
      throw new TypeError('option "numericSeparator", if provided, must be `true` or `false`');
    var je = M.numericSeparator;
    if (typeof S > "u")
      return "undefined";
    if (S === null)
      return "null";
    if (typeof S == "boolean")
      return S ? "true" : "false";
    if (typeof S == "string")
      return Se(S, M);
    if (typeof S == "number") {
      if (S === 0)
        return 1 / 0 / S > 0 ? "0" : "-0";
      var de = String(S);
      return je ? N(S, de) : de;
    }
    if (typeof S == "bigint") {
      var be = String(S) + "n";
      return je ? N(S, be) : be;
    }
    var at = typeof M.depth > "u" ? 5 : M.depth;
    if (typeof ee > "u" && (ee = 0), ee >= at && at > 0 && typeof S == "object")
      return U(S) ? "[Array]" : "[Object]";
    var Ce = ja(M, ee);
    if (typeof ie > "u")
      ie = [];
    else if (he(ie, S) >= 0)
      return "[Circular]";
    function me(Ee, We, Ta) {
      if (We && (ie = A.call(ie), ie.push(We)), Ta) {
        var Nr = {
          depth: M.depth
        };
        return W(M, "quoteStyle") && (Nr.quoteStyle = M.quoteStyle), w(Ee, Nr, ee + 1, ie);
      }
      return w(Ee, M, ee + 1, ie);
    }
    if (typeof S == "function" && !z(S)) {
      var Tr = ae(S), Pr = Me(S, me);
      return "[Function" + (Tr ? ": " + Tr : " (anonymous)") + "]" + (Pr.length > 0 ? " { " + q.call(Pr, ", ") + " }" : "");
    }
    if (_(S)) {
      var qr = H ? j.call(String(S), /^(Symbol\(.*\))_[^)]*$/, "$1") : oe.call(S);
      return typeof S == "object" && !H ? ke(qr) : qr;
    }
    if (qe(S)) {
      for (var Ne = "<" + p.call(String(S.nodeName)), ot = S.attributes || [], ze = 0; ze < ot.length; ze++)
        Ne += " " + ot[ze].name + "=" + F(C(ot[ze].value), "double", M);
      return Ne += ">", S.childNodes && S.childNodes.length && (Ne += "..."), Ne += "</" + p.call(String(S.nodeName)) + ">", Ne;
    }
    if (U(S)) {
      if (S.length === 0)
        return "[]";
      var it = Me(S, me);
      return Ce && !Sa(it) ? "[" + nt(it, Ce) + "]" : "[ " + q.call(it, ", ") + " ]";
    }
    if (K(S)) {
      var st = Me(S, me);
      return !("cause" in Error.prototype) && "cause" in S && !se.call(S, "cause") ? "{ [" + String(S) + "] " + q.call(b.call("[cause]: " + me(S.cause), st), ", ") + " }" : st.length === 0 ? "[" + String(S) + "]" : "{ [" + String(S) + "] " + q.call(st, ", ") + " }";
    }
    if (typeof S == "object" && ve) {
      if (k && typeof S[k] == "function" && re)
        return re(S, { depth: at - ee });
      if (ve !== "symbol" && typeof S.inspect == "function")
        return S.inspect();
    }
    if ($e(S)) {
      var Cr = [];
      return a && a.call(S, function(Ee, We) {
        Cr.push(me(We, S, !0) + " => " + me(Ee, S));
      }), Or("Map", t.call(S), Cr, Ce);
    }
    if (ye(S)) {
      var Er = [];
      return o && o.call(S, function(Ee) {
        Er.push(me(Ee, S));
      }), Or("Set", r.call(S), Er, Ce);
    }
    if (Te(S))
      return rt("WeakMap");
    if (Pe(S))
      return rt("WeakSet");
    if (we(S))
      return rt("WeakRef");
    if (G(S))
      return ke(me(Number(S)));
    if (B(S))
      return ke(me(L.call(S)));
    if (I(S))
      return ke(x.call(S));
    if (ne(S))
      return ke(me(String(S)));
    if (typeof window < "u" && S === window)
      return "{ [object Window] }";
    if (typeof globalThis < "u" && S === globalThis || typeof Ke < "u" && S === Ke)
      return "{ [object globalThis] }";
    if (!J(S) && !z(S)) {
      var lt = Me(S, me), Rr = Z ? Z(S) === Object.prototype : S instanceof Object || S.constructor === Object, ct = S instanceof Object ? "" : "null prototype", kr = !Rr && D && Object(S) === S && D in S ? d.call(Y(S), 8, -1) : ct ? "Object" : "", Oa = Rr || typeof S.constructor != "function" ? "" : S.constructor.name ? S.constructor.name + " " : "", ut = Oa + (kr || ct ? "[" + q.call(b.call([], kr || [], ct || []), ": ") + "] " : "");
      return lt.length === 0 ? ut + "{}" : Ce ? ut + "{" + nt(lt, Ce) + "}" : ut + "{ " + q.call(lt, ", ") + " }";
    }
    return String(S);
  };
  function F(w, S, X) {
    var ee = (X.quoteStyle || S) === "double" ? '"' : "'";
    return ee + w + ee;
  }
  function C(w) {
    return j.call(String(w), /"/g, "&quot;");
  }
  function U(w) {
    return Y(w) === "[object Array]" && (!D || !(typeof w == "object" && D in w));
  }
  function J(w) {
    return Y(w) === "[object Date]" && (!D || !(typeof w == "object" && D in w));
  }
  function z(w) {
    return Y(w) === "[object RegExp]" && (!D || !(typeof w == "object" && D in w));
  }
  function K(w) {
    return Y(w) === "[object Error]" && (!D || !(typeof w == "object" && D in w));
  }
  function ne(w) {
    return Y(w) === "[object String]" && (!D || !(typeof w == "object" && D in w));
  }
  function G(w) {
    return Y(w) === "[object Number]" && (!D || !(typeof w == "object" && D in w));
  }
  function I(w) {
    return Y(w) === "[object Boolean]" && (!D || !(typeof w == "object" && D in w));
  }
  function _(w) {
    if (H)
      return w && typeof w == "object" && w instanceof Symbol;
    if (typeof w == "symbol")
      return !0;
    if (!w || typeof w != "object" || !oe)
      return !1;
    try {
      return oe.call(w), !0;
    } catch {
    }
    return !1;
  }
  function B(w) {
    if (!w || typeof w != "object" || !L)
      return !1;
    try {
      return L.call(w), !0;
    } catch {
    }
    return !1;
  }
  var Q = Object.prototype.hasOwnProperty || function(w) {
    return w in this;
  };
  function W(w, S) {
    return Q.call(w, S);
  }
  function Y(w) {
    return v.call(w);
  }
  function ae(w) {
    if (w.name)
      return w.name;
    var S = h.call(u.call(w), /^function\s*([\w$]+)/);
    return S ? S[1] : null;
  }
  function he(w, S) {
    if (w.indexOf)
      return w.indexOf(S);
    for (var X = 0, ee = w.length; X < ee; X++)
      if (w[X] === S)
        return X;
    return -1;
  }
  function $e(w) {
    if (!t || !w || typeof w != "object")
      return !1;
    try {
      t.call(w);
      try {
        r.call(w);
      } catch {
        return !0;
      }
      return w instanceof Map;
    } catch {
    }
    return !1;
  }
  function Te(w) {
    if (!l || !w || typeof w != "object")
      return !1;
    try {
      l.call(w, l);
      try {
        y.call(w, y);
      } catch {
        return !0;
      }
      return w instanceof WeakMap;
    } catch {
    }
    return !1;
  }
  function we(w) {
    if (!m || !w || typeof w != "object")
      return !1;
    try {
      return m.call(w), !0;
    } catch {
    }
    return !1;
  }
  function ye(w) {
    if (!r || !w || typeof w != "object")
      return !1;
    try {
      r.call(w);
      try {
        t.call(w);
      } catch {
        return !0;
      }
      return w instanceof Set;
    } catch {
    }
    return !1;
  }
  function Pe(w) {
    if (!y || !w || typeof w != "object")
      return !1;
    try {
      y.call(w, y);
      try {
        l.call(w, l);
      } catch {
        return !0;
      }
      return w instanceof WeakSet;
    } catch {
    }
    return !1;
  }
  function qe(w) {
    return !w || typeof w != "object" ? !1 : typeof HTMLElement < "u" && w instanceof HTMLElement ? !0 : typeof w.nodeName == "string" && typeof w.getAttribute == "function";
  }
  function Se(w, S) {
    if (w.length > S.maxStringLength) {
      var X = w.length - S.maxStringLength, ee = "... " + X + " more character" + (X > 1 ? "s" : "");
      return Se(d.call(w, 0, S.maxStringLength), S) + ee;
    }
    var ie = j.call(j.call(w, /(['\\])/g, "\\$1"), /[\x00-\x1f]/g, wa);
    return F(ie, "single", S);
  }
  function wa(w) {
    var S = w.charCodeAt(0), X = {
      8: "b",
      9: "t",
      10: "n",
      12: "f",
      13: "r"
    }[S];
    return X ? "\\" + X : "\\x" + (S < 16 ? "0" : "") + T.call(S.toString(16));
  }
  function ke(w) {
    return "Object(" + w + ")";
  }
  function rt(w) {
    return w + " { ? }";
  }
  function Or(w, S, X, ee) {
    var ie = ee ? nt(X, ee) : q.call(X, ", ");
    return w + " (" + S + ") {" + ie + "}";
  }
  function Sa(w) {
    for (var S = 0; S < w.length; S++)
      if (he(w[S], `
`) >= 0)
        return !1;
    return !0;
  }
  function ja(w, S) {
    var X;
    if (w.indent === "	")
      X = "	";
    else if (typeof w.indent == "number" && w.indent > 0)
      X = q.call(Array(w.indent + 1), " ");
    else
      return null;
    return {
      base: X,
      prev: q.call(Array(S + 1), X)
    };
  }
  function nt(w, S) {
    if (w.length === 0)
      return "";
    var X = `
` + S.prev + S.base;
    return X + q.call(w, "," + X) + `
` + S.prev;
  }
  function Me(w, S) {
    var X = U(w), ee = [];
    if (X) {
      ee.length = w.length;
      for (var ie = 0; ie < w.length; ie++)
        ee[ie] = W(w, ie) ? S(w[ie], w) : "";
    }
    var M = typeof P == "function" ? P(w) : [], ve;
    if (H) {
      ve = {};
      for (var je = 0; je < M.length; je++)
        ve["$" + M[je]] = M[je];
    }
    for (var de in w)
      W(w, de) && (X && String(Number(de)) === de && de < w.length || H && ve["$" + de] instanceof Symbol || (O.call(/[^\w$]/, de) ? ee.push(S(de, w) + ": " + S(w[de], w)) : ee.push(de + ": " + S(w[de], w))));
    if (typeof P == "function")
      for (var be = 0; be < M.length; be++)
        se.call(w, M[be]) && ee.push("[" + S(M[be]) + "]: " + S(w[M[be]], w));
    return ee;
  }
  return tr;
}
var rr, zn;
function ds() {
  if (zn) return rr;
  zn = 1;
  var e = /* @__PURE__ */ $r(), n = ls(), t = /* @__PURE__ */ ps(), a = /* @__PURE__ */ Be(), c = e("%WeakMap%", !0), s = e("%Map%", !0), r = n("WeakMap.prototype.get", !0), o = n("WeakMap.prototype.set", !0), i = n("WeakMap.prototype.has", !0), l = n("Map.prototype.get", !0), f = n("Map.prototype.set", !0), y = n("Map.prototype.has", !0), g = function(u, h) {
    for (var d = u, j; (j = d.next) !== null; d = j)
      if (j.key === h)
        return d.next = j.next, j.next = /** @type {NonNullable<typeof list.next>} */
        u.next, u.next = j, j;
  }, m = function(u, h) {
    var d = g(u, h);
    return d && d.value;
  }, x = function(u, h, d) {
    var j = g(u, h);
    j ? j.value = d : u.next = /** @type {import('.').ListNode<typeof value>} */
    {
      // eslint-disable-line no-param-reassign, no-extra-parens
      key: h,
      next: u.next,
      value: d
    };
  }, v = function(u, h) {
    return !!g(u, h);
  };
  return rr = function() {
    var h, d, j, T = {
      assert: function(p) {
        if (!T.has(p))
          throw new a("Side channel does not contain " + t(p));
      },
      get: function(p) {
        if (c && p && (typeof p == "object" || typeof p == "function")) {
          if (h)
            return r(h, p);
        } else if (s) {
          if (d)
            return l(d, p);
        } else if (j)
          return m(j, p);
      },
      has: function(p) {
        if (c && p && (typeof p == "object" || typeof p == "function")) {
          if (h)
            return i(h, p);
        } else if (s) {
          if (d)
            return y(d, p);
        } else if (j)
          return v(j, p);
        return !1;
      },
      set: function(p, O) {
        c && p && (typeof p == "object" || typeof p == "function") ? (h || (h = new c()), o(h, p, O)) : s ? (d || (d = new s()), f(d, p, O)) : (j || (j = { key: {}, next: null }), x(j, p, O));
      }
    };
    return T;
  }, rr;
}
var nr, Wn;
function wr() {
  if (Wn) return nr;
  Wn = 1;
  var e = String.prototype.replace, n = /%20/g, t = {
    RFC1738: "RFC1738",
    RFC3986: "RFC3986"
  };
  return nr = {
    default: t.RFC3986,
    formatters: {
      RFC1738: function(a) {
        return e.call(a, n, "+");
      },
      RFC3986: function(a) {
        return String(a);
      }
    },
    RFC1738: t.RFC1738,
    RFC3986: t.RFC3986
  }, nr;
}
var ar, Jn;
function ma() {
  if (Jn) return ar;
  Jn = 1;
  var e = /* @__PURE__ */ wr(), n = Object.prototype.hasOwnProperty, t = Array.isArray, a = function() {
    for (var u = [], h = 0; h < 256; ++h)
      u.push("%" + ((h < 16 ? "0" : "") + h.toString(16)).toUpperCase());
    return u;
  }(), c = function(h) {
    for (; h.length > 1; ) {
      var d = h.pop(), j = d.obj[d.prop];
      if (t(j)) {
        for (var T = [], p = 0; p < j.length; ++p)
          typeof j[p] < "u" && T.push(j[p]);
        d.obj[d.prop] = T;
      }
    }
  }, s = function(h, d) {
    for (var j = d && d.plainObjects ? /* @__PURE__ */ Object.create(null) : {}, T = 0; T < h.length; ++T)
      typeof h[T] < "u" && (j[T] = h[T]);
    return j;
  }, r = function u(h, d, j) {
    if (!d)
      return h;
    if (typeof d != "object") {
      if (t(h))
        h.push(d);
      else if (h && typeof h == "object")
        (j && (j.plainObjects || j.allowPrototypes) || !n.call(Object.prototype, d)) && (h[d] = !0);
      else
        return [h, d];
      return h;
    }
    if (!h || typeof h != "object")
      return [h].concat(d);
    var T = h;
    return t(h) && !t(d) && (T = s(h, j)), t(h) && t(d) ? (d.forEach(function(p, O) {
      if (n.call(h, O)) {
        var b = h[O];
        b && typeof b == "object" && p && typeof p == "object" ? h[O] = u(b, p, j) : h.push(p);
      } else
        h[O] = p;
    }), h) : Object.keys(d).reduce(function(p, O) {
      var b = d[O];
      return n.call(p, O) ? p[O] = u(p[O], b, j) : p[O] = b, p;
    }, T);
  }, o = function(h, d) {
    return Object.keys(d).reduce(function(j, T) {
      return j[T] = d[T], j;
    }, h);
  }, i = function(u, h, d) {
    var j = u.replace(/\+/g, " ");
    if (d === "iso-8859-1")
      return j.replace(/%[0-9a-f]{2}/gi, unescape);
    try {
      return decodeURIComponent(j);
    } catch {
      return j;
    }
  }, l = 1024, f = function(h, d, j, T, p) {
    if (h.length === 0)
      return h;
    var O = h;
    if (typeof h == "symbol" ? O = Symbol.prototype.toString.call(h) : typeof h != "string" && (O = String(h)), j === "iso-8859-1")
      return escape(O).replace(/%u[0-9a-f]{4}/gi, function(oe) {
        return "%26%23" + parseInt(oe.slice(2), 16) + "%3B";
      });
    for (var b = "", q = 0; q < O.length; q += l) {
      for (var A = O.length >= l ? O.slice(q, q + l) : O, E = [], L = 0; L < A.length; ++L) {
        var P = A.charCodeAt(L);
        if (P === 45 || P === 46 || P === 95 || P === 126 || P >= 48 && P <= 57 || P >= 65 && P <= 90 || P >= 97 && P <= 122 || p === e.RFC1738 && (P === 40 || P === 41)) {
          E[E.length] = A.charAt(L);
          continue;
        }
        if (P < 128) {
          E[E.length] = a[P];
          continue;
        }
        if (P < 2048) {
          E[E.length] = a[192 | P >> 6] + a[128 | P & 63];
          continue;
        }
        if (P < 55296 || P >= 57344) {
          E[E.length] = a[224 | P >> 12] + a[128 | P >> 6 & 63] + a[128 | P & 63];
          continue;
        }
        L += 1, P = 65536 + ((P & 1023) << 10 | A.charCodeAt(L) & 1023), E[E.length] = a[240 | P >> 18] + a[128 | P >> 12 & 63] + a[128 | P >> 6 & 63] + a[128 | P & 63];
      }
      b += E.join("");
    }
    return b;
  }, y = function(h) {
    for (var d = [{ obj: { o: h }, prop: "o" }], j = [], T = 0; T < d.length; ++T)
      for (var p = d[T], O = p.obj[p.prop], b = Object.keys(O), q = 0; q < b.length; ++q) {
        var A = b[q], E = O[A];
        typeof E == "object" && E !== null && j.indexOf(E) === -1 && (d.push({ obj: O, prop: A }), j.push(E));
      }
    return c(d), h;
  }, g = function(h) {
    return Object.prototype.toString.call(h) === "[object RegExp]";
  }, m = function(h) {
    return !h || typeof h != "object" ? !1 : !!(h.constructor && h.constructor.isBuffer && h.constructor.isBuffer(h));
  }, x = function(h, d) {
    return [].concat(h, d);
  }, v = function(h, d) {
    if (t(h)) {
      for (var j = [], T = 0; T < h.length; T += 1)
        j.push(d(h[T]));
      return j;
    }
    return d(h);
  };
  return ar = {
    arrayToObject: s,
    assign: o,
    combine: x,
    compact: y,
    decode: i,
    encode: f,
    isBuffer: m,
    isRegExp: g,
    maybeMap: v,
    merge: r
  }, ar;
}
var or, Gn;
function hs() {
  if (Gn) return or;
  Gn = 1;
  var e = ds(), n = /* @__PURE__ */ ma(), t = /* @__PURE__ */ wr(), a = Object.prototype.hasOwnProperty, c = {
    brackets: function(u) {
      return u + "[]";
    },
    comma: "comma",
    indices: function(u, h) {
      return u + "[" + h + "]";
    },
    repeat: function(u) {
      return u;
    }
  }, s = Array.isArray, r = Array.prototype.push, o = function(v, u) {
    r.apply(v, s(u) ? u : [u]);
  }, i = Date.prototype.toISOString, l = t.default, f = {
    addQueryPrefix: !1,
    allowDots: !1,
    allowEmptyArrays: !1,
    arrayFormat: "indices",
    charset: "utf-8",
    charsetSentinel: !1,
    delimiter: "&",
    encode: !0,
    encodeDotInKeys: !1,
    encoder: n.encode,
    encodeValuesOnly: !1,
    format: l,
    formatter: t.formatters[l],
    // deprecated
    indices: !1,
    serializeDate: function(u) {
      return i.call(u);
    },
    skipNulls: !1,
    strictNullHandling: !1
  }, y = function(u) {
    return typeof u == "string" || typeof u == "number" || typeof u == "boolean" || typeof u == "symbol" || typeof u == "bigint";
  }, g = {}, m = function v(u, h, d, j, T, p, O, b, q, A, E, L, P, oe, H, D, se, Z) {
    for (var N = u, re = Z, R = 0, k = !1; (re = re.get(g)) !== void 0 && !k; ) {
      var F = re.get(u);
      if (R += 1, typeof F < "u") {
        if (F === R)
          throw new RangeError("Cyclic object value");
        k = !0;
      }
      typeof re.get(g) > "u" && (R = 0);
    }
    if (typeof A == "function" ? N = A(h, N) : N instanceof Date ? N = P(N) : d === "comma" && s(N) && (N = n.maybeMap(N, function(Y) {
      return Y instanceof Date ? P(Y) : Y;
    })), N === null) {
      if (p)
        return q && !D ? q(h, f.encoder, se, "key", oe) : h;
      N = "";
    }
    if (y(N) || n.isBuffer(N)) {
      if (q) {
        var C = D ? h : q(h, f.encoder, se, "key", oe);
        return [H(C) + "=" + H(q(N, f.encoder, se, "value", oe))];
      }
      return [H(h) + "=" + H(String(N))];
    }
    var U = [];
    if (typeof N > "u")
      return U;
    var J;
    if (d === "comma" && s(N))
      D && q && (N = n.maybeMap(N, q)), J = [{ value: N.length > 0 ? N.join(",") || null : void 0 }];
    else if (s(A))
      J = A;
    else {
      var z = Object.keys(N);
      J = E ? z.sort(E) : z;
    }
    var K = b ? h.replace(/\./g, "%2E") : h, ne = j && s(N) && N.length === 1 ? K + "[]" : K;
    if (T && s(N) && N.length === 0)
      return ne + "[]";
    for (var G = 0; G < J.length; ++G) {
      var I = J[G], _ = typeof I == "object" && typeof I.value < "u" ? I.value : N[I];
      if (!(O && _ === null)) {
        var B = L && b ? I.replace(/\./g, "%2E") : I, Q = s(N) ? typeof d == "function" ? d(ne, B) : ne : ne + (L ? "." + B : "[" + B + "]");
        Z.set(u, R);
        var W = e();
        W.set(g, Z), o(U, v(
          _,
          Q,
          d,
          j,
          T,
          p,
          O,
          b,
          d === "comma" && D && s(N) ? null : q,
          A,
          E,
          L,
          P,
          oe,
          H,
          D,
          se,
          W
        ));
      }
    }
    return U;
  }, x = function(u) {
    if (!u)
      return f;
    if (typeof u.allowEmptyArrays < "u" && typeof u.allowEmptyArrays != "boolean")
      throw new TypeError("`allowEmptyArrays` option can only be `true` or `false`, when provided");
    if (typeof u.encodeDotInKeys < "u" && typeof u.encodeDotInKeys != "boolean")
      throw new TypeError("`encodeDotInKeys` option can only be `true` or `false`, when provided");
    if (u.encoder !== null && typeof u.encoder < "u" && typeof u.encoder != "function")
      throw new TypeError("Encoder has to be a function.");
    var h = u.charset || f.charset;
    if (typeof u.charset < "u" && u.charset !== "utf-8" && u.charset !== "iso-8859-1")
      throw new TypeError("The charset option must be either utf-8, iso-8859-1, or undefined");
    var d = t.default;
    if (typeof u.format < "u") {
      if (!a.call(t.formatters, u.format))
        throw new TypeError("Unknown format option provided.");
      d = u.format;
    }
    var j = t.formatters[d], T = f.filter;
    (typeof u.filter == "function" || s(u.filter)) && (T = u.filter);
    var p;
    if (u.arrayFormat in c ? p = u.arrayFormat : "indices" in u ? p = u.indices ? "indices" : "repeat" : p = f.arrayFormat, "commaRoundTrip" in u && typeof u.commaRoundTrip != "boolean")
      throw new TypeError("`commaRoundTrip` must be a boolean, or absent");
    var O = typeof u.allowDots > "u" ? u.encodeDotInKeys === !0 ? !0 : f.allowDots : !!u.allowDots;
    return {
      addQueryPrefix: typeof u.addQueryPrefix == "boolean" ? u.addQueryPrefix : f.addQueryPrefix,
      allowDots: O,
      allowEmptyArrays: typeof u.allowEmptyArrays == "boolean" ? !!u.allowEmptyArrays : f.allowEmptyArrays,
      arrayFormat: p,
      charset: h,
      charsetSentinel: typeof u.charsetSentinel == "boolean" ? u.charsetSentinel : f.charsetSentinel,
      commaRoundTrip: u.commaRoundTrip,
      delimiter: typeof u.delimiter > "u" ? f.delimiter : u.delimiter,
      encode: typeof u.encode == "boolean" ? u.encode : f.encode,
      encodeDotInKeys: typeof u.encodeDotInKeys == "boolean" ? u.encodeDotInKeys : f.encodeDotInKeys,
      encoder: typeof u.encoder == "function" ? u.encoder : f.encoder,
      encodeValuesOnly: typeof u.encodeValuesOnly == "boolean" ? u.encodeValuesOnly : f.encodeValuesOnly,
      filter: T,
      format: d,
      formatter: j,
      serializeDate: typeof u.serializeDate == "function" ? u.serializeDate : f.serializeDate,
      skipNulls: typeof u.skipNulls == "boolean" ? u.skipNulls : f.skipNulls,
      sort: typeof u.sort == "function" ? u.sort : null,
      strictNullHandling: typeof u.strictNullHandling == "boolean" ? u.strictNullHandling : f.strictNullHandling
    };
  };
  return or = function(v, u) {
    var h = v, d = x(u), j, T;
    typeof d.filter == "function" ? (T = d.filter, h = T("", h)) : s(d.filter) && (T = d.filter, j = T);
    var p = [];
    if (typeof h != "object" || h === null)
      return "";
    var O = c[d.arrayFormat], b = O === "comma" && d.commaRoundTrip;
    j || (j = Object.keys(h)), d.sort && j.sort(d.sort);
    for (var q = e(), A = 0; A < j.length; ++A) {
      var E = j[A];
      d.skipNulls && h[E] === null || o(p, m(
        h[E],
        E,
        O,
        b,
        d.allowEmptyArrays,
        d.strictNullHandling,
        d.skipNulls,
        d.encodeDotInKeys,
        d.encode ? d.encoder : null,
        d.filter,
        d.sort,
        d.allowDots,
        d.serializeDate,
        d.format,
        d.formatter,
        d.encodeValuesOnly,
        d.charset,
        q
      ));
    }
    var L = p.join(d.delimiter), P = d.addQueryPrefix === !0 ? "?" : "";
    return d.charsetSentinel && (d.charset === "iso-8859-1" ? P += "utf8=%26%2310003%3B&" : P += "utf8=%E2%9C%93&"), L.length > 0 ? P + L : "";
  }, or;
}
var ir, Vn;
function ys() {
  if (Vn) return ir;
  Vn = 1;
  var e = /* @__PURE__ */ ma(), n = Object.prototype.hasOwnProperty, t = Array.isArray, a = {
    allowDots: !1,
    allowEmptyArrays: !1,
    allowPrototypes: !1,
    allowSparse: !1,
    arrayLimit: 20,
    charset: "utf-8",
    charsetSentinel: !1,
    comma: !1,
    decodeDotInKeys: !1,
    decoder: e.decode,
    delimiter: "&",
    depth: 5,
    duplicates: "combine",
    ignoreQueryPrefix: !1,
    interpretNumericEntities: !1,
    parameterLimit: 1e3,
    parseArrays: !0,
    plainObjects: !1,
    strictDepth: !1,
    strictNullHandling: !1
  }, c = function(g) {
    return g.replace(/&#(\d+);/g, function(m, x) {
      return String.fromCharCode(parseInt(x, 10));
    });
  }, s = function(g, m) {
    return g && typeof g == "string" && m.comma && g.indexOf(",") > -1 ? g.split(",") : g;
  }, r = "utf8=%26%2310003%3B", o = "utf8=%E2%9C%93", i = function(m, x) {
    var v = { __proto__: null }, u = x.ignoreQueryPrefix ? m.replace(/^\?/, "") : m;
    u = u.replace(/%5B/gi, "[").replace(/%5D/gi, "]");
    var h = x.parameterLimit === 1 / 0 ? void 0 : x.parameterLimit, d = u.split(x.delimiter, h), j = -1, T, p = x.charset;
    if (x.charsetSentinel)
      for (T = 0; T < d.length; ++T)
        d[T].indexOf("utf8=") === 0 && (d[T] === o ? p = "utf-8" : d[T] === r && (p = "iso-8859-1"), j = T, T = d.length);
    for (T = 0; T < d.length; ++T)
      if (T !== j) {
        var O = d[T], b = O.indexOf("]="), q = b === -1 ? O.indexOf("=") : b + 1, A, E;
        q === -1 ? (A = x.decoder(O, a.decoder, p, "key"), E = x.strictNullHandling ? null : "") : (A = x.decoder(O.slice(0, q), a.decoder, p, "key"), E = e.maybeMap(
          s(O.slice(q + 1), x),
          function(P) {
            return x.decoder(P, a.decoder, p, "value");
          }
        )), E && x.interpretNumericEntities && p === "iso-8859-1" && (E = c(E)), O.indexOf("[]=") > -1 && (E = t(E) ? [E] : E);
        var L = n.call(v, A);
        L && x.duplicates === "combine" ? v[A] = e.combine(v[A], E) : (!L || x.duplicates === "last") && (v[A] = E);
      }
    return v;
  }, l = function(g, m, x, v) {
    for (var u = v ? m : s(m, x), h = g.length - 1; h >= 0; --h) {
      var d, j = g[h];
      if (j === "[]" && x.parseArrays)
        d = x.allowEmptyArrays && (u === "" || x.strictNullHandling && u === null) ? [] : [].concat(u);
      else {
        d = x.plainObjects ? /* @__PURE__ */ Object.create(null) : {};
        var T = j.charAt(0) === "[" && j.charAt(j.length - 1) === "]" ? j.slice(1, -1) : j, p = x.decodeDotInKeys ? T.replace(/%2E/g, ".") : T, O = parseInt(p, 10);
        !x.parseArrays && p === "" ? d = { 0: u } : !isNaN(O) && j !== p && String(O) === p && O >= 0 && x.parseArrays && O <= x.arrayLimit ? (d = [], d[O] = u) : p !== "__proto__" && (d[p] = u);
      }
      u = d;
    }
    return u;
  }, f = function(m, x, v, u) {
    if (m) {
      var h = v.allowDots ? m.replace(/\.([^.[]+)/g, "[$1]") : m, d = /(\[[^[\]]*])/, j = /(\[[^[\]]*])/g, T = v.depth > 0 && d.exec(h), p = T ? h.slice(0, T.index) : h, O = [];
      if (p) {
        if (!v.plainObjects && n.call(Object.prototype, p) && !v.allowPrototypes)
          return;
        O.push(p);
      }
      for (var b = 0; v.depth > 0 && (T = j.exec(h)) !== null && b < v.depth; ) {
        if (b += 1, !v.plainObjects && n.call(Object.prototype, T[1].slice(1, -1)) && !v.allowPrototypes)
          return;
        O.push(T[1]);
      }
      if (T) {
        if (v.strictDepth === !0)
          throw new RangeError("Input depth exceeded depth option of " + v.depth + " and strictDepth is true");
        O.push("[" + h.slice(T.index) + "]");
      }
      return l(O, x, v, u);
    }
  }, y = function(m) {
    if (!m)
      return a;
    if (typeof m.allowEmptyArrays < "u" && typeof m.allowEmptyArrays != "boolean")
      throw new TypeError("`allowEmptyArrays` option can only be `true` or `false`, when provided");
    if (typeof m.decodeDotInKeys < "u" && typeof m.decodeDotInKeys != "boolean")
      throw new TypeError("`decodeDotInKeys` option can only be `true` or `false`, when provided");
    if (m.decoder !== null && typeof m.decoder < "u" && typeof m.decoder != "function")
      throw new TypeError("Decoder has to be a function.");
    if (typeof m.charset < "u" && m.charset !== "utf-8" && m.charset !== "iso-8859-1")
      throw new TypeError("The charset option must be either utf-8, iso-8859-1, or undefined");
    var x = typeof m.charset > "u" ? a.charset : m.charset, v = typeof m.duplicates > "u" ? a.duplicates : m.duplicates;
    if (v !== "combine" && v !== "first" && v !== "last")
      throw new TypeError("The duplicates option must be either combine, first, or last");
    var u = typeof m.allowDots > "u" ? m.decodeDotInKeys === !0 ? !0 : a.allowDots : !!m.allowDots;
    return {
      allowDots: u,
      allowEmptyArrays: typeof m.allowEmptyArrays == "boolean" ? !!m.allowEmptyArrays : a.allowEmptyArrays,
      allowPrototypes: typeof m.allowPrototypes == "boolean" ? m.allowPrototypes : a.allowPrototypes,
      allowSparse: typeof m.allowSparse == "boolean" ? m.allowSparse : a.allowSparse,
      arrayLimit: typeof m.arrayLimit == "number" ? m.arrayLimit : a.arrayLimit,
      charset: x,
      charsetSentinel: typeof m.charsetSentinel == "boolean" ? m.charsetSentinel : a.charsetSentinel,
      comma: typeof m.comma == "boolean" ? m.comma : a.comma,
      decodeDotInKeys: typeof m.decodeDotInKeys == "boolean" ? m.decodeDotInKeys : a.decodeDotInKeys,
      decoder: typeof m.decoder == "function" ? m.decoder : a.decoder,
      delimiter: typeof m.delimiter == "string" || e.isRegExp(m.delimiter) ? m.delimiter : a.delimiter,
      // eslint-disable-next-line no-implicit-coercion, no-extra-parens
      depth: typeof m.depth == "number" || m.depth === !1 ? +m.depth : a.depth,
      duplicates: v,
      ignoreQueryPrefix: m.ignoreQueryPrefix === !0,
      interpretNumericEntities: typeof m.interpretNumericEntities == "boolean" ? m.interpretNumericEntities : a.interpretNumericEntities,
      parameterLimit: typeof m.parameterLimit == "number" ? m.parameterLimit : a.parameterLimit,
      parseArrays: m.parseArrays !== !1,
      plainObjects: typeof m.plainObjects == "boolean" ? m.plainObjects : a.plainObjects,
      strictDepth: typeof m.strictDepth == "boolean" ? !!m.strictDepth : a.strictDepth,
      strictNullHandling: typeof m.strictNullHandling == "boolean" ? m.strictNullHandling : a.strictNullHandling
    };
  };
  return ir = function(g, m) {
    var x = y(m);
    if (g === "" || g === null || typeof g > "u")
      return x.plainObjects ? /* @__PURE__ */ Object.create(null) : {};
    for (var v = typeof g == "string" ? i(g, x) : g, u = x.plainObjects ? /* @__PURE__ */ Object.create(null) : {}, h = Object.keys(v), d = 0; d < h.length; ++d) {
      var j = h[d], T = f(j, v[j], x, typeof g == "string");
      u = e.merge(u, T, x);
    }
    return x.allowSparse === !0 ? u : e.compact(u);
  }, ir;
}
var sr, Dn;
function ga() {
  if (Dn) return sr;
  Dn = 1;
  var e = /* @__PURE__ */ hs(), n = /* @__PURE__ */ ys(), t = /* @__PURE__ */ wr();
  return sr = {
    formats: t,
    parse: n,
    stringify: e
  }, sr;
}
var Qn;
function ms() {
  if (Qn) return Oe;
  Qn = 1;
  var e = Ii();
  function n() {
    this.protocol = null, this.slashes = null, this.auth = null, this.host = null, this.port = null, this.hostname = null, this.hash = null, this.search = null, this.query = null, this.pathname = null, this.path = null, this.href = null;
  }
  var t = /^([a-z0-9.+-]+:)/i, a = /:[0-9]*$/, c = /^(\/\/?(?!\/)[^?\s]*)(\?[^\s]*)?$/, s = [
    "<",
    ">",
    '"',
    "`",
    " ",
    "\r",
    `
`,
    "	"
  ], r = [
    "{",
    "}",
    "|",
    "\\",
    "^",
    "`"
  ].concat(s), o = ["'"].concat(r), i = [
    "%",
    "/",
    "?",
    ";",
    "#"
  ].concat(o), l = [
    "/",
    "?",
    "#"
  ], f = 255, y = /^[+a-z0-9A-Z_-]{0,63}$/, g = /^([+a-z0-9A-Z_-]{0,63})(.*)$/, m = {
    javascript: !0,
    "javascript:": !0
  }, x = {
    javascript: !0,
    "javascript:": !0
  }, v = {
    http: !0,
    https: !0,
    ftp: !0,
    gopher: !0,
    file: !0,
    "http:": !0,
    "https:": !0,
    "ftp:": !0,
    "gopher:": !0,
    "file:": !0
  }, u = /* @__PURE__ */ ga();
  function h(p, O, b) {
    if (p && typeof p == "object" && p instanceof n)
      return p;
    var q = new n();
    return q.parse(p, O, b), q;
  }
  n.prototype.parse = function(p, O, b) {
    if (typeof p != "string")
      throw new TypeError("Parameter 'url' must be a string, not " + typeof p);
    var q = p.indexOf("?"), A = q !== -1 && q < p.indexOf("#") ? "?" : "#", E = p.split(A), L = /\\/g;
    E[0] = E[0].replace(L, "/"), p = E.join(A);
    var P = p;
    if (P = P.trim(), !b && p.split("#").length === 1) {
      var oe = c.exec(P);
      if (oe)
        return this.path = P, this.href = P, this.pathname = oe[1], oe[2] ? (this.search = oe[2], O ? this.query = u.parse(this.search.substr(1)) : this.query = this.search.substr(1)) : O && (this.search = "", this.query = {}), this;
    }
    var H = t.exec(P);
    if (H) {
      H = H[0];
      var D = H.toLowerCase();
      this.protocol = D, P = P.substr(H.length);
    }
    if (b || H || P.match(/^\/\/[^@/]+@[^@/]+/)) {
      var se = P.substr(0, 2) === "//";
      se && !(H && x[H]) && (P = P.substr(2), this.slashes = !0);
    }
    if (!x[H] && (se || H && !v[H])) {
      for (var Z = -1, N = 0; N < l.length; N++) {
        var re = P.indexOf(l[N]);
        re !== -1 && (Z === -1 || re < Z) && (Z = re);
      }
      var R, k;
      Z === -1 ? k = P.lastIndexOf("@") : k = P.lastIndexOf("@", Z), k !== -1 && (R = P.slice(0, k), P = P.slice(k + 1), this.auth = decodeURIComponent(R)), Z = -1;
      for (var N = 0; N < i.length; N++) {
        var re = P.indexOf(i[N]);
        re !== -1 && (Z === -1 || re < Z) && (Z = re);
      }
      Z === -1 && (Z = P.length), this.host = P.slice(0, Z), P = P.slice(Z), this.parseHost(), this.hostname = this.hostname || "";
      var F = this.hostname[0] === "[" && this.hostname[this.hostname.length - 1] === "]";
      if (!F)
        for (var C = this.hostname.split(/\./), N = 0, U = C.length; N < U; N++) {
          var J = C[N];
          if (J && !J.match(y)) {
            for (var z = "", K = 0, ne = J.length; K < ne; K++)
              J.charCodeAt(K) > 127 ? z += "x" : z += J[K];
            if (!z.match(y)) {
              var G = C.slice(0, N), I = C.slice(N + 1), _ = J.match(g);
              _ && (G.push(_[1]), I.unshift(_[2])), I.length && (P = "/" + I.join(".") + P), this.hostname = G.join(".");
              break;
            }
          }
        }
      this.hostname.length > f ? this.hostname = "" : this.hostname = this.hostname.toLowerCase(), F || (this.hostname = e.toASCII(this.hostname));
      var B = this.port ? ":" + this.port : "", Q = this.hostname || "";
      this.host = Q + B, this.href += this.host, F && (this.hostname = this.hostname.substr(1, this.hostname.length - 2), P[0] !== "/" && (P = "/" + P));
    }
    if (!m[D])
      for (var N = 0, U = o.length; N < U; N++) {
        var W = o[N];
        if (P.indexOf(W) !== -1) {
          var Y = encodeURIComponent(W);
          Y === W && (Y = escape(W)), P = P.split(W).join(Y);
        }
      }
    var ae = P.indexOf("#");
    ae !== -1 && (this.hash = P.substr(ae), P = P.slice(0, ae));
    var he = P.indexOf("?");
    if (he !== -1 ? (this.search = P.substr(he), this.query = P.substr(he + 1), O && (this.query = u.parse(this.query)), P = P.slice(0, he)) : O && (this.search = "", this.query = {}), P && (this.pathname = P), v[D] && this.hostname && !this.pathname && (this.pathname = "/"), this.pathname || this.search) {
      var B = this.pathname || "", $e = this.search || "";
      this.path = B + $e;
    }
    return this.href = this.format(), this;
  };
  function d(p) {
    return typeof p == "string" && (p = h(p)), p instanceof n ? p.format() : n.prototype.format.call(p);
  }
  n.prototype.format = function() {
    var p = this.auth || "";
    p && (p = encodeURIComponent(p), p = p.replace(/%3A/i, ":"), p += "@");
    var O = this.protocol || "", b = this.pathname || "", q = this.hash || "", A = !1, E = "";
    this.host ? A = p + this.host : this.hostname && (A = p + (this.hostname.indexOf(":") === -1 ? this.hostname : "[" + this.hostname + "]"), this.port && (A += ":" + this.port)), this.query && typeof this.query == "object" && Object.keys(this.query).length && (E = u.stringify(this.query, {
      arrayFormat: "repeat",
      addQueryPrefix: !1
    }));
    var L = this.search || E && "?" + E || "";
    return O && O.substr(-1) !== ":" && (O += ":"), this.slashes || (!O || v[O]) && A !== !1 ? (A = "//" + (A || ""), b && b.charAt(0) !== "/" && (b = "/" + b)) : A || (A = ""), q && q.charAt(0) !== "#" && (q = "#" + q), L && L.charAt(0) !== "?" && (L = "?" + L), b = b.replace(/[?#]/g, function(P) {
      return encodeURIComponent(P);
    }), L = L.replace("#", "%23"), O + A + b + L + q;
  };
  function j(p, O) {
    return h(p, !1, !0).resolve(O);
  }
  n.prototype.resolve = function(p) {
    return this.resolveObject(h(p, !1, !0)).format();
  };
  function T(p, O) {
    return p ? h(p, !1, !0).resolveObject(O) : O;
  }
  return n.prototype.resolveObject = function(p) {
    if (typeof p == "string") {
      var O = new n();
      O.parse(p, !1, !0), p = O;
    }
    for (var b = new n(), q = Object.keys(this), A = 0; A < q.length; A++) {
      var E = q[A];
      b[E] = this[E];
    }
    if (b.hash = p.hash, p.href === "")
      return b.href = b.format(), b;
    if (p.slashes && !p.protocol) {
      for (var L = Object.keys(p), P = 0; P < L.length; P++) {
        var oe = L[P];
        oe !== "protocol" && (b[oe] = p[oe]);
      }
      return v[b.protocol] && b.hostname && !b.pathname && (b.pathname = "/", b.path = b.pathname), b.href = b.format(), b;
    }
    if (p.protocol && p.protocol !== b.protocol) {
      if (!v[p.protocol]) {
        for (var H = Object.keys(p), D = 0; D < H.length; D++) {
          var se = H[D];
          b[se] = p[se];
        }
        return b.href = b.format(), b;
      }
      if (b.protocol = p.protocol, !p.host && !x[p.protocol]) {
        for (var U = (p.pathname || "").split("/"); U.length && !(p.host = U.shift()); )
          ;
        p.host || (p.host = ""), p.hostname || (p.hostname = ""), U[0] !== "" && U.unshift(""), U.length < 2 && U.unshift(""), b.pathname = U.join("/");
      } else
        b.pathname = p.pathname;
      if (b.search = p.search, b.query = p.query, b.host = p.host || "", b.auth = p.auth, b.hostname = p.hostname || p.host, b.port = p.port, b.pathname || b.search) {
        var Z = b.pathname || "", N = b.search || "";
        b.path = Z + N;
      }
      return b.slashes = b.slashes || p.slashes, b.href = b.format(), b;
    }
    var re = b.pathname && b.pathname.charAt(0) === "/", R = p.host || p.pathname && p.pathname.charAt(0) === "/", k = R || re || b.host && p.pathname, F = k, C = b.pathname && b.pathname.split("/") || [], U = p.pathname && p.pathname.split("/") || [], J = b.protocol && !v[b.protocol];
    if (J && (b.hostname = "", b.port = null, b.host && (C[0] === "" ? C[0] = b.host : C.unshift(b.host)), b.host = "", p.protocol && (p.hostname = null, p.port = null, p.host && (U[0] === "" ? U[0] = p.host : U.unshift(p.host)), p.host = null), k = k && (U[0] === "" || C[0] === "")), R)
      b.host = p.host || p.host === "" ? p.host : b.host, b.hostname = p.hostname || p.hostname === "" ? p.hostname : b.hostname, b.search = p.search, b.query = p.query, C = U;
    else if (U.length)
      C || (C = []), C.pop(), C = C.concat(U), b.search = p.search, b.query = p.query;
    else if (p.search != null) {
      if (J) {
        b.host = C.shift(), b.hostname = b.host;
        var z = b.host && b.host.indexOf("@") > 0 ? b.host.split("@") : !1;
        z && (b.auth = z.shift(), b.hostname = z.shift(), b.host = b.hostname);
      }
      return b.search = p.search, b.query = p.query, (b.pathname !== null || b.search !== null) && (b.path = (b.pathname ? b.pathname : "") + (b.search ? b.search : "")), b.href = b.format(), b;
    }
    if (!C.length)
      return b.pathname = null, b.search ? b.path = "/" + b.search : b.path = null, b.href = b.format(), b;
    for (var K = C.slice(-1)[0], ne = (b.host || p.host || C.length > 1) && (K === "." || K === "..") || K === "", G = 0, I = C.length; I >= 0; I--)
      K = C[I], K === "." ? C.splice(I, 1) : K === ".." ? (C.splice(I, 1), G++) : G && (C.splice(I, 1), G--);
    if (!k && !F)
      for (; G--; G)
        C.unshift("..");
    k && C[0] !== "" && (!C[0] || C[0].charAt(0) !== "/") && C.unshift(""), ne && C.join("/").substr(-1) !== "/" && C.push("");
    var _ = C[0] === "" || C[0] && C[0].charAt(0) === "/";
    if (J) {
      b.hostname = _ ? "" : C.length ? C.shift() : "", b.host = b.hostname;
      var z = b.host && b.host.indexOf("@") > 0 ? b.host.split("@") : !1;
      z && (b.auth = z.shift(), b.hostname = z.shift(), b.host = b.hostname);
    }
    return k = k || b.host && C.length, k && !_ && C.unshift(""), C.length > 0 ? b.pathname = C.join("/") : (b.pathname = null, b.path = null), (b.pathname !== null || b.search !== null) && (b.path = (b.pathname ? b.pathname : "") + (b.search ? b.search : "")), b.auth = p.auth || b.auth, b.slashes = b.slashes || p.slashes, b.href = b.format(), b;
  }, n.prototype.parseHost = function() {
    var p = this.host, O = a.exec(p);
    O && (O = O[0], O !== ":" && (this.port = O.substr(1)), p = p.substr(0, p.length - O.length)), p && (this.hostname = p);
  }, Oe.parse = h, Oe.resolve = j, Oe.resolveObject = T, Oe.format = d, Oe.Url = n, Oe;
}
var lr = ms(), cr = /* @__PURE__ */ ga(), gs = (e) => typeof e == "object" && "log" in e && typeof e.log == "object" && "entries" in e.log && Array.isArray(e.log.entries), vs = class {
  constructor(e, n = {}) {
    this.initCalled = !1, this.entries = [], this.requests = [], this.options = {}, this.options = {
      harIsAlreadyEncoded: !1,
      ...n
    }, this.requests = [], gs(e) ? this.entries = e.log.entries : this.entries = [
      {
        request: e
      }
    ];
  }
  init() {
    return this.initCalled = !0, this.requests = this.entries.map(({ request: e }) => {
      var t;
      const n = {
        bodySize: 0,
        headersSize: 0,
        headers: [],
        cookies: [],
        httpVersion: "HTTP/1.1",
        queryString: [],
        postData: {
          mimeType: ((t = e.postData) == null ? void 0 : t.mimeType) || "application/octet-stream"
        },
        ...e
      };
      return n.postData && !n.postData.mimeType && (n.postData.mimeType = "application/octet-stream"), this.prepare(n, this.options);
    }), this;
  }
  prepare(e, n) {
    var f, y, g;
    const t = {
      ...e,
      fullUrl: "",
      uriObj: {},
      queryObj: {},
      headersObj: {},
      cookiesObj: {},
      allHeaders: {}
    };
    if (t.queryString && t.queryString.length && (t.queryObj = t.queryString.reduce(_r, {})), t.headers && t.headers.length) {
      const m = /^HTTP\/2/;
      t.headersObj = t.headers.reduce((x, { name: v, value: u }) => {
        const h = m.exec(t.httpVersion) ? v.toLocaleLowerCase() : v;
        return {
          ...x,
          [h]: u
        };
      }, {});
    }
    t.cookies && t.cookies.length && (t.cookiesObj = t.cookies.reduceRight(
      (m, { name: x, value: v }) => ({
        ...m,
        [x]: v
      }),
      {}
    ));
    const a = (f = t.cookies) == null ? void 0 : f.map(({ name: m, value: x }) => n.harIsAlreadyEncoded ? `${m}=${x}` : `${encodeURIComponent(m)}=${encodeURIComponent(x)}`);
    switch (a != null && a.length && (t.allHeaders.cookie = a.join("; ")), t.postData.mimeType) {
      case "multipart/mixed":
      case "multipart/related":
      case "multipart/form-data":
      case "multipart/alternative":
        if (t.postData.text = "", t.postData.mimeType = "multipart/form-data", (y = t.postData) != null && y.params) {
          const m = "---011000010111000001101001", x = `${m}--`, v = `\r
`, u = (T) => T.replace(/\n/g, "%0A").replace(/\r/g, "%0D").replace(/"/g, "%22"), h = (T) => T.replace(/\r?\n|\r/g, `\r
`), d = [`--${m}`];
          (g = t.postData) == null || g.params.forEach((T, p) => {
            const O = T.name, b = T.value || "", q = T.fileName || null, A = T.contentType || "application/octet-stream";
            q ? (d.push(
              `Content-Disposition: form-data; name="${u(h(O))}"; filename="${q}"`
            ), d.push(`Content-Type: ${A}`)) : d.push(`Content-Disposition: form-data; name="${u(h(O))}"`), d.push(""), d.push(h(b)), p !== t.postData.params.length - 1 && d.push(`--${m}`);
          }), d.push(`--${x}`), t.postData.boundary = m, t.postData.text = d.join(v);
          const j = fe(t.headersObj, "content-type") || "content-type";
          t.headersObj[j] = `multipart/form-data; boundary=${m}`;
        }
        break;
      case "application/x-www-form-urlencoded":
        t.postData.params ? (t.postData.paramsObj = t.postData.params.reduce(_r, {}), t.postData.text = cr.stringify(t.postData.paramsObj)) : t.postData.text = "";
        break;
      case "text/json":
      case "text/x-json":
      case "application/json":
      case "application/x-json":
        if (t.postData.mimeType = "application/json", t.postData.text)
          try {
            t.postData.jsonObj = JSON.parse(t.postData.text);
          } catch {
            t.postData.mimeType = "text/plain";
          }
        break;
    }
    const c = {
      ...t.allHeaders,
      ...t.headersObj
    }, s = lr.parse(t.url, !0, !0);
    t.queryObj = {
      ...t.queryObj,
      ...s.query
    };
    let r;
    n.harIsAlreadyEncoded ? r = cr.stringify(t.queryObj, {
      encode: !1,
      indices: !1
    }) : r = cr.stringify(t.queryObj, {
      indices: !1
    });
    const o = {
      ...s,
      query: t.queryObj,
      search: r,
      path: r ? `${s.pathname}?${r}` : s.pathname
    }, i = lr.format({
      ...s,
      query: null,
      search: null
    }), l = lr.format({
      ...s,
      ...o
    });
    return {
      ...t,
      allHeaders: c,
      fullUrl: l,
      url: i,
      uriObj: o
    };
  }
  convert(e, n, t) {
    this.initCalled || this.init(), !t && n && (t = n);
    const a = Ni[e];
    if (!a)
      return !1;
    const { convert: c } = a.clientsById[n || a.info.default];
    return this.requests.map((r) => c(r, t));
  }
};
const bs = ({
  rootMargin: e = "0px",
  threshold: n,
  root: t
} = {}) => {
  const [a, c] = xe(!1), s = aa(null);
  return gr(() => {
    const r = new IntersectionObserver(
      ([i]) => {
        i && c(i.isIntersecting);
      },
      { rootMargin: e, threshold: n, root: t }
    ), o = s.current;
    if (o)
      return r.observe(o), () => r.unobserve(o);
  }, [t, e, n]), [s, a];
}, xs = typeof window < "u" ? Qa : gr, va = ({
  children: e,
  maxHeight: n = 250
}) => {
  const t = aa(null), [a, c] = xe(!1), [s, r] = xe(!1);
  return xs(() => {
    const o = t.current;
    if (!o) return;
    c(o.scrollHeight > n);
    const i = new ResizeObserver(() => {
      c(o.scrollHeight > n);
    });
    return i.observe(o), () => i.disconnect();
  }, [n]), /* @__PURE__ */ $.jsxs(
    Zn,
    {
      className: "group",
      open: s,
      onOpenChange: r,
      style: { "--max-height": `${n}px` },
      children: [
        /* @__PURE__ */ $.jsxs(
          ea,
          {
            forceMount: !0,
            className: pe(
              "relative overflow-hidden group",
              !s && a && "max-h-[--max-height]"
            ),
            children: [
              !s && a && /* @__PURE__ */ $.jsx("div", { className: " absolute inset-0 bg-gradient-to-b from-transparent to-zinc-50/60 dark:to-zinc-950/90 z-10 group-hover:to-transparent" }),
              /* @__PURE__ */ $.jsx("div", { ref: t, children: e }),
              !s && a && /* @__PURE__ */ $.jsx(
                ur,
                {
                  className: "absolute inset-0 grid place-items-center z-10 cursor-pointer peer",
                  asChild: !0,
                  children: /* @__PURE__ */ $.jsx("div", { children: /* @__PURE__ */ $.jsxs(Ur, { variant: "outline", className: "hidden group-hover:flex", children: [
                    /* @__PURE__ */ $.jsx(Na, { size: 14, className: "mr-1.5" }),
                    "Click to expand"
                  ] }) })
                }
              )
            ]
          }
        ),
        a && /* @__PURE__ */ $.jsx(
          "div",
          {
            className: pe("flex justify-center w-full mb-2", !s && "hidden"),
            children: /* @__PURE__ */ $.jsx(ur, { asChild: !0, children: /* @__PURE__ */ $.jsxs(Ur, { variant: "outline", size: "sm", children: [
              "Collapse",
              /* @__PURE__ */ $.jsx(Aa, { size: 14, className: "ml-1.5" })
            ] }) })
          }
        )
      ]
    }
  );
}, $s = ({
  server: e,
  servers: n,
  operation: t,
  examples: a
}) => {
  var g, m, x;
  const { isAuthEnabled: c, login: s, signup: r, isPending: o, isAuthenticated: i } = lo(), l = (g = t.parameters) == null ? void 0 : g.filter((v) => v.in === "header").sort((v, u) => v.required && !u.required ? -1 : 1).map((v) => {
    var u, h, d, j, T, p, O, b;
    return {
      name: v.name,
      defaultValue: ((u = v.schema) == null ? void 0 : u.default) ?? ((d = (h = v.examples) == null ? void 0 : h.find((q) => q.value)) == null ? void 0 : d.value) ?? "",
      defaultActive: v.required ?? !1,
      isRequired: v.required ?? !1,
      enum: ((j = v.schema) == null ? void 0 : j.type) == "array" ? (p = (T = v.schema) == null ? void 0 : T.items) == null ? void 0 : p.enum : (O = v.schema) == null ? void 0 : O.enum,
      type: ((b = v.schema) == null ? void 0 : b.type) ?? "string"
    };
  }), f = (m = t.parameters) == null ? void 0 : m.filter((v) => v.in === "query").sort((v, u) => v.required && !u.required ? -1 : 1).map((v) => {
    var u, h, d, j, T;
    return {
      name: v.name,
      defaultActive: v.required ?? !1,
      isRequired: v.required ?? !1,
      enum: ((u = v.schema) == null ? void 0 : u.type) == "array" ? (d = (h = v.schema) == null ? void 0 : h.items) == null ? void 0 : d.enum : (j = v.schema) == null ? void 0 : j.enum,
      type: ((T = v.schema) == null ? void 0 : T.type) ?? "string"
    };
  }), y = (x = t.parameters) == null ? void 0 : x.filter((v) => v.in === "path").map((v) => ({ name: v.name }));
  return /* @__PURE__ */ $.jsx(
    Fa,
    {
      server: e,
      servers: n,
      method: t.method,
      url: t.path,
      headers: l,
      queryParams: f,
      pathParams: y,
      examples: a,
      requiresLogin: c && !i && !o,
      onLogin: () => s(),
      onSignUp: () => r()
    }
  );
}, Sr = ({ children: e, className: n }) => /* @__PURE__ */ $.jsx(
  "div",
  {
    className: pe("rounded-xl overflow-hidden border border-border", n),
    children: e
  }
), jr = ({ children: e, className: n }) => /* @__PURE__ */ $.jsx(
  "div",
  {
    className: pe(
      "border-b bg-muted dark:bg-transparent text-card-foreground p-3 py-2.5",
      n
    ),
    children: e
  }
), ba = ({ children: e, className: n }) => /* @__PURE__ */ $.jsx("div", { className: pe("bg-card overflow-auto p-2", n), children: e }), xa = ({ children: e, className: n }) => /* @__PURE__ */ $.jsx("div", { className: pe("border-t bg-muted dark:bg-transparent p-3", n), children: e }), Ue = (e, n) => {
  if (e.example !== void 0)
    return e.example;
  if (e.examples && typeof e.examples == "object" && "default" in e.examples) {
    const t = e.examples.default;
    if (t !== null)
      return typeof t == "object" && "value" in t ? t.value : t;
  }
  if (e.type === "object" && e.properties) {
    const t = {};
    for (const [a, c] of Object.entries(e.properties))
      typeof c == "object" && (t[a] = Ue(c, a));
    return t;
  }
  if (e.type === "array")
    return Array.isArray(e.items) ? e.items.map(
      (t) => Ue(t)
    ) : e.items ? [Ue(e.items)] : [];
  if (e.enum)
    return e.enum[0];
  switch (e.type) {
    case "string":
      return n || "string";
    case "number":
    case "integer":
      return 0;
    case "boolean":
      return !0;
    case "null":
      return null;
    case "object":
      return {};
    case void 0:
    default:
      return {};
  }
}, ws = (e) => e == null ? "No example" : typeof e == "string" ? e.trim() : JSON.stringify(e, null, 2), Ss = (e) => e ? {
  "application/json": "json",
  "application/xml": "xml",
  "application/x-yaml": "yaml",
  "text/csv": "csv",
  "application/javascript": "javascript",
  "application/graphql": "graphql"
}[e] ?? "plain" : "plain", $a = ({
  content: e,
  description: n,
  onExampleChange: t
}) => {
  var m;
  const [a, c] = xe(0), [s, r] = xe(0), o = Array.isArray(e) && e.length === 1 ? e[0] : e[a], i = (o == null ? void 0 : o.examples) ?? [], l = i[s], f = oa(() => {
    if (l)
      return "value" in l ? l.value : l;
    if (o != null && o.schema)
      return Ue(o.schema);
  }, [l, o == null ? void 0 : o.schema]);
  gr(() => {
    f && (t == null || t(f));
  }, [f, t]);
  const y = ws(f), g = Ss(o == null ? void 0 : o.mediaType);
  return /* @__PURE__ */ $.jsxs($.Fragment, { children: [
    /* @__PURE__ */ $.jsxs(ba, { className: "p-0", children: [
      l != null && l.externalValue ? /* @__PURE__ */ $.jsx("div", { className: "p-2", children: /* @__PURE__ */ $.jsx(
        "a",
        {
          href: l.externalValue,
          target: "_blank",
          rel: "noopener noreferrer",
          className: "text-xs text-primary hover:underline",
          children: "View External Example →"
        }
      ) }) : /* @__PURE__ */ $.jsx(va, { children: /* @__PURE__ */ $.jsx(
        ia,
        {
          language: g,
          noBackground: !0,
          className: "[--scrollbar-color:gray] text-xs max-h-[500px] p-2",
          code: y
        }
      ) }),
      (l == null ? void 0 : l.description) && /* @__PURE__ */ $.jsx("div", { className: "border-t text-xs px-3 py-1.5 text-muted-foreground", children: l.description })
    ] }),
    /* @__PURE__ */ $.jsxs(xa, { className: "text-xs p-0 divide-y divide-border", children: [
      n && /* @__PURE__ */ $.jsx("div", { className: "text-muted-foreground text-xs px-3 py-2", children: n }),
      (i.length !== 0 || e.length !== 0) && /* @__PURE__ */ $.jsxs("div", { className: "flex items-center gap-2 justify-between min-w-0 px-3 py-2", children: [
        /* @__PURE__ */ $.jsx("div", { className: "flex items-center gap-2 min-w-0", children: e.length > 1 ? /* @__PURE__ */ $.jsx(
          Ye,
          {
            className: "max-w-[200px]",
            value: a.toString(),
            onChange: (x) => c(Number(x.target.value)),
            options: e.map((x, v) => ({
              value: v.toString(),
              label: x.mediaType
            }))
          }
        ) : /* @__PURE__ */ $.jsx("span", { className: "font-mono text-[11px]", children: (m = e[0]) == null ? void 0 : m.mediaType }) }),
        i.length > 1 && /* @__PURE__ */ $.jsx("div", { className: "flex items-center gap-1", children: /* @__PURE__ */ $.jsx(
          Ye,
          {
            className: "max-w-[180px]",
            value: s.toString(),
            onChange: (x) => r(Number(x.target.value)),
            options: i.map((x, v) => ({
              value: v.toString(),
              label: x.summary || x.name || x.description || `Example ${v + 1}`
            }))
          }
        ) })
      ] })
    ] })
  ] });
}, js = ({
  content: e,
  onExampleChange: n
}) => e.length === 0 ? null : /* @__PURE__ */ $.jsxs(Sr, { children: [
  /* @__PURE__ */ $.jsx(jr, { className: "text-xs flex justify-between items-center", children: /* @__PURE__ */ $.jsx("span", { className: "font-mono", children: "Request Body Example" }) }),
  /* @__PURE__ */ $.jsx($a, { content: e, onExampleChange: n })
] }), Os = ({ response: e }) => /* @__PURE__ */ $.jsx(Ge.Content, { value: e.statusCode, children: /* @__PURE__ */ $.jsx(
  $a,
  {
    content: e.content ?? [],
    description: e.description ?? void 0
  }
) }), Ts = ({
  responses: e,
  selectedResponse: n,
  onSelectResponse: t
}) => {
  var a;
  return /* @__PURE__ */ $.jsx(Sr, { children: /* @__PURE__ */ $.jsxs(
    Ge.Root,
    {
      defaultValue: (a = e[0]) == null ? void 0 : a.statusCode,
      value: n,
      onValueChange: (c) => t(c),
      children: [
        /* @__PURE__ */ $.jsxs(jr, { className: "text-xs flex flex-col gap-2 pb-0", children: [
          /* @__PURE__ */ $.jsx("span", { className: "font-mono", children: "Example Responses" }),
          /* @__PURE__ */ $.jsx(Ge.List, { className: "flex gap-2", children: e.map((c) => /* @__PURE__ */ $.jsx(
            Ge.Trigger,
            {
              value: c.statusCode,
              className: pe(
                "text-xs font-mono px-1.5 py-1 pb-px translate-y-px border-b-2 border-transparent rounded-t cursor-pointer",
                "data-[state=active]:text-primary data-[state=active]:dark:text-inherit data-[state=active]:border-primary",
                "hover:border-accent-foreground/25"
              ),
              children: c.statusCode
            },
            c.statusCode
          )) })
        ] }),
        e.map((c) => /* @__PURE__ */ $.jsx(Os, { response: c }, c.statusCode))
      ]
    }
  ) });
}, Ps = (e, n) => {
  let t;
  switch (n) {
    case "shell":
      t = e.convert("shell", "curl");
      break;
    case "js":
      t = e.convert("javascript", "fetch");
      break;
    case "python":
      t = e.convert("python", "requests");
      break;
    case "java":
      t = e.convert("java", "okhttp");
      break;
    case "go":
      t = e.convert("go", "native");
      break;
    case "csharp":
      t = e.convert("csharp", "httpclient");
      break;
    case "kotlin":
      t = e.convert("kotlin", "okhttp");
      break;
    case "objc":
      t = e.convert("objc", "nsurlsession");
      break;
    case "php":
      t = e.convert("php", "http2");
      break;
    case "ruby":
      t = e.convert("ruby");
      break;
    case "swift":
      t = e.convert("swift");
      break;
    default:
      t = e.convert("shell");
      break;
  }
  return t ? t[0] : "";
}, qs = _e(
  /* GraphQL */
  `
  query getServerQuery($input: JSON!, $type: SchemaType!) {
    schema(input: $input, type: $type) {
      url
      servers {
        url
      }
    }
  }
`
), Cs = [
  { value: "shell", label: "cURL" },
  { value: "js", label: "JavaScript" },
  { value: "python", label: "Python" },
  { value: "java", label: "Java" },
  { value: "go", label: "Go" },
  { value: "csharp", label: "C#" },
  { value: "kotlin", label: "Kotlin" },
  { value: "objc", label: "Objective-C" },
  { value: "php", label: "PHP" },
  { value: "ruby", label: "Ruby" },
  { value: "swift", label: "Swift" }
], Es = ({
  operation: e,
  selectedResponse: n,
  onSelectResponse: t
}) => {
  var A;
  const { input: a, type: c, options: s } = vr(), r = Pa(), o = Qe(qs, { input: a, type: c }), i = yr(o), l = co(), f = ta(e.method), [y, g] = Ua(), [, m] = na(), [x, v] = xe(), u = y.get("lang") ?? (s == null ? void 0 : s.examplesLanguage) ?? "shell", h = (A = e.requestBody) == null ? void 0 : A.content, d = h && (s != null && s.transformExamples) ? s.transformExamples({
    auth: r,
    type: "request",
    operation: e,
    content: h,
    context: l
  }) : h, j = /* @__PURE__ */ $.jsx(
    _a,
    {
      path: e.path,
      renderParam: ({ name: E }) => /* @__PURE__ */ $.jsx(
        Xn,
        {
          name: E,
          backgroundOpacity: "0",
          slug: `${e.slug}-${E}`,
          children: `{${E}}`
        }
      )
    }
  ), { selectedServer: T } = mr(i.data.schema.servers), p = oa(() => {
    var P;
    const E = x ?? ((P = d == null ? void 0 : d[0]) != null && P.schema ? Ue(
      d[0].schema
    ) : void 0), L = new vs({
      method: e.method.toLocaleUpperCase(),
      url: T + e.path.replaceAll("{", ":").replaceAll("}", ""),
      postData: E ? {
        text: JSON.stringify(E, null, 2),
        mimeType: "application/json"
      } : {},
      headers: [{ name: "Content-Type", value: "application/json" }],
      queryString: [],
      httpVersion: "",
      cookies: [],
      headersSize: 0,
      bodySize: 0
    });
    return Ps(L, u);
  }, [
    x,
    d,
    e.method,
    e.path,
    T,
    u
  ]), [O, b] = bs({ rootMargin: "200px 0px 200px 0px" }), q = b && (e.extensions["x-explorer-enabled"] === !0 || e.extensions["x-zudoku-playground-enabled"] === !0 || e.extensions["x-explorer-enabled"] === void 0 && e.extensions["x-zudoku-playground-enabled"] === void 0 && !(s != null && s.disablePlayground));
  return /* @__PURE__ */ $.jsxs(
    "aside",
    {
      ref: O,
      className: "flex flex-col overflow-hidden sticky top-[--scroll-padding] gap-4",
      "data-pagefind-ignore": "all",
      children: [
        /* @__PURE__ */ $.jsxs(Sr, { children: [
          /* @__PURE__ */ $.jsxs(jr, { className: "flex justify-between items-center flex-nowrap py-2.5 gap-2 text-xs", children: [
            /* @__PURE__ */ $.jsxs("span", { className: "font-mono break-words leading-6", children: [
              /* @__PURE__ */ $.jsx("span", { className: pe("font-semibold", f), children: e.method.toLocaleUpperCase() }),
              " ",
              j
            ] }),
            q && /* @__PURE__ */ $.jsx(
              $s,
              {
                servers: i.data.schema.servers.map((E) => E.url),
                operation: e,
                examples: h ?? void 0
              }
            )
          ] }),
          b && /* @__PURE__ */ $.jsxs($.Fragment, { children: [
            /* @__PURE__ */ $.jsx(ba, { className: "p-0", children: /* @__PURE__ */ $.jsx(va, { children: /* @__PURE__ */ $.jsx(
              ia,
              {
                language: u,
                noBackground: !0,
                className: "[--scrollbar-color:gray] text-xs max-h-[500px] p-2",
                code: p
              }
            ) }) }),
            /* @__PURE__ */ $.jsxs(xa, { className: "flex items-center text-xs gap-2 justify-end py-2.5", children: [
              /* @__PURE__ */ $.jsx("span", { children: "Show example in" }),
              /* @__PURE__ */ $.jsx(
                Ye,
                {
                  className: "self-start max-w-[150px]",
                  value: u,
                  onChange: (E) => {
                    m(() => {
                      g((L) => (L.set("lang", E.target.value), L));
                    });
                  },
                  options: Cs
                }
              )
            ] })
          ] })
        ] }),
        b && d && /* @__PURE__ */ $.jsx(
          js,
          {
            content: d,
            onExampleChange: v
          }
        ),
        b && e.responses.length > 0 && /* @__PURE__ */ $.jsx(
          Ts,
          {
            selectedResponse: n,
            onSelectResponse: t,
            responses: e.responses.map((E) => ({
              ...E,
              content: E.content && (s != null && s.transformExamples) ? s.transformExamples({
                auth: r,
                type: "response",
                context: l,
                operation: e,
                content: E.content
              }) : E.content
            }))
          }
        )
      ]
    }
  );
}, Rs = ["path", "query", "header", "cookie"], ks = ({
  operationFragment: e,
  serverUrl: n
}) => {
  var o, i, l;
  const t = uo(Ns, e), a = Za(
    t.parameters ?? [],
    (f) => f.in
  ), c = t.responses.at(0), [s, r] = xe(c == null ? void 0 : c.statusCode);
  return /* @__PURE__ */ $.jsxs("div", { children: [
    t.deprecated && /* @__PURE__ */ $.jsx(Fr, { variant: "muted", className: "text-xs mb-4", children: "deprecated" }),
    /* @__PURE__ */ $.jsxs(
      "div",
      {
        className: pe(
          "grid grid-cols-1 lg:grid-cols-[minmax(0,4fr)_minmax(0,3fr)] gap-x-8 gap-y-4 items-start",
          t.deprecated && "opacity-50 transition hover:opacity-100"
        ),
        children: [
          /* @__PURE__ */ $.jsx(
            Ie,
            {
              level: 2,
              id: t.slug,
              registerSidebarAnchor: !0,
              className: "break-all col-span-full",
              children: t.summary
            }
          ),
          /* @__PURE__ */ $.jsxs("div", { className: "text-sm flex gap-2 font-mono col-span-full", children: [
            /* @__PURE__ */ $.jsx("span", { className: ta(t.method), children: t.method.toUpperCase() }),
            /* @__PURE__ */ $.jsxs(ra, { className: "max-w-full truncate flex cursor-pointer", children: [
              n && /* @__PURE__ */ $.jsx("div", { className: "text-neutral-400 dark:text-neutral-500 truncate", children: n.replace(/\/$/, "") }),
              /* @__PURE__ */ $.jsx("div", { className: "text-neutral-900 dark:text-neutral-200", children: t.path })
            ] })
          ] }),
          /* @__PURE__ */ $.jsxs("div", { className: "flex flex-col gap-4", children: [
            t.description && /* @__PURE__ */ $.jsx(
              De,
              {
                className: `${fr} max-w-full prose-img:max-w-prose`,
                content: t.description
              }
            ),
            t.parameters && t.parameters.length > 0 && /* @__PURE__ */ $.jsx($.Fragment, { children: Rs.flatMap(
              (f) => {
                var y;
                return (y = a[f]) != null && y.length ? /* @__PURE__ */ $.jsx(
                  vo,
                  {
                    summary: t.summary ?? void 0,
                    id: t.slug,
                    parameters: a[f],
                    group: f
                  },
                  f
                ) : [];
              }
            ) }),
            yo((l = (i = (o = t.requestBody) == null ? void 0 : o.content) == null ? void 0 : i.at(0)) == null ? void 0 : l.schema, (f) => {
              var y;
              return /* @__PURE__ */ $.jsxs("div", { className: "mt-4 flex flex-col gap-4", children: [
                /* @__PURE__ */ $.jsxs(
                  Ie,
                  {
                    level: 3,
                    className: "capitalize flex items-center gap-2",
                    id: `${t.slug}/request-body`,
                    children: [
                      t.summary && /* @__PURE__ */ $.jsxs(pr, { children: [
                        t.summary,
                        " › "
                      ] }),
                      "Request Body",
                      " ",
                      ((y = t.requestBody) == null ? void 0 : y.required) === !1 ? /* @__PURE__ */ $.jsx(Fr, { variant: "muted", children: "optional" }) : ""
                    ]
                  }
                ),
                /* @__PURE__ */ $.jsx(Ir, { schema: f })
              ] });
            }),
            t.responses.length > 0 && /* @__PURE__ */ $.jsxs($.Fragment, { children: [
              /* @__PURE__ */ $.jsxs(
                Ie,
                {
                  level: 3,
                  className: "capitalize mt-8 pt-8 border-t",
                  id: `${t.slug}/responses`,
                  children: [
                    t.summary && /* @__PURE__ */ $.jsxs(pr, { children: [
                      t.summary,
                      " › "
                    ] }),
                    "Responses"
                  ]
                }
              ),
              /* @__PURE__ */ $.jsxs(
                to,
                {
                  onValueChange: (f) => r(f),
                  value: s,
                  children: [
                    t.responses.length > 1 && /* @__PURE__ */ $.jsx(ro, { children: t.responses.map((f) => /* @__PURE__ */ $.jsx(
                      no,
                      {
                        value: f.statusCode,
                        title: f.description ?? void 0,
                        children: f.statusCode
                      },
                      f.statusCode
                    )) }),
                    /* @__PURE__ */ $.jsx("ul", { className: "list-none m-0 px-0", children: t.responses.map((f) => {
                      var y, g;
                      return /* @__PURE__ */ $.jsx(
                        ao,
                        {
                          value: f.statusCode,
                          children: /* @__PURE__ */ $.jsx(
                            Ir,
                            {
                              schema: (g = (y = f.content) == null ? void 0 : y.find((m) => m.schema)) == null ? void 0 : g.schema
                            }
                          )
                        },
                        f.statusCode
                      );
                    }) })
                  ]
                }
              )
            ] })
          ] }),
          /* @__PURE__ */ $.jsx(
            Es,
            {
              selectedResponse: s,
              onSelectResponse: r,
              operation: t
            }
          )
        ]
      },
      t.operationId
    )
  ] });
};
function Kn(e, n = 160) {
  return e ? e.replace(/\[([^\]]+)\]\([^)]+\)/g, "$1").replace(/!\[.*?\]\(.*?\)/g, "").replace(/[_*`~]/g, "").replace(/^(?:>|\s*#+|-{3,}|\*{3,})/gm, "").replace(/[|>{}[\]]/g, "").replace(/\s+/g, " ").trim().substring(0, n).replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;").replace(/"/g, "&quot;").replace(/'/g, "&#039;") : "";
}
const Ns = _e(
  /* GraphQL */
  `
  fragment OperationsFragment on OperationItem {
    slug
    summary
    method
    description
    operationId
    contentTypes
    path
    deprecated
    extensions
    parameters {
      name
      in
      description
      required
      schema
      style
      explode
      examples {
        name
        description
        externalValue
        value
        summary
      }
    }
    requestBody {
      content {
        mediaType
        encoding {
          name
        }
        examples {
          name
          description
          externalValue
          value
          summary
        }
        schema
      }
      description
      required
    }
    responses {
      statusCode
      links
      description
      content {
        examples {
          name
          description
          externalValue
          value
          summary
        }
        mediaType
        encoding {
          name
        }
        schema
      }
    }
  }
`
), As = _e(
  /* GraphQL */
  `
  query SchemaWarmup($input: JSON!, $type: SchemaType!) {
    schema(input: $input, type: $type) {
      openapi
    }
  }
`
), Is = _e(
  /* GraphQL */
  `
  query OperationsForTag(
    $input: JSON!
    $type: SchemaType!
    $tag: String
    $untagged: Boolean
  ) {
    schema(input: $input, type: $type) {
      servers {
        url
      }
      description
      summary
      title
      url
      version
      tag(slug: $tag, untagged: $untagged) {
        name
        description
        operations {
          slug
          ...OperationsFragment
        }
        next {
          name
          slug
        }
        prev {
          name
          slug
        }
      }
    }
  }
`
), ol = ({
  tag: e,
  untagged: n
}) => {
  const { input: t, type: a, versions: c, version: s, options: r } = vr(), o = Qe(Is, {
    input: t,
    type: a,
    tag: e,
    untagged: n
  }), i = yr(o), {
    data: { schema: l }
  } = i, { selectedServer: f } = mr(l.servers), y = l.title, g = l.summary, m = l.description, x = La(), v = Qe(As, { input: t, type: a });
  if (qa({
    ...v,
    enabled: typeof window < "u",
    notifyOnChangeProps: []
  }), Ca(), !l.tag) return null;
  const { operations: u, next: h, prev: d, description: j } = l.tag, T = j ? Kn(j) : g || (m ? Kn(m) : void 0), p = Object.entries(c).length > 1, O = (r == null ? void 0 : r.showVersionSelect) === "always" || p && (r == null ? void 0 : r.showVersionSelect) !== "hide", b = {
    prev: d != null && d.name ? { to: `../${d.slug}`, label: d.name } : void 0,
    next: h ? {
      to: `../${h.slug ?? Ba}`,
      label: h.name ?? "Other endpoints"
    } : void 0
  };
  return /* @__PURE__ */ $.jsxs(
    "div",
    {
      className: "pt-[--padding-content-top]",
      "data-pagefind-filter": "section:openapi",
      "data-pagefind-meta": "section:openapi",
      children: [
        /* @__PURE__ */ $.jsxs(Ea, { children: [
          /* @__PURE__ */ $.jsx("title", { children: [l.tag.name, y].filter(Boolean).join(" - ") }),
          T && /* @__PURE__ */ $.jsx("meta", { name: "description", content: T })
        ] }),
        /* @__PURE__ */ $.jsxs("div", { className: "mb-8", children: [
          /* @__PURE__ */ $.jsxs(Zn, { className: "w-full", children: [
            /* @__PURE__ */ $.jsxs("div", { className: "flex flex-col gap-y-4 sm:flex-row justify-around items-start sm:items-end", children: [
              /* @__PURE__ */ $.jsxs("div", { className: "flex flex-col flex-1 gap-2", children: [
                /* @__PURE__ */ $.jsx(Ga, { children: y }),
                /* @__PURE__ */ $.jsxs(
                  Ie,
                  {
                    level: 1,
                    id: "description",
                    registerSidebarAnchor: !0,
                    className: "mb-0",
                    children: [
                      l.tag.name ?? "Other endpoints",
                      O && /* @__PURE__ */ $.jsxs("span", { className: "text-xl text-muted-foreground ml-1.5", children: [
                        " ",
                        "(",
                        s,
                        ")"
                      ] })
                    ]
                  }
                ),
                /* @__PURE__ */ $.jsx(po, {})
              ] }),
              /* @__PURE__ */ $.jsxs("div", { className: "flex flex-col gap-4 sm:items-end", children: [
                O && /* @__PURE__ */ $.jsxs(
                  Ha,
                  {
                    onValueChange: (q) => x(c[q]),
                    defaultValue: s,
                    disabled: !p,
                    children: [
                      /* @__PURE__ */ $.jsx(Ma, { className: "w-[180px]", children: /* @__PURE__ */ $.jsx(za, { placeholder: "Select version" }) }),
                      /* @__PURE__ */ $.jsx(Wa, { children: Object.entries(c).map(([q]) => /* @__PURE__ */ $.jsx(Ja, { value: q, children: q }, q)) })
                    ]
                  }
                ),
                l.description && /* @__PURE__ */ $.jsxs(ur, { className: "flex items-center gap-1 text-sm font-medium text-muted-foreground group", children: [
                  /* @__PURE__ */ $.jsx("span", { children: "API information" }),
                  /* @__PURE__ */ $.jsx(
                    Yn,
                    {
                      className: "group-data-[state=open]:hidden translate-y-px",
                      size: 14
                    }
                  ),
                  /* @__PURE__ */ $.jsx(
                    Ia,
                    {
                      className: "group-data-[state=closed]:hidden translate-y-px",
                      size: 13
                    }
                  )
                ] })
              ] })
            ] }),
            l.description && /* @__PURE__ */ $.jsx(ea, { className: "CollapsibleContent", children: /* @__PURE__ */ $.jsx(
              "div",
              {
                className: pe(
                  fr,
                  "pt-4 max-w-full prose-img:max-w-prose"
                ),
                children: /* @__PURE__ */ $.jsx(
                  De,
                  {
                    className: "border rounded bg-muted/25 border-border px-2.5 md:px-4",
                    content: l.description
                  }
                )
              }
            ) })
          ] }),
          j && /* @__PURE__ */ $.jsx(
            "div",
            {
              className: pe(
                fr,
                "my-4 max-w-full prose-img:max-w-prose"
              ),
              children: /* @__PURE__ */ $.jsx(De, { content: j })
            }
          )
        ] }),
        /* @__PURE__ */ $.jsx("hr", {}),
        /* @__PURE__ */ $.jsxs("div", { className: "px-6 mt-6 -mx-6 [content-visibility:auto]", children: [
          u.map((q) => /* @__PURE__ */ $.jsxs("div", { children: [
            /* @__PURE__ */ $.jsx(
              ks,
              {
                serverUrl: f,
                operationFragment: q
              }
            ),
            /* @__PURE__ */ $.jsx("hr", { className: "my-10" })
          ] }, q.slug)),
          /* @__PURE__ */ $.jsx(Da, { className: "mb-4", ...b })
        ] })
      ]
    }
  );
};
export {
  ol as OperationList,
  Ns as OperationsFragment
};
