// zudoku/lib/SchemaList-CsTIT1pi.js
import { j as e } from "./jsx-runtime-C5mzlN2N.js";
import { s as i } from "./index-LNp6rxyU.js";
import { d as j } from "./hook-pPrHCB6G.js";
import { ChevronsUpDownIcon as N, ChevronsDownUpIcon as y, ChevronRightIcon as C } from "lucide-react";
import { B as b } from "./Button-BBNrKpQd.js";
import { C as o, a as c, b as m, c as v } from "./index-Ba_X7EnR.js";
import { C as w } from "./CategoryHeading-D06WK_Wo.js";
import { H as l, M as S, P as z } from "./Markdown-C5h6bxbE.js";
import { T } from "./Toc-BK39DQvI.js";
import { c as V } from "./cn-qaFjX9_3.js";
import { u as A, S as H } from "./SchemaView-DCSWtnYr.js";
import { u as I } from "./context-Lrf2Y9bR.js";
const M = v(
  /* GraphQL */
  `
  query GetSchemas($input: JSON!, $type: SchemaType!) {
    schema(input: $input, type: $type) {
      title
      description
      summary
      components {
        schemas {
          name
          schema
          extensions
        }
      }
    }
  }
`
);
function J() {
  var n;
  const { input: d, type: p, versions: h, version: x, options: a } = I(), u = A(M, {
    input: d,
    type: p
  }), { data: t } = j(u), r = ((n = t.schema.components) == null ? void 0 : n.schemas) ?? [];
  if (!r.length)
    return /* @__PURE__ */ e.jsx("div", { children: "No schemas found" });
  const f = Object.entries(h).length > 1, g = (a == null ? void 0 : a.showVersionSelect) === "always" || f && (a == null ? void 0 : a.showVersionSelect) !== "hide";
  return /* @__PURE__ */ e.jsxs(
    "div",
    {
      className: "grid grid-cols-[--sidecar-grid-cols] gap-8 justify-between",
      "data-pagefind-filter": "section:openapi",
      "data-pagefind-meta": "section:openapi",
      children: [
        /* @__PURE__ */ e.jsxs("div", { className: "pt-[--padding-content-top] pb-[--padding-content-bottom]", children: [
          /* @__PURE__ */ e.jsxs(o, { className: "w-full", children: [
            /* @__PURE__ */ e.jsxs("div", { className: "flex flex-col gap-y-4 sm:flex-row justify-around items-start sm:items-end", children: [
              /* @__PURE__ */ e.jsxs("div", { className: "flex-1", children: [
                /* @__PURE__ */ e.jsx(w, { children: t.schema.title }),
                /* @__PURE__ */ e.jsxs(
                  l,
                  {
                    level: 1,
                    id: "schemas",
                    registerSidebarAnchor: !0,
                    className: "mb-0",
                    children: [
                      "Schemas",
                      g && /* @__PURE__ */ e.jsxs("span", { className: "text-xl text-muted-foreground ml-1.5", children: [
                        "(",
                        x,
                        ")"
                      ] })
                    ]
                  }
                )
              ] }),
              t.schema.description && /* @__PURE__ */ e.jsxs(c, { className: "flex items-center gap-1 text-sm font-medium text-muted-foreground group", children: [
                /* @__PURE__ */ e.jsx("span", { children: "API information" }),
                /* @__PURE__ */ e.jsx(
                  N,
                  {
                    className: "group-data-[state=open]:hidden translate-y-px",
                    size: 14
                  }
                ),
                /* @__PURE__ */ e.jsx(
                  y,
                  {
                    className: "group-data-[state=closed]:hidden translate-y-px",
                    size: 13
                  }
                )
              ] })
            ] }),
            t.schema.description && /* @__PURE__ */ e.jsx(m, { className: "CollapsibleContent", children: /* @__PURE__ */ e.jsx(
              "div",
              {
                className: V(
                  z,
                  "pt-4 max-w-full prose-img:max-w-prose"
                ),
                children: /* @__PURE__ */ e.jsx(
                  S,
                  {
                    className: "border rounded bg-muted/25 border-border px-2.5 md:px-4",
                    content: t.schema.description
                  }
                )
              }
            ) })
          ] }),
          /* @__PURE__ */ e.jsx("hr", { className: "my-8" }),
          /* @__PURE__ */ e.jsx("div", { className: "flex flex-col gap-y-5", children: r.map((s) => /* @__PURE__ */ e.jsxs(o, { className: "group", defaultOpen: !0, children: [
            /* @__PURE__ */ e.jsxs(
              l,
              {
                registerSidebarAnchor: !0,
                level: 2,
                className: "flex items-center gap-1 justify-between w-fit",
                id: i(s.name),
                children: [
                  s.name,
                  " ",
                  /* @__PURE__ */ e.jsx(c, { asChild: !0, children: /* @__PURE__ */ e.jsx(b, { variant: "ghost", size: "icon", className: "size-6", children: /* @__PURE__ */ e.jsx(
                    C,
                    {
                      size: 16,
                      className: "group-data-[state=open]:rotate-90 transition cursor-pointer"
                    }
                  ) }) })
                ]
              }
            ),
            /* @__PURE__ */ e.jsx(m, { className: "mt-4 CollapsibleContent", children: /* @__PURE__ */ e.jsx(H, { schema: s.schema }) })
          ] }, s.name)) })
        ] }),
        /* @__PURE__ */ e.jsx(
          T,
          {
            entries: r.map((s) => ({
              id: i(s.name),
              value: s.name,
              depth: 1
            }))
          }
        )
      ]
    }
  );
}
export {
  J as SchemaList
};
