// zudoku/lib/zudoku.plugin-search-pagefind.js
import { j as t } from "./jsx-runtime-C5mzlN2N.js";
import { C as k } from "./ClientOnly-E7hGysn1.js";
import { VisuallyHidden as v } from "@radix-ui/react-visually-hidden";
import { u as N, e as j, m as S, o as _, f as L } from "./hook-pPrHCB6G.js";
import { useRef as b, useLayoutEffect as T, useState as w } from "react";
import { B as E } from "./Button-BBNrKpQd.js";
import { C as R, a as p, b as x, c as F, d as I, e as q, f as P } from "./Callout-D3Ja4OPX.js";
import { b as $ } from "./Dialog-ByYz4ABw.js";
import { S as A } from "./RouteGuard-BZ_VsiXc.js";
import { BracketsIcon as D, FileTextIcon as O } from "lucide-react";
import { a as B, L as g } from "./chunk-KNED5TY2-BUPjb3LQ.js";
const U = async ({
  search: r,
  options: n,
  auth: s,
  context: o
}) => {
  const c = n.maxResults ?? 10, a = n.transformResults ?? (() => !0), e = [], i = z({
    search: r,
    transformFn: a,
    auth: s,
    context: o
  });
  for await (const l of i)
    if (e.push(l), e.length >= c) break;
  return e;
};
async function* z({
  search: r,
  transformFn: n,
  auth: s,
  context: o
}) {
  let a = 0;
  for (; a < r.results.length; ) {
    const e = r.results.slice(
      a,
      a + 5
    );
    a += e.length;
    const i = await Promise.all(e.map((l) => l.data()));
    for (const l of i) {
      const d = n({ result: l, auth: s, context: o });
      d !== !1 && (d === !0 || d == null ? yield l : yield d);
    }
  }
}
const G = (r, n) => {
  const s = r.weighted_locations.reduce(
    (c, a) => c + a.balanced_score,
    0
  );
  return n.weighted_locations.reduce(
    (c, a) => c + a.balanced_score,
    0
  ) - s;
}, y = "cursor-pointer border border-transparent data-[selected=true]:border-border", H = ({
  searchResults: r,
  searchTerm: n,
  onClose: s,
  maxSubResults: o = 4
}) => {
  const c = B(), a = b(null);
  return T(() => {
    requestIdleCallback(() => {
      var e;
      (e = a.current) == null || e.scrollTo({ top: 0 });
    });
  }, [n]), /* @__PURE__ */ t.jsxs(R, { className: "max-h-[450px]", ref: a, children: [
    n && r.length > 0 && /* @__PURE__ */ t.jsx(
      p,
      {
        className: "text-sm text-muted-foreground",
        heading: `${r.length} results for "${n}"`
      }
    ),
    n && r.map((e) => /* @__PURE__ */ t.jsxs(
      p,
      {
        children: [
          /* @__PURE__ */ t.jsx(
            x,
            {
              asChild: !0,
              value: `${e.meta.title}-${e.url}`,
              className: y,
              onSelect: () => {
                c(e.url), s();
              },
              children: /* @__PURE__ */ t.jsxs(g, { to: e.url, children: [
                e.meta.section === "openapi" ? /* @__PURE__ */ t.jsx(D, {}) : /* @__PURE__ */ t.jsx(O, {}),
                e.meta.title
              ] })
            }
          ),
          e.sub_results.sort(G).slice(0, o).map((i) => /* @__PURE__ */ t.jsx(
            x,
            {
              asChild: !0,
              value: `sub-${e.meta.title}-${i.url}`,
              className: y,
              onSelect: () => {
                c(i.url), s();
              },
              children: /* @__PURE__ */ t.jsx(g, { to: i.url, onClick: s, children: /* @__PURE__ */ t.jsxs("div", { className: "flex flex-col items-start gap-2 ms-2.5 ps-5 border-l border-muted-foreground/50", children: [
                /* @__PURE__ */ t.jsx("span", { className: "font-bold", children: i.title }),
                /* @__PURE__ */ t.jsx(
                  "span",
                  {
                    className: "text-[13px] [&_mark]:bg-primary [&_mark]:text-primary-foreground",
                    dangerouslySetInnerHTML: { __html: i.excerpt }
                  }
                )
              ] }) })
            },
            `sub-${e.meta.title}-${i.url}`
          ))
        ]
      },
      [e.meta.title ?? e.excerpt, e.url].join("-")
    ))
  ] });
}, u = {
  // Slightly lower than default because API docs tend to have repetitive terms (parameter names, HTTP methods, etc.)
  termFrequency: 0.8,
  // Lower than default because API documentation pages tend to be longer due to comprehensive endpoint documentation
  pageLength: 0.6,
  // Slightly higher than default because in technical documentation, exact matches should be prioritized
  termSimilarity: 1.2,
  // Slightly lower than default because API docs might have legitimate repetition of terms
  termSaturation: 1.2
}, K = (r) => import(
  /* @vite-ignore */
  L(r, "/pagefind/pagefind.js")
), V = (r) => {
  const {
    options: { basePath: n }
  } = j(), { data: s, ...o } = S({
    queryKey: ["pagefind", r.ranking],
    retry: !1,
    queryFn: async () => {
      var a, e, i, l;
      const c = await K(n);
      return await c.init(), await c.options({
        ranking: {
          termFrequency: ((a = r.ranking) == null ? void 0 : a.termFrequency) ?? u.termFrequency,
          pageLength: ((e = r.ranking) == null ? void 0 : e.pageLength) ?? u.pageLength,
          termSimilarity: ((i = r.ranking) == null ? void 0 : i.termSimilarity) ?? u.termSimilarity,
          termSaturation: ((l = r.ranking) == null ? void 0 : l.termSaturation) ?? u.termSaturation
        }
      }), c;
    },
    enabled: typeof window < "u"
  });
  return o.isError && o.error.message !== "NOT_BUILT_YET" && console.error(o.error), { ...o, pagefind: s };
}, Y = ({
  isOpen: r,
  onClose: n,
  options: s
}) => {
  const { pagefind: o, error: c, isError: a } = V(s), [e, i] = w(""), l = N(), d = j(), h = b(null), { data: C } = S({
    queryKey: ["pagefind-search", e, l.isAuthenticated],
    queryFn: async () => {
      const m = l.isAuthenticated ? void 0 : { not: { section: A } }, f = await (o == null ? void 0 : o.search(e, { filters: m }));
      return f ? U({ search: f, options: s, auth: l, context: d }) : [];
    },
    placeholderData: _,
    enabled: !!o && !!e
  });
  return /* @__PURE__ */ t.jsxs(
    F,
    {
      command: { shouldFilter: !1 },
      content: { className: "max-w-[750px]" },
      open: r,
      onOpenChange: n,
      children: [
        /* @__PURE__ */ t.jsx(v, { children: /* @__PURE__ */ t.jsx($, { children: "Search" }) }),
        /* @__PURE__ */ t.jsx(
          I,
          {
            ref: h,
            placeholder: "Search...",
            value: e,
            onValueChange: i,
            disabled: a
          }
        ),
        /* @__PURE__ */ t.jsx(q, { children: e ? /* @__PURE__ */ t.jsxs("div", { className: "flex flex-col items-center", children: [
          "No results found.",
          /* @__PURE__ */ t.jsx(
            E,
            {
              variant: "link",
              onClick: () => {
                var m;
                i(""), (m = h.current) == null || m.focus();
              },
              children: "Clear search"
            }
          )
        ] }) : "Start typing to search" }),
        a ? /* @__PURE__ */ t.jsx("div", { className: "p-4 text-sm", children: c.message === "NOT_BUILT_YET" ? /* @__PURE__ */ t.jsxs(P, { type: "info", children: [
          "Search is currently not available in development mode by default.",
          /* @__PURE__ */ t.jsx("br", {}),
          "To still use search in development, run ",
          /* @__PURE__ */ t.jsx("code", { children: "zudoku build" }),
          " ",
          "and copy the ",
          /* @__PURE__ */ t.jsx("code", { children: "dist/pagefind" }),
          " directory to your",
          " ",
          /* @__PURE__ */ t.jsx("code", { children: "public" }),
          " directory."
        ] }) : "An error occurred while loading search." }) : /* @__PURE__ */ t.jsx(
          H,
          {
            searchResults: C ?? [],
            searchTerm: e,
            onClose: n,
            maxSubResults: s.maxSubResults
          }
        )
      ]
    }
  );
}, se = (r) => ({
  renderSearch: ({ isOpen: n, onClose: s }) => /* @__PURE__ */ t.jsx(k, { children: /* @__PURE__ */ t.jsx(Y, { isOpen: n, onClose: s, options: r }) })
});
export {
  se as pagefindSearchPlugin
};
